// @zeplin/sdk/dist/models/asset-content.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToAssetContent = exports.transformAssetContentToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformAssetContentToJSON = function transformAssetContentToJSON(value) {
  return {
    url: value.url,
    format: value.format,
    density: value.density
  };
};

exports.transformAssetContentToJSON = transformAssetContentToJSON;

var transformJSONToAssetContent = function transformJSONToAssetContent(value) {
  return {
    url: value.url,
    format: value.format,
    density: value.density
  };
};
/**
 * 
 * @export
 * @interface AssetContent
 */


exports.transformJSONToAssetContent = transformJSONToAssetContent;
