// @zeplin/sdk/dist/models/blend-mode-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToBlendModeEnum = exports.transformBlendModeEnumToJSON = exports.BlendModeEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformBlendModeEnumToJSON = function transformBlendModeEnumToJSON(value) {
  return value;
};

exports.transformBlendModeEnumToJSON = transformBlendModeEnumToJSON;

var transformJSONToBlendModeEnum = function transformJSONToBlendModeEnum(value) {
  return value;
};
/**
 * 
 * @export
 * @enum {string}
 */


exports.transformJSONToBlendModeEnum = transformJSONToBlendModeEnum;
var BlendModeEnum;
exports.BlendModeEnum = BlendModeEnum;

(function (BlendModeEnum) {
  BlendModeEnum["NORMAL"] = "normal";
  BlendModeEnum["DARKEN"] = "darken";
  BlendModeEnum["MULTIPLY"] = "multiply";
  BlendModeEnum["COLOR_BURN"] = "color-burn";
  BlendModeEnum["LIGHTEN"] = "lighten";
  BlendModeEnum["SCREEN"] = "screen";
  BlendModeEnum["COLOR_DODGE"] = "color-dodge";
  BlendModeEnum["OVERLAY"] = "overlay";
  BlendModeEnum["SOFT_LIGHT"] = "soft-light";
  BlendModeEnum["HARD_LIGHT"] = "hard-light";
  BlendModeEnum["DIFFERENCE"] = "difference";
  BlendModeEnum["EXCLUSION"] = "exclusion";
  BlendModeEnum["HUE"] = "hue";
  BlendModeEnum["SATURATION"] = "saturation";
  BlendModeEnum["COLOR"] = "color";
  BlendModeEnum["LUMINOSITY"] = "luminosity";
  BlendModeEnum["SOURCE_IN"] = "source-in";
  BlendModeEnum["SOURCE_OUT"] = "source-out";
  BlendModeEnum["SOURCE_ATOP"] = "source-atop";
  BlendModeEnum["DESTINATION_OVER"] = "destination-over";
  BlendModeEnum["DESTINATION_IN"] = "destination-in";
  BlendModeEnum["DESTINATION_OUT"] = "destination-out";
  BlendModeEnum["DESTINATION_ATOP"] = "destination-atop";
  BlendModeEnum["DISSOLVE"] = "dissolve";
  BlendModeEnum["LINEAR_BURN"] = "linear-burn";
  BlendModeEnum["LINEAR_DODGE"] = "linear-dodge";
  BlendModeEnum["DARKER_COLOR"] = "darker-color";
  BlendModeEnum["LIGHTER_COLOR"] = "lighter-color";
  BlendModeEnum["VIVID_LIGHT"] = "vivid-light";
  BlendModeEnum["LINEAR_LIGHT"] = "linear-light";
  BlendModeEnum["PIN_LIGHT"] = "pin-light";
  BlendModeEnum["HARD_MIX"] = "hard-mix";
  BlendModeEnum["SUBTRACT"] = "subtract";
  BlendModeEnum["DIVIDE"] = "divide";
})(BlendModeEnum || (exports.BlendModeEnum = BlendModeEnum = {}));
