// @zeplin/sdk/dist/models/bounding-rectangle.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToBoundingRectangle = exports.transformBoundingRectangleToJSON = void 0;

var _layerPosition = require("./layer-position");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformBoundingRectangleToJSON = function transformBoundingRectangleToJSON(value) {
  return {
    width: value.width,
    height: value.height,
    x: value.x,
    y: value.y,
    absolute: (0, _layerPosition.transformLayerPositionToJSON)(value.absolute)
  };
};

exports.transformBoundingRectangleToJSON = transformBoundingRectangleToJSON;

var transformJSONToBoundingRectangle = function transformJSONToBoundingRectangle(value) {
  return {
    width: value.width,
    height: value.height,
    x: value.x,
    y: value.y,
    absolute: (0, _layerPosition.transformJSONToLayerPosition)(value.absolute)
  };
};
/**
 * 
 * @export
 * @interface BoundingRectangle
 */


exports.transformJSONToBoundingRectangle = transformJSONToBoundingRectangle;
