// @zeplin/sdk/dist/models/code.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToCode = exports.transformCodeToJSON = void 0;

var _languageEnum = require("./language-enum");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformCodeToJSON = function transformCodeToJSON(value) {
  return {
    snippet: value.snippet,
    language: value.language && (0, _languageEnum.transformLanguageEnumToJSON)(value.language)
  };
};

exports.transformCodeToJSON = transformCodeToJSON;

var transformJSONToCode = function transformJSONToCode(value) {
  return {
    snippet: value.snippet,
    language: value.language && (0, _languageEnum.transformJSONToLanguageEnum)(value.language)
  };
};
/**
 * 
 * @export
 * @interface Code
 */


exports.transformJSONToCode = transformJSONToCode;
