// @zeplin/sdk/dist/models/color-create-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToColorCreateBody = exports.transformColorCreateBodyToJSON = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformColorCreateBodyToJSON = function transformColorCreateBodyToJSON(value) {
  return {
    name: value.name,
    source_id: value.sourceId,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};

exports.transformColorCreateBodyToJSON = transformColorCreateBodyToJSON;

var transformJSONToColorCreateBody = function transformJSONToColorCreateBody(value) {
  return {
    name: value.name,
    sourceId: value.source_id,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};
/**
 * 
 * @export
 * @interface ColorCreateBody
 */


exports.transformJSONToColorCreateBody = transformJSONToColorCreateBody;
