// @zeplin/sdk/dist/models/color-stop.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToColorStop = exports.transformColorStopToJSON = void 0;

var _colorData = require("./color-data");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformColorStopToJSON = function transformColorStopToJSON(value) {
  return {
    color: (0, _colorData.transformColorDataToJSON)(value.color),
    position: value.position
  };
};

exports.transformColorStopToJSON = transformColorStopToJSON;

var transformJSONToColorStop = function transformJSONToColorStop(value) {
  return {
    color: (0, _colorData.transformJSONToColorData)(value.color),
    position: value.position
  };
};
/**
 * 
 * @export
 * @interface ColorStop
 */


exports.transformJSONToColorStop = transformJSONToColorStop;
