// @zeplin/sdk/dist/models/color-data.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToColorData = exports.transformColorDataToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformColorDataToJSON = function transformColorDataToJSON(value) {
  return {
    source_id: value.sourceId,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};

exports.transformColorDataToJSON = transformColorDataToJSON;

var transformJSONToColorData = function transformJSONToColorData(value) {
  return {
    sourceId: value.source_id,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};
/**
 * 
 * @export
 * @interface ColorData
 */


exports.transformJSONToColorData = transformJSONToColorData;
