// @zeplin/sdk/dist/models/color.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToColor = exports.transformColorToJSON = void 0;

require("core-js/modules/es.function.name.js");

var _resourceSource = require("./resource-source");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformColorToJSON = function transformColorToJSON(value) {
  return {
    id: value.id,
    created: value.created,
    source_id: value.sourceId,
    name: value.name,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a,
    source: value.source && (0, _resourceSource.transformResourceSourceToJSON)(value.source)
  };
};

exports.transformColorToJSON = transformColorToJSON;

var transformJSONToColor = function transformJSONToColor(value) {
  return {
    id: value.id,
    created: value.created,
    sourceId: value.source_id,
    name: value.name,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a,
    source: value.source && (0, _resourceSource.transformJSONToResourceSource)(value.source)
  };
};
/**
 * 
 * @export
 * @interface Color
 */


exports.transformJSONToColor = transformJSONToColor;
