// @zeplin/sdk/dist/models/component-variant-property.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponentVariantProperty = exports.transformComponentVariantPropertyToJSON = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentVariantPropertyToJSON = function transformComponentVariantPropertyToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    value: value.value
  };
};

exports.transformComponentVariantPropertyToJSON = transformComponentVariantPropertyToJSON;

var transformJSONToComponentVariantProperty = function transformJSONToComponentVariantProperty(value) {
  return {
    id: value.id,
    name: value.name,
    value: value.value
  };
};
/**
 * 
 * @export
 * @interface ComponentVariantProperty
 */


exports.transformJSONToComponentVariantProperty = transformJSONToComponentVariantProperty;
