// @zeplin/sdk/dist/models/component-section-variant-property.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponentSectionVariantProperty = exports.transformComponentSectionVariantPropertyToJSON = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.object.to-string.js");

require("core-js/modules/web.dom-collections.iterator.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentSectionVariantPropertyToJSON = function transformComponentSectionVariantPropertyToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    values: value.values
  };
};

exports.transformComponentSectionVariantPropertyToJSON = transformComponentSectionVariantPropertyToJSON;

var transformJSONToComponentSectionVariantProperty = function transformJSONToComponentSectionVariantProperty(value) {
  return {
    id: value.id,
    name: value.name,
    values: value.values
  };
};
/**
 * 
 * @export
 * @interface ComponentSectionVariantProperty
 */


exports.transformJSONToComponentSectionVariantProperty = transformJSONToComponentSectionVariantProperty;
