// @zeplin/sdk/dist/models/component.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponent = exports.transformComponentToJSON = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.array.map.js");

var _componentSectionReference = require("./component-section-reference");

var _componentVariantProperty = require("./component-variant-property");

var _componentVersion = require("./component-version");

var _resourceSource = require("./resource-source");

var _snapshotImage = require("./snapshot-image");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentToJSON = function transformComponentToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    image: (0, _snapshotImage.transformSnapshotImageToJSON)(value.image),
    created: value.created,
    updated: value.updated,
    section: value.section && (0, _componentSectionReference.transformComponentSectionReferenceToJSON)(value.section),
    variant_properties: value.variantProperties && value.variantProperties.map(_componentVariantProperty.transformComponentVariantPropertyToJSON),
    source: value.source && (0, _resourceSource.transformResourceSourceToJSON)(value.source),
    latest_version: value.latestVersion && (0, _componentVersion.transformComponentVersionToJSON)(value.latestVersion)
  };
};

exports.transformComponentToJSON = transformComponentToJSON;

var transformJSONToComponent = function transformJSONToComponent(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    image: (0, _snapshotImage.transformJSONToSnapshotImage)(value.image),
    created: value.created,
    updated: value.updated,
    section: value.section && (0, _componentSectionReference.transformJSONToComponentSectionReference)(value.section),
    variantProperties: value.variant_properties && value.variant_properties.map(_componentVariantProperty.transformJSONToComponentVariantProperty),
    source: value.source && (0, _resourceSource.transformJSONToResourceSource)(value.source),
    latestVersion: value.latest_version && (0, _componentVersion.transformJSONToComponentVersion)(value.latest_version)
  };
};
/**
 * 
 * @export
 * @interface Component
 */


exports.transformJSONToComponent = transformJSONToComponent;
