// @zeplin/sdk/dist/models/component-section.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponentSection = exports.transformComponentSectionToJSON = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.array.map.js");

var _componentGroup = require("./component-group");

var _componentSectionVariant = require("./component-section-variant");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentSectionToJSON = function transformComponentSectionToJSON(value) {
  return {
    id: value.id,
    created: value.created,
    name: value.name,
    description: value.description,
    groups: value.groups.map(_componentGroup.transformComponentGroupToJSON),
    variant: value.variant && (0, _componentSectionVariant.transformComponentSectionVariantToJSON)(value.variant)
  };
};

exports.transformComponentSectionToJSON = transformComponentSectionToJSON;

var transformJSONToComponentSection = function transformJSONToComponentSection(value) {
  return {
    id: value.id,
    created: value.created,
    name: value.name,
    description: value.description,
    groups: value.groups.map(_componentGroup.transformJSONToComponentGroup),
    variant: value.variant && (0, _componentSectionVariant.transformJSONToComponentSectionVariant)(value.variant)
  };
};
/**
 * 
 * @export
 * @interface ComponentSection
 */


exports.transformJSONToComponentSection = transformJSONToComponentSection;
