// @zeplin/sdk/dist/models/component-section-variant.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponentSectionVariant = exports.transformComponentSectionVariantToJSON = void 0;

require("core-js/modules/es.array.map.js");

var _componentSectionVariantProperty = require("./component-section-variant-property");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentSectionVariantToJSON = function transformComponentSectionVariantToJSON(value) {
  return {
    properties: value.properties.map(_componentSectionVariantProperty.transformComponentSectionVariantPropertyToJSON),
    source_id: value.sourceId
  };
};

exports.transformComponentSectionVariantToJSON = transformComponentSectionVariantToJSON;

var transformJSONToComponentSectionVariant = function transformJSONToComponentSectionVariant(value) {
  return {
    properties: value.properties.map(_componentSectionVariantProperty.transformJSONToComponentSectionVariantProperty),
    sourceId: value.source_id
  };
};
/**
 * Variant information for this component section
 * @export
 * @interface ComponentSectionVariant
 */


exports.transformJSONToComponentSectionVariant = transformJSONToComponentSectionVariant;
