// @zeplin/sdk/dist/models/component-section-reference.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponentSectionReference = exports.transformComponentSectionReferenceToJSON = void 0;

var _entityReference = require("./entity-reference");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentSectionReferenceToJSON = function transformComponentSectionReferenceToJSON(value) {
  return {
    id: value.id,
    group: value.group && (0, _entityReference.transformEntityReferenceToJSON)(value.group)
  };
};

exports.transformComponentSectionReferenceToJSON = transformComponentSectionReferenceToJSON;

var transformJSONToComponentSectionReference = function transformJSONToComponentSectionReference(value) {
  return {
    id: value.id,
    group: value.group && (0, _entityReference.transformJSONToEntityReference)(value.group)
  };
};
/**
 * 
 * @export
 * @interface ComponentSectionReference
 */


exports.transformJSONToComponentSectionReference = transformJSONToComponentSectionReference;
