// @zeplin/sdk/dist/models/component-group.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToComponentGroup = exports.transformComponentGroupToJSON = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformComponentGroupToJSON = function transformComponentGroupToJSON(value) {
  return {
    id: value.id,
    created: value.created,
    name: value.name,
    description: value.description
  };
};

exports.transformComponentGroupToJSON = transformComponentGroupToJSON;

var transformJSONToComponentGroup = function transformJSONToComponentGroup(value) {
  return {
    id: value.id,
    created: value.created,
    name: value.name,
    description: value.description
  };
};
/**
 * 
 * @export
 * @interface ComponentGroup
 */


exports.transformJSONToComponentGroup = transformJSONToComponentGroup;
