// @zeplin/sdk/dist/models/connected-component.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToConnectedComponent = exports.transformConnectedComponentToJSON = void 0;

require("core-js/modules/es.array.map.js");

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

var _code = require("./code");

var _connectedComponentLink = require("./connected-component-link");

var _entityReference = require("./entity-reference");

var _resourceSource = require("./resource-source");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformConnectedComponentToJSON = function transformConnectedComponentToJSON(value) {
  return {
    components: value.components.map(_entityReference.transformEntityReferenceToJSON),
    file_path: value.filePath,
    name: value.name,
    description: value.description,
    code: value.code && (0, _code.transformCodeToJSON)(value.code),
    links: value.links.map(_connectedComponentLink.transformConnectedComponentLinkToJSON),
    source: value.source && (0, _resourceSource.transformResourceSourceToJSON)(value.source)
  };
};

exports.transformConnectedComponentToJSON = transformConnectedComponentToJSON;

var transformJSONToConnectedComponent = function transformJSONToConnectedComponent(value) {
  return {
    components: value.components.map(_entityReference.transformJSONToEntityReference),
    filePath: value.file_path,
    name: value.name,
    description: value.description,
    code: value.code && (0, _code.transformJSONToCode)(value.code),
    links: value.links.map(_connectedComponentLink.transformJSONToConnectedComponentLink),
    source: value.source && (0, _resourceSource.transformJSONToResourceSource)(value.source)
  };
};
/**
 * 
 * @export
 * @interface ConnectedComponent
 */


exports.transformJSONToConnectedComponent = transformJSONToConnectedComponent;
