// @zeplin/sdk/dist/models/connected-component-link.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToConnectedComponentLink = exports.transformConnectedComponentLinkToJSON = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformConnectedComponentLinkToJSON = function transformConnectedComponentLinkToJSON(value) {
  return {
    type: value.type,
    url: value.url,
    name: value.name
  };
};

exports.transformConnectedComponentLinkToJSON = transformConnectedComponentLinkToJSON;

var transformJSONToConnectedComponentLink = function transformJSONToConnectedComponentLink(value) {
  return {
    type: value.type,
    url: value.url,
    name: value.name
  };
};
/**
 * 
 * @export
 * @interface ConnectedComponentLink
 */


exports.transformJSONToConnectedComponentLink = transformJSONToConnectedComponentLink;
