// @zeplin/sdk/dist/models/design-token-font.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToDesignTokenFont = exports.transformDesignTokenFontToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformDesignTokenFontToJSON = function transformDesignTokenFontToJSON(value) {
  return {
    family: value.family,
    size: value.size,
    weight: value.weight,
    stretch: value.stretch
  };
};

exports.transformDesignTokenFontToJSON = transformDesignTokenFontToJSON;

var transformJSONToDesignTokenFont = function transformJSONToDesignTokenFont(value) {
  return {
    family: value.family,
    size: value.size,
    weight: value.weight,
    stretch: value.stretch
  };
};
/**
 * Font information for the text style
 * @export
 * @interface DesignTokenFont
 */


exports.transformJSONToDesignTokenFont = transformJSONToDesignTokenFont;
