// @zeplin/sdk/dist/models/design-token-source.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToDesignTokenSource = exports.transformDesignTokenSourceToJSON = void 0;

var _designTokenProject = require("./design-token-project");

var _designTokenStyleguide = require("./design-token-styleguide");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformDesignTokenSourceToJSON = function transformDesignTokenSourceToJSON(value) {
  return {
    project: value.project && (0, _designTokenProject.transformDesignTokenProjectToJSON)(value.project),
    styleguide: value.styleguide && (0, _designTokenStyleguide.transformDesignTokenStyleguideToJSON)(value.styleguide)
  };
};

exports.transformDesignTokenSourceToJSON = transformDesignTokenSourceToJSON;

var transformJSONToDesignTokenSource = function transformJSONToDesignTokenSource(value) {
  return {
    project: value.project && (0, _designTokenProject.transformJSONToDesignTokenProject)(value.project),
    styleguide: value.styleguide && (0, _designTokenStyleguide.transformJSONToDesignTokenStyleguide)(value.styleguide)
  };
};
/**
 * Source details for design token. It has to be either `project` or `styleguide`.
 * @export
 * @interface DesignTokenSource
 */


exports.transformJSONToDesignTokenSource = transformJSONToDesignTokenSource;
