// @zeplin/sdk/dist/models/design-token-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToDesignTokenResource = exports.transformDesignTokenResourceToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformDesignTokenResourceToJSON = function transformDesignTokenResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformDesignTokenResourceToJSON = transformDesignTokenResourceToJSON;

var transformJSONToDesignTokenResource = function transformJSONToDesignTokenResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * Details of the resource that the design token is generated from.
 * @export
 * @interface DesignTokenResource
 */


exports.transformJSONToDesignTokenResource = transformJSONToDesignTokenResource;
