// @zeplin/sdk/dist/models/design-token-project.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToDesignTokenProject = exports.transformDesignTokenProjectToJSON = void 0;

require("core-js/modules/es.function.name.js");

var _entityReference = require("./entity-reference");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformDesignTokenProjectToJSON = function transformDesignTokenProjectToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    platform: value.platform,
    linked_styleguide: value.linkedStyleguide && (0, _entityReference.transformEntityReferenceToJSON)(value.linkedStyleguide)
  };
};

exports.transformDesignTokenProjectToJSON = transformDesignTokenProjectToJSON;

var transformJSONToDesignTokenProject = function transformJSONToDesignTokenProject(value) {
  return {
    id: value.id,
    name: value.name,
    platform: value.platform,
    linkedStyleguide: value.linked_styleguide && (0, _entityReference.transformJSONToEntityReference)(value.linked_styleguide)
  };
};
/**
 * The source project of the token
 * @export
 * @interface DesignTokenProject
 */


exports.transformJSONToDesignTokenProject = transformJSONToDesignTokenProject;
