// @zeplin/sdk/dist/models/enabled-rem-preferences.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToEnabledRemPreferences = exports.transformEnabledRemPreferencesToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformEnabledRemPreferencesToJSON = function transformEnabledRemPreferencesToJSON(value) {
  return {
    status: value.status,
    root_font_size: value.rootFontSize,
    use_for_font_sizes: value.useForFontSizes,
    use_for_measurements: value.useForMeasurements
  };
};

exports.transformEnabledRemPreferencesToJSON = transformEnabledRemPreferencesToJSON;

var transformJSONToEnabledRemPreferences = function transformJSONToEnabledRemPreferences(value) {
  return {
    status: value.status,
    rootFontSize: value.root_font_size,
    useForFontSizes: value.use_for_font_sizes,
    useForMeasurements: value.use_for_measurements
  };
};
/**
 * 
 * @export
 * @interface EnabledRemPreferences
 */


exports.transformJSONToEnabledRemPreferences = transformJSONToEnabledRemPreferences;
