// @zeplin/sdk/dist/models/error-response.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToErrorResponse = exports.transformErrorResponseToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformErrorResponseToJSON = function transformErrorResponseToJSON(value) {
  return {
    message: value.message,
    detail: value.detail,
    code: value.code
  };
};

exports.transformErrorResponseToJSON = transformErrorResponseToJSON;

var transformJSONToErrorResponse = function transformJSONToErrorResponse(value) {
  return {
    message: value.message,
    detail: value.detail,
    code: value.code
  };
};
/**
 * 
 * @export
 * @interface ErrorResponse
 */


exports.transformJSONToErrorResponse = transformJSONToErrorResponse;
