// @zeplin/sdk/dist/models/flow-board-position.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardPosition = exports.transformFlowBoardPositionToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardPositionToJSON = function transformFlowBoardPositionToJSON(value) {
  return {
    x: value.x,
    y: value.y
  };
};

exports.transformFlowBoardPositionToJSON = transformFlowBoardPositionToJSON;

var transformJSONToFlowBoardPosition = function transformJSONToFlowBoardPosition(value) {
  return {
    x: value.x,
    y: value.y
  };
};
/**
 * 
 * @export
 * @interface FlowBoardPosition
 */


exports.transformJSONToFlowBoardPosition = transformJSONToFlowBoardPosition;
