// @zeplin/sdk/dist/models/flow-board-group.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardGroup = exports.transformFlowBoardGroupToJSON = void 0;

require("core-js/modules/es.function.name.js");

var _colorData = require("./color-data");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardGroupToJSON = function transformFlowBoardGroupToJSON(value) {
  return {
    id: value.id,
    created: value.created,
    name: value.name,
    color: (0, _colorData.transformColorDataToJSON)(value.color)
  };
};

exports.transformFlowBoardGroupToJSON = transformFlowBoardGroupToJSON;

var transformJSONToFlowBoardGroup = function transformJSONToFlowBoardGroup(value) {
  return {
    id: value.id,
    created: value.created,
    name: value.name,
    color: (0, _colorData.transformJSONToColorData)(value.color)
  };
};
/**
 * 
 * @export
 * @interface FlowBoardGroup
 */


exports.transformJSONToFlowBoardGroup = transformJSONToFlowBoardGroup;
