// @zeplin/sdk/dist/models/flow-board-screen-node.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardScreenNode = exports.transformFlowBoardScreenNodeToJSON = void 0;

var _entityReference = require("./entity-reference");

var _flowBoardPosition = require("./flow-board-position");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardScreenNodeToJSON = function transformFlowBoardScreenNodeToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    position: (0, _flowBoardPosition.transformFlowBoardPositionToJSON)(value.position),
    screen: (0, _entityReference.transformEntityReferenceToJSON)(value.screen)
  };
};

exports.transformFlowBoardScreenNodeToJSON = transformFlowBoardScreenNodeToJSON;

var transformJSONToFlowBoardScreenNode = function transformJSONToFlowBoardScreenNode(value) {
  return {
    id: value.id,
    type: value.type,
    position: (0, _flowBoardPosition.transformJSONToFlowBoardPosition)(value.position),
    screen: (0, _entityReference.transformJSONToEntityReference)(value.screen)
  };
};
/**
 * 
 * @export
 * @interface FlowBoardScreenNode
 */


exports.transformJSONToFlowBoardScreenNode = transformJSONToFlowBoardScreenNode;
