// @zeplin/sdk/dist/models/flow-board-connection.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardConnection = exports.transformFlowBoardConnectionToJSON = void 0;

var _entityReference = require("./entity-reference");

var _flowBoardConnectionPosition = require("./flow-board-connection-position");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardConnectionToJSON = function transformFlowBoardConnectionToJSON(value) {
  return {
    node: (0, _entityReference.transformEntityReferenceToJSON)(value.node),
    style: value.style,
    layer_source_id: value.layerSourceId,
    connection_position: value.connectionPosition && (0, _flowBoardConnectionPosition.transformFlowBoardConnectionPositionToJSON)(value.connectionPosition)
  };
};

exports.transformFlowBoardConnectionToJSON = transformFlowBoardConnectionToJSON;

var transformJSONToFlowBoardConnection = function transformJSONToFlowBoardConnection(value) {
  return {
    node: (0, _entityReference.transformJSONToEntityReference)(value.node),
    style: value.style,
    layerSourceId: value.layer_source_id,
    connectionPosition: value.connection_position && (0, _flowBoardConnectionPosition.transformJSONToFlowBoardConnectionPosition)(value.connection_position)
  };
};
/**
 * 
 * @export
 * @interface FlowBoardConnection
 */


exports.transformJSONToFlowBoardConnection = transformJSONToFlowBoardConnection;
