// @zeplin/sdk/dist/models/flow-board-connection-position.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardConnectionPosition = exports.transformFlowBoardConnectionPositionToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardConnectionPositionToJSON = function transformFlowBoardConnectionPositionToJSON(value) {
  return {
    dot_index: value.dotIndex,
    side: value.side
  };
};

exports.transformFlowBoardConnectionPositionToJSON = transformFlowBoardConnectionPositionToJSON;

var transformJSONToFlowBoardConnectionPosition = function transformJSONToFlowBoardConnectionPosition(value) {
  return {
    dotIndex: value.dot_index,
    side: value.side
  };
};
/**
 * 
 * @export
 * @interface FlowBoardConnectionPosition
 */


exports.transformJSONToFlowBoardConnectionPosition = transformJSONToFlowBoardConnectionPosition;
