// @zeplin/sdk/dist/models/flow-board-connector.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardConnector = exports.transformFlowBoardConnectorToJSON = void 0;

var _flowBoardConnection = require("./flow-board-connection");

var _flowBoardConnectorLabel = require("./flow-board-connector-label");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardConnectorToJSON = function transformFlowBoardConnectorToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    start: (0, _flowBoardConnection.transformFlowBoardConnectionToJSON)(value.start),
    end: (0, _flowBoardConnection.transformFlowBoardConnectionToJSON)(value.end),
    label: value.label && (0, _flowBoardConnectorLabel.transformFlowBoardConnectorLabelToJSON)(value.label),
    color: value.color
  };
};

exports.transformFlowBoardConnectorToJSON = transformFlowBoardConnectorToJSON;

var transformJSONToFlowBoardConnector = function transformJSONToFlowBoardConnector(value) {
  return {
    id: value.id,
    type: value.type,
    start: (0, _flowBoardConnection.transformJSONToFlowBoardConnection)(value.start),
    end: (0, _flowBoardConnection.transformJSONToFlowBoardConnection)(value.end),
    label: value.label && (0, _flowBoardConnectorLabel.transformJSONToFlowBoardConnectorLabel)(value.label),
    color: value.color
  };
};
/**
 * 
 * @export
 * @interface FlowBoardConnector
 */


exports.transformJSONToFlowBoardConnector = transformJSONToFlowBoardConnector;
