// @zeplin/sdk/dist/models/flow-board-text-node.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardTextNode = exports.transformFlowBoardTextNodeToJSON = void 0;

var _flowBoardPosition = require("./flow-board-position");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardTextNodeToJSON = function transformFlowBoardTextNodeToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    position: (0, _flowBoardPosition.transformFlowBoardPositionToJSON)(value.position),
    text: value.text,
    width: value.width
  };
};

exports.transformFlowBoardTextNodeToJSON = transformFlowBoardTextNodeToJSON;

var transformJSONToFlowBoardTextNode = function transformJSONToFlowBoardTextNode(value) {
  return {
    id: value.id,
    type: value.type,
    position: (0, _flowBoardPosition.transformJSONToFlowBoardPosition)(value.position),
    text: value.text,
    width: value.width
  };
};
/**
 * 
 * @export
 * @interface FlowBoardTextNode
 */


exports.transformJSONToFlowBoardTextNode = transformJSONToFlowBoardTextNode;
