// @zeplin/sdk/dist/models/flow-board-connector-label.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardConnectorLabel = exports.transformFlowBoardConnectorLabelToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardConnectorLabelToJSON = function transformFlowBoardConnectorLabelToJSON(value) {
  return {
    text: value.text,
    position: value.position,
    width: value.width
  };
};

exports.transformFlowBoardConnectorLabelToJSON = transformFlowBoardConnectorLabelToJSON;

var transformJSONToFlowBoardConnectorLabel = function transformJSONToFlowBoardConnectorLabel(value) {
  return {
    text: value.text,
    position: value.position,
    width: value.width
  };
};
/**
 * 
 * @export
 * @interface FlowBoardConnectorLabel
 */


exports.transformJSONToFlowBoardConnectorLabel = transformJSONToFlowBoardConnectorLabel;
