// @zeplin/sdk/dist/models/flow-board-variant-group-node.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoardVariantGroupNode = exports.transformFlowBoardVariantGroupNodeToJSON = void 0;

var _entityReference = require("./entity-reference");

var _flowBoardPosition = require("./flow-board-position");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardVariantGroupNodeToJSON = function transformFlowBoardVariantGroupNodeToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    position: (0, _flowBoardPosition.transformFlowBoardPositionToJSON)(value.position),
    variant_group: (0, _entityReference.transformEntityReferenceToJSON)(value.variantGroup),
    default_variant: (0, _entityReference.transformEntityReferenceToJSON)(value.defaultVariant)
  };
};

exports.transformFlowBoardVariantGroupNodeToJSON = transformFlowBoardVariantGroupNodeToJSON;

var transformJSONToFlowBoardVariantGroupNode = function transformJSONToFlowBoardVariantGroupNode(value) {
  return {
    id: value.id,
    type: value.type,
    position: (0, _flowBoardPosition.transformJSONToFlowBoardPosition)(value.position),
    variantGroup: (0, _entityReference.transformJSONToEntityReference)(value.variant_group),
    defaultVariant: (0, _entityReference.transformJSONToEntityReference)(value.default_variant)
  };
};
/**
 * 
 * @export
 * @interface FlowBoardVariantGroupNode
 */


exports.transformJSONToFlowBoardVariantGroupNode = transformJSONToFlowBoardVariantGroupNode;
