// @zeplin/sdk/dist/models/flow-board.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToFlowBoard = exports.transformFlowBoardToJSON = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformFlowBoardToJSON = function transformFlowBoardToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    number_of_connectors: value.numberOfConnectors,
    number_of_nodes: value.numberOfNodes,
    number_of_groups: value.numberOfGroups
  };
};

exports.transformFlowBoardToJSON = transformFlowBoardToJSON;

var transformJSONToFlowBoard = function transformJSONToFlowBoard(value) {
  return {
    id: value.id,
    name: value.name,
    numberOfConnectors: value.number_of_connectors,
    numberOfNodes: value.number_of_nodes,
    numberOfGroups: value.number_of_groups
  };
};
/**
 * 
 * @export
 * @interface FlowBoard
 */


exports.transformJSONToFlowBoard = transformJSONToFlowBoard;
