// @zeplin/sdk/dist/models/gradient.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToGradient = exports.transformGradientToJSON = void 0;

require("core-js/modules/es.array.map.js");

var _colorStop = require("./color-stop");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformGradientToJSON = function transformGradientToJSON(value) {
  return {
    type: value.type,
    angle: value.angle,
    scale: value.scale,
    color_stops: value.colorStops && value.colorStops.map(_colorStop.transformColorStopToJSON),
    opacity: value.opacity
  };
};

exports.transformGradientToJSON = transformGradientToJSON;

var transformJSONToGradient = function transformJSONToGradient(value) {
  return {
    type: value.type,
    angle: value.angle,
    scale: value.scale,
    colorStops: value.color_stops && value.color_stops.map(_colorStop.transformJSONToColorStop),
    opacity: value.opacity
  };
};
/**
 * 
 * @export
 * @interface Gradient
 */


exports.transformJSONToGradient = transformJSONToGradient;
