// @zeplin/sdk/dist/models/grid.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToGrid = exports.transformGridToJSON = void 0;

var _horizontalGrid = require("./horizontal-grid");

var _verticalGrid = require("./vertical-grid");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformGridToJSON = function transformGridToJSON(value) {
  return {
    horizontal_offset: value.horizontalOffset,
    vertical_offset: value.verticalOffset,
    vertical: value.vertical && (0, _verticalGrid.transformVerticalGridToJSON)(value.vertical),
    horizontal: value.horizontal && (0, _horizontalGrid.transformHorizontalGridToJSON)(value.horizontal)
  };
};

exports.transformGridToJSON = transformGridToJSON;

var transformJSONToGrid = function transformJSONToGrid(value) {
  return {
    horizontalOffset: value.horizontal_offset,
    verticalOffset: value.vertical_offset,
    vertical: value.vertical && (0, _verticalGrid.transformJSONToVerticalGrid)(value.vertical),
    horizontal: value.horizontal && (0, _horizontalGrid.transformJSONToHorizontalGrid)(value.horizontal)
  };
};
/**
 * 
 * @export
 * @interface Grid
 */


exports.transformJSONToGrid = transformJSONToGrid;
