// @zeplin/sdk/dist/models/horizontal-grid.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToHorizontalGrid = exports.transformHorizontalGridToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformHorizontalGridToJSON = function transformHorizontalGridToJSON(value) {
  return {
    gutter_height: value.gutterHeight,
    row_height: value.rowHeight,
    number_of_rows: value.numberOfRows
  };
};

exports.transformHorizontalGridToJSON = transformHorizontalGridToJSON;

var transformJSONToHorizontalGrid = function transformJSONToHorizontalGrid(value) {
  return {
    gutterHeight: value.gutter_height,
    rowHeight: value.row_height,
    numberOfRows: value.number_of_rows
  };
};
/**
 * 
 * @export
 * @interface HorizontalGrid
 */


exports.transformJSONToHorizontalGrid = transformJSONToHorizontalGrid;
