// @zeplin/sdk/dist/models/hotspot-bounding-rectangle.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformJSONToHotspotBoundingRectangle = exports.transformHotspotBoundingRectangleToJSON = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformHotspotBoundingRectangleToJSON = function transformHotspotBoundingRectangleToJSON(value) {
  return {
    width: value.width,
    height: value.height,
    x: value.x,
    y: value.y
  };
};

exports.transformHotspotBoundingRectangleToJSON = transformHotspotBoundingRectangleToJSON;

var transformJSONToHotspotBoundingRectangle = function transformJSONToHotspotBoundingRectangle(value) {
  return {
    width: value.width,
    height: value.height,
    x: value.x,
    y: value.y
  };
};
/**
 * 
 * @export
 * @interface HotspotBoundingRectangle
 */


exports.transformJSONToHotspotBoundingRectangle = transformJSONToHotspotBoundingRectangle;
