// @zeplin/sdk/dist/models/language-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLanguageEnumToJSON = exports.transformJSONToLanguageEnum = exports.LanguageEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLanguageEnumToJSON = function transformLanguageEnumToJSON(value) {
  return value;
};

exports.transformLanguageEnumToJSON = transformLanguageEnumToJSON;

var transformJSONToLanguageEnum = function transformJSONToLanguageEnum(value) {
  return value;
};
/**
 * Programming language
 * @export
 * @enum {string}
 */


exports.transformJSONToLanguageEnum = transformJSONToLanguageEnum;
var LanguageEnum;
exports.LanguageEnum = LanguageEnum;

(function (LanguageEnum) {
  LanguageEnum["MARKUP"] = "markup";
  LanguageEnum["HTML"] = "html";
  LanguageEnum["XML"] = "xml";
  LanguageEnum["SVG"] = "svg";
  LanguageEnum["CSS"] = "css";
  LanguageEnum["CLIKE"] = "clike";
  LanguageEnum["JAVASCRIPT"] = "javascript";
  LanguageEnum["ACTIONSCRIPT"] = "actionscript";
  LanguageEnum["C"] = "c";
  LanguageEnum["CS"] = "cs";
  LanguageEnum["CPP"] = "cpp";
  LanguageEnum["COFFEESCRIPT"] = "coffeescript";
  LanguageEnum["DART"] = "dart";
  LanguageEnum["EJS"] = "ejs";
  LanguageEnum["ELM"] = "elm";
  LanguageEnum["ERB"] = "erb";
  LanguageEnum["FLOW"] = "flow";
  LanguageEnum["FTL"] = "ftl";
  LanguageEnum["HANDLEBARS"] = "handlebars";
  LanguageEnum["JAVA"] = "java";
  LanguageEnum["KOTLIN"] = "kotlin";
  LanguageEnum["LESS"] = "less";
  LanguageEnum["LIVESCRIPT"] = "livescript";
  LanguageEnum["MARKDOWN"] = "markdown";
  LanguageEnum["OBJECTIVEC"] = "objectivec";
  LanguageEnum["PHP"] = "php";
  LanguageEnum["PUG"] = "pug";
  LanguageEnum["PYTHON"] = "python";
  LanguageEnum["R"] = "r";
  LanguageEnum["JSX"] = "jsx";
  LanguageEnum["TSX"] = "tsx";
  LanguageEnum["SASS"] = "sass";
  LanguageEnum["SCSS"] = "scss";
  LanguageEnum["SCALA"] = "scala";
  LanguageEnum["SWIFT"] = "swift";
  LanguageEnum["TS"] = "ts";
  LanguageEnum["VELOCITY"] = "velocity";
  LanguageEnum["YAML"] = "yaml";
})(LanguageEnum || (exports.LanguageEnum = LanguageEnum = {}));
