// @zeplin/sdk/dist/models/layer-shadow.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLayerShadowToJSON = exports.transformJSONToLayerShadow = void 0;

var _colorData = require("./color-data");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLayerShadowToJSON = function transformLayerShadowToJSON(value) {
  return {
    type: value.type,
    offset_x: value.offsetX,
    offset_y: value.offsetY,
    blur_radius: value.blurRadius,
    spread: value.spread,
    color: value.color && (0, _colorData.transformColorDataToJSON)(value.color)
  };
};

exports.transformLayerShadowToJSON = transformLayerShadowToJSON;

var transformJSONToLayerShadow = function transformJSONToLayerShadow(value) {
  return {
    type: value.type,
    offsetX: value.offset_x,
    offsetY: value.offset_y,
    blurRadius: value.blur_radius,
    spread: value.spread,
    color: value.color && (0, _colorData.transformJSONToColorData)(value.color)
  };
};
/**
 * 
 * @export
 * @interface LayerShadow
 */


exports.transformJSONToLayerShadow = transformJSONToLayerShadow;
