// @zeplin/sdk/dist/models/layer-text-style.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLayerTextStyleToJSON = exports.transformJSONToLayerTextStyle = void 0;

var _layerTextStyleRange = require("./layer-text-style-range");

var _textStyleData = require("./text-style-data");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLayerTextStyleToJSON = function transformLayerTextStyleToJSON(value) {
  return {
    range: value.range && (0, _layerTextStyleRange.transformLayerTextStyleRangeToJSON)(value.range),
    style: value.style && (0, _textStyleData.transformTextStyleDataToJSON)(value.style)
  };
};

exports.transformLayerTextStyleToJSON = transformLayerTextStyleToJSON;

var transformJSONToLayerTextStyle = function transformJSONToLayerTextStyle(value) {
  return {
    range: value.range && (0, _layerTextStyleRange.transformJSONToLayerTextStyleRange)(value.range),
    style: value.style && (0, _textStyleData.transformJSONToTextStyleData)(value.style)
  };
};
/**
 * 
 * @export
 * @interface LayerTextStyle
 */


exports.transformJSONToLayerTextStyle = transformJSONToLayerTextStyle;
