// @zeplin/sdk/dist/models/layer-text-style-range.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLayerTextStyleRangeToJSON = exports.transformJSONToLayerTextStyleRange = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLayerTextStyleRangeToJSON = function transformLayerTextStyleRangeToJSON(value) {
  return {
    location: value.location,
    length: value.length
  };
};

exports.transformLayerTextStyleRangeToJSON = transformLayerTextStyleRangeToJSON;

var transformJSONToLayerTextStyleRange = function transformJSONToLayerTextStyleRange(value) {
  return {
    location: value.location,
    length: value.length
  };
};
/**
 * 
 * @export
 * @interface LayerTextStyleRange
 */


exports.transformJSONToLayerTextStyleRange = transformJSONToLayerTextStyleRange;
