// @zeplin/sdk/dist/models/layer-fill.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLayerFillToJSON = exports.transformJSONToLayerFill = void 0;

var _blendModeEnum = require("./blend-mode-enum");

var _colorData = require("./color-data");

var _gradient = require("./gradient");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLayerFillToJSON = function transformLayerFillToJSON(value) {
  return {
    type: value.type,
    color: value.color && (0, _colorData.transformColorDataToJSON)(value.color),
    gradient: value.gradient && (0, _gradient.transformGradientToJSON)(value.gradient),
    blend_mode: value.blendMode && (0, _blendModeEnum.transformBlendModeEnumToJSON)(value.blendMode)
  };
};

exports.transformLayerFillToJSON = transformLayerFillToJSON;

var transformJSONToLayerFill = function transformJSONToLayerFill(value) {
  return {
    type: value.type,
    color: value.color && (0, _colorData.transformJSONToColorData)(value.color),
    gradient: value.gradient && (0, _gradient.transformJSONToGradient)(value.gradient),
    blendMode: value.blend_mode && (0, _blendModeEnum.transformJSONToBlendModeEnum)(value.blend_mode)
  };
};
/**
 * 
 * @export
 * @interface LayerFill
 */


exports.transformJSONToLayerFill = transformJSONToLayerFill;
