// @zeplin/sdk/dist/models/layer-border.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLayerBorderToJSON = exports.transformJSONToLayerBorder = void 0;

require("core-js/modules/es.array.fill.js");

var _layerFill = require("./layer-fill");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLayerBorderToJSON = function transformLayerBorderToJSON(value) {
  return {
    position: value.position,
    thickness: value.thickness,
    fill: value.fill && (0, _layerFill.transformLayerFillToJSON)(value.fill)
  };
};

exports.transformLayerBorderToJSON = transformLayerBorderToJSON;

var transformJSONToLayerBorder = function transformJSONToLayerBorder(value) {
  return {
    position: value.position,
    thickness: value.thickness,
    fill: value.fill && (0, _layerFill.transformJSONToLayerFill)(value.fill)
  };
};
/**
 * 
 * @export
 * @interface LayerBorder
 */


exports.transformJSONToLayerBorder = transformJSONToLayerBorder;
