// @zeplin/sdk/dist/models/link.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLinkToJSON = exports.transformJSONToLink = void 0;

var _hotspotBoundingRectangle = require("./hotspot-bounding-rectangle");

var _linkDestination = require("./link-destination");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLinkToJSON = function transformLinkToJSON(value) {
  return {
    rect: (0, _hotspotBoundingRectangle.transformHotspotBoundingRectangleToJSON)(value.rect),
    destination: (0, _linkDestination.transformLinkDestinationToJSON)(value.destination)
  };
};

exports.transformLinkToJSON = transformLinkToJSON;

var transformJSONToLink = function transformJSONToLink(value) {
  return {
    rect: (0, _hotspotBoundingRectangle.transformJSONToHotspotBoundingRectangle)(value.rect),
    destination: (0, _linkDestination.transformJSONToLinkDestination)(value.destination)
  };
};
/**
 * 
 * @export
 * @interface Link
 */


exports.transformJSONToLink = transformJSONToLink;
