// @zeplin/sdk/dist/models/link-destination.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformLinkDestinationToJSON = exports.transformJSONToLinkDestination = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformLinkDestinationToJSON = function transformLinkDestinationToJSON(value) {
  return {
    name: value.name,
    type: value.type
  };
};

exports.transformLinkDestinationToJSON = transformLinkDestinationToJSON;

var transformJSONToLinkDestination = function transformJSONToLinkDestination(value) {
  return {
    name: value.name,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface LinkDestination
 */


exports.transformJSONToLinkDestination = transformJSONToLinkDestination;
