// @zeplin/sdk/dist/models/notification-update-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformNotificationUpdateBodyToJSON = exports.transformJSONToNotificationUpdateBody = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformNotificationUpdateBodyToJSON = function transformNotificationUpdateBodyToJSON(value) {
  return {
    is_read: value.isRead
  };
};

exports.transformNotificationUpdateBodyToJSON = transformNotificationUpdateBodyToJSON;

var transformJSONToNotificationUpdateBody = function transformJSONToNotificationUpdateBody(value) {
  return {
    isRead: value.is_read
  };
};
/**
 * 
 * @export
 * @interface NotificationUpdateBody
 */


exports.transformJSONToNotificationUpdateBody = transformJSONToNotificationUpdateBody;
