// @zeplin/sdk/dist/models/organization-member.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformOrganizationMemberToJSON = exports.transformJSONToOrganizationMember = void 0;

var _user = require("./user");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformOrganizationMemberToJSON = function transformOrganizationMemberToJSON(value) {
  return {
    user: (0, _user.transformUserToJSON)(value.user),
    tags: value.tags,
    role: value.role,
    restricted: value.restricted,
    invited: value.invited
  };
};

exports.transformOrganizationMemberToJSON = transformOrganizationMemberToJSON;

var transformJSONToOrganizationMember = function transformJSONToOrganizationMember(value) {
  return {
    user: (0, _user.transformJSONToUser)(value.user),
    tags: value.tags,
    role: value.role,
    restricted: value.restricted,
    invited: value.invited
  };
};
/**
 * 
 * @export
 * @interface OrganizationMember
 */


exports.transformJSONToOrganizationMember = transformJSONToOrganizationMember;
