// @zeplin/sdk/dist/models/organization-billing.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformOrganizationBillingToJSON = exports.transformJSONToOrganizationBilling = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformOrganizationBillingToJSON = function transformOrganizationBillingToJSON(value) {
  return {
    total_seat_count: value.totalSeatCount,
    used_seat_count: value.usedSeatCount
  };
};

exports.transformOrganizationBillingToJSON = transformOrganizationBillingToJSON;

var transformJSONToOrganizationBilling = function transformJSONToOrganizationBilling(value) {
  return {
    totalSeatCount: value.total_seat_count,
    usedSeatCount: value.used_seat_count
  };
};
/**
 * 
 * @export
 * @interface OrganizationBilling
 */


exports.transformJSONToOrganizationBilling = transformJSONToOrganizationBilling;
