// @zeplin/sdk/dist/models/organization-summary.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformOrganizationSummaryToJSON = exports.transformJSONToOrganizationSummary = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformOrganizationSummaryToJSON = function transformOrganizationSummaryToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    logo: value.logo
  };
};

exports.transformOrganizationSummaryToJSON = transformOrganizationSummaryToJSON;

var transformJSONToOrganizationSummary = function transformJSONToOrganizationSummary(value) {
  return {
    id: value.id,
    name: value.name,
    logo: value.logo
  };
};
/**
 * 
 * @export
 * @interface OrganizationSummary
 */


exports.transformJSONToOrganizationSummary = transformJSONToOrganizationSummary;
