// @zeplin/sdk/dist/models/organization.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformOrganizationToJSON = exports.transformJSONToOrganization = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.array.map.js");

var _organizationMember = require("./organization-member");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformOrganizationToJSON = function transformOrganizationToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    logo: value.logo,
    members: value.members && value.members.map(_organizationMember.transformOrganizationMemberToJSON)
  };
};

exports.transformOrganizationToJSON = transformOrganizationToJSON;

var transformJSONToOrganization = function transformJSONToOrganization(value) {
  return {
    id: value.id,
    name: value.name,
    logo: value.logo,
    members: value.members && value.members.map(_organizationMember.transformJSONToOrganizationMember)
  };
};
/**
 * 
 * @export
 * @interface Organization
 */


exports.transformJSONToOrganization = transformJSONToOrganization;
