// @zeplin/sdk/dist/models/organization-webhook-event-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformOrganizationWebhookEventEnumToJSON = exports.transformJSONToOrganizationWebhookEventEnum = exports.OrganizationWebhookEventEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformOrganizationWebhookEventEnumToJSON = function transformOrganizationWebhookEventEnumToJSON(value) {
  return value;
};

exports.transformOrganizationWebhookEventEnumToJSON = transformOrganizationWebhookEventEnumToJSON;

var transformJSONToOrganizationWebhookEventEnum = function transformJSONToOrganizationWebhookEventEnum(value) {
  return value;
};
/**
 * 
 * @export
 * @enum {string}
 */


exports.transformJSONToOrganizationWebhookEventEnum = transformJSONToOrganizationWebhookEventEnum;
var OrganizationWebhookEventEnum;
exports.OrganizationWebhookEventEnum = OrganizationWebhookEventEnum;

(function (OrganizationWebhookEventEnum) {
  OrganizationWebhookEventEnum["STAR"] = "*";
  OrganizationWebhookEventEnum["PROJECT"] = "project.*";
  OrganizationWebhookEventEnum["PROJECT_COLOR"] = "project.color";
  OrganizationWebhookEventEnum["PROJECT_COMPONENT"] = "project.component";
  OrganizationWebhookEventEnum["PROJECT_MEMBER"] = "project.member";
  OrganizationWebhookEventEnum["PROJECT_NOTE"] = "project.note";
  OrganizationWebhookEventEnum["PROJECT_NOTE_COMMENT"] = "project.note.comment";
  OrganizationWebhookEventEnum["PROJECT_SCREEN"] = "project.screen";
  OrganizationWebhookEventEnum["PROJECT_SCREEN_VERSION"] = "project.screen.version";
  OrganizationWebhookEventEnum["PROJECT_SPACING_TOKEN"] = "project.spacing_token";
  OrganizationWebhookEventEnum["PROJECT_TEXT_STYLE"] = "project.text_style";
  OrganizationWebhookEventEnum["STYLEGUIDE"] = "styleguide.*";
  OrganizationWebhookEventEnum["STYLEGUIDE_COLOR"] = "styleguide.color";
  OrganizationWebhookEventEnum["STYLEGUIDE_COMPONENT"] = "styleguide.component";
  OrganizationWebhookEventEnum["STYLEGUIDE_MEMBER"] = "styleguide.member";
  OrganizationWebhookEventEnum["STYLEGUIDE_SPACING_TOKEN"] = "styleguide.spacing_token";
  OrganizationWebhookEventEnum["STYLEGUIDE_TEXT_STYLE"] = "styleguide.text_style";
  OrganizationWebhookEventEnum["WORKSPACE"] = "workspace.*";
  OrganizationWebhookEventEnum["WORKSPACE_PROJECT"] = "workspace.project";
  OrganizationWebhookEventEnum["WORKSPACE_STYLEGUIDE"] = "workspace.styleguide";
  OrganizationWebhookEventEnum["WORKSPACE_ORGANIZATION"] = "workspace.organization";
  OrganizationWebhookEventEnum["WORKSPACE_ORGANIZATION_MEMBER"] = "workspace.organization.member";
})(OrganizationWebhookEventEnum || (exports.OrganizationWebhookEventEnum = OrganizationWebhookEventEnum = {}));
