// @zeplin/sdk/dist/models/ping-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformPingEventToJSON = exports.transformJSONToPingEvent = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformPingEventToJSON = function transformPingEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: value.resource,
    context: value.context
  };
};

exports.transformPingEventToJSON = transformPingEventToJSON;

var transformJSONToPingEvent = function transformJSONToPingEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: value.resource,
    context: value.context
  };
};
/**
 * 
 * @export
 * @interface PingEvent
 */


exports.transformJSONToPingEvent = transformJSONToPingEvent;
