// @zeplin/sdk/dist/models/project-member-invited-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectMemberInvitedEventToJSON = exports.transformJSONToProjectMemberInvitedEvent = void 0;

var _projectMemberEventContext = require("./project-member-event-context");

var _projectMemberInvitedEventResource = require("./project-member-invited-event-resource");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectMemberInvitedEventToJSON = function transformProjectMemberInvitedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectMemberInvitedEventResource.transformProjectMemberInvitedEventResourceToJSON)(value.resource),
    context: (0, _projectMemberEventContext.transformProjectMemberEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectMemberInvitedEventToJSON = transformProjectMemberInvitedEventToJSON;

var transformJSONToProjectMemberInvitedEvent = function transformJSONToProjectMemberInvitedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectMemberInvitedEventResource.transformJSONToProjectMemberInvitedEventResource)(value.resource),
    context: (0, _projectMemberEventContext.transformJSONToProjectMemberEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectMemberInvitedEvent
 */


exports.transformJSONToProjectMemberInvitedEvent = transformJSONToProjectMemberInvitedEvent;
