// @zeplin/sdk/dist/models/project-screen-event-context.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenEventContextToJSON = exports.transformJSONToProjectScreenEventContext = void 0;

var _project = require("./project");

var _screenVersionSummary = require("./screen-version-summary");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenEventContextToJSON = function transformProjectScreenEventContextToJSON(value) {
  return {
    project: (0, _project.transformProjectToJSON)(value.project),
    version: (0, _screenVersionSummary.transformScreenVersionSummaryToJSON)(value.version)
  };
};

exports.transformProjectScreenEventContextToJSON = transformProjectScreenEventContextToJSON;

var transformJSONToProjectScreenEventContext = function transformJSONToProjectScreenEventContext(value) {
  return {
    project: (0, _project.transformJSONToProject)(value.project),
    version: (0, _screenVersionSummary.transformJSONToScreenVersionSummary)(value.version)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenEventContext
 */


exports.transformJSONToProjectScreenEventContext = transformJSONToProjectScreenEventContext;
