// @zeplin/sdk/dist/models/project-screen-version-event-context.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenVersionEventContextToJSON = exports.transformJSONToProjectScreenVersionEventContext = void 0;

var _project = require("./project");

var _screen = require("./screen");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenVersionEventContextToJSON = function transformProjectScreenVersionEventContextToJSON(value) {
  return {
    project: (0, _project.transformProjectToJSON)(value.project),
    screen: (0, _screen.transformScreenToJSON)(value.screen)
  };
};

exports.transformProjectScreenVersionEventContextToJSON = transformProjectScreenVersionEventContextToJSON;

var transformJSONToProjectScreenVersionEventContext = function transformJSONToProjectScreenVersionEventContext(value) {
  return {
    project: (0, _project.transformJSONToProject)(value.project),
    screen: (0, _screen.transformJSONToScreen)(value.screen)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenVersionEventContext
 */


exports.transformJSONToProjectScreenVersionEventContext = transformJSONToProjectScreenVersionEventContext;
