// @zeplin/sdk/dist/models/project-component-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectComponentCreatedEventResourceToJSON = exports.transformJSONToProjectComponentCreatedEventResource = void 0;

var _component = require("./component");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectComponentCreatedEventResourceToJSON = function transformProjectComponentCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _component.transformComponentToJSON)(value.data)
  };
};

exports.transformProjectComponentCreatedEventResourceToJSON = transformProjectComponentCreatedEventResourceToJSON;

var transformJSONToProjectComponentCreatedEventResource = function transformJSONToProjectComponentCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _component.transformJSONToComponent)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectComponentCreatedEventResource
 */


exports.transformJSONToProjectComponentCreatedEventResource = transformJSONToProjectComponentCreatedEventResource;
