// @zeplin/sdk/dist/models/project-member-removed-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectMemberRemovedEventToJSON = exports.transformJSONToProjectMemberRemovedEvent = void 0;

var _projectMemberEventContext = require("./project-member-event-context");

var _projectMemberRemovedEventResource = require("./project-member-removed-event-resource");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectMemberRemovedEventToJSON = function transformProjectMemberRemovedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectMemberRemovedEventResource.transformProjectMemberRemovedEventResourceToJSON)(value.resource),
    context: (0, _projectMemberEventContext.transformProjectMemberEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectMemberRemovedEventToJSON = transformProjectMemberRemovedEventToJSON;

var transformJSONToProjectMemberRemovedEvent = function transformJSONToProjectMemberRemovedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectMemberRemovedEventResource.transformJSONToProjectMemberRemovedEventResource)(value.resource),
    context: (0, _projectMemberEventContext.transformJSONToProjectMemberEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectMemberRemovedEvent
 */


exports.transformJSONToProjectMemberRemovedEvent = transformJSONToProjectMemberRemovedEvent;
