// @zeplin/sdk/dist/models/project-spacing-token-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectSpacingTokenCreatedEventResourceToJSON = exports.transformJSONToProjectSpacingTokenCreatedEventResource = void 0;

var _spacingToken = require("./spacing-token");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectSpacingTokenCreatedEventResourceToJSON = function transformProjectSpacingTokenCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _spacingToken.transformSpacingTokenToJSON)(value.data)
  };
};

exports.transformProjectSpacingTokenCreatedEventResourceToJSON = transformProjectSpacingTokenCreatedEventResourceToJSON;

var transformJSONToProjectSpacingTokenCreatedEventResource = function transformJSONToProjectSpacingTokenCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _spacingToken.transformJSONToSpacingToken)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectSpacingTokenCreatedEventResource
 */


exports.transformJSONToProjectSpacingTokenCreatedEventResource = transformJSONToProjectSpacingTokenCreatedEventResource;
