// @zeplin/sdk/dist/models/project-note-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteCreatedEventToJSON = exports.transformJSONToProjectNoteCreatedEvent = void 0;

var _projectNoteCreatedEventResource = require("./project-note-created-event-resource");

var _projectNoteEventContext = require("./project-note-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteCreatedEventToJSON = function transformProjectNoteCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCreatedEventResource.transformProjectNoteCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectNoteEventContext.transformProjectNoteEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectNoteCreatedEventToJSON = transformProjectNoteCreatedEventToJSON;

var transformJSONToProjectNoteCreatedEvent = function transformJSONToProjectNoteCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCreatedEventResource.transformJSONToProjectNoteCreatedEventResource)(value.resource),
    context: (0, _projectNoteEventContext.transformJSONToProjectNoteEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteCreatedEvent
 */


exports.transformJSONToProjectNoteCreatedEvent = transformJSONToProjectNoteCreatedEvent;
