// @zeplin/sdk/dist/models/project-note-comment-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteCommentCreatedEventToJSON = exports.transformJSONToProjectNoteCommentCreatedEvent = void 0;

var _projectNoteCommentCreatedEventResource = require("./project-note-comment-created-event-resource");

var _projectNoteCommentEventContext = require("./project-note-comment-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteCommentCreatedEventToJSON = function transformProjectNoteCommentCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCommentCreatedEventResource.transformProjectNoteCommentCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectNoteCommentEventContext.transformProjectNoteCommentEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectNoteCommentCreatedEventToJSON = transformProjectNoteCommentCreatedEventToJSON;

var transformJSONToProjectNoteCommentCreatedEvent = function transformJSONToProjectNoteCommentCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCommentCreatedEventResource.transformJSONToProjectNoteCommentCreatedEventResource)(value.resource),
    context: (0, _projectNoteCommentEventContext.transformJSONToProjectNoteCommentEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteCommentCreatedEvent
 */


exports.transformJSONToProjectNoteCommentCreatedEvent = transformJSONToProjectNoteCommentCreatedEvent;
