// @zeplin/sdk/dist/models/project-screen-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenCreatedEventToJSON = exports.transformJSONToProjectScreenCreatedEvent = void 0;

var _projectScreenCreatedEventResource = require("./project-screen-created-event-resource");

var _projectScreenEventContext = require("./project-screen-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenCreatedEventToJSON = function transformProjectScreenCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenCreatedEventResource.transformProjectScreenCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectScreenEventContext.transformProjectScreenEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectScreenCreatedEventToJSON = transformProjectScreenCreatedEventToJSON;

var transformJSONToProjectScreenCreatedEvent = function transformJSONToProjectScreenCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenCreatedEventResource.transformJSONToProjectScreenCreatedEventResource)(value.resource),
    context: (0, _projectScreenEventContext.transformJSONToProjectScreenEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenCreatedEvent
 */


exports.transformJSONToProjectScreenCreatedEvent = transformJSONToProjectScreenCreatedEvent;
