// @zeplin/sdk/dist/models/project-spacing-token-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectSpacingTokenCreatedEventToJSON = exports.transformJSONToProjectSpacingTokenCreatedEvent = void 0;

var _projectSpacingTokenCreatedEventResource = require("./project-spacing-token-created-event-resource");

var _projectSpacingTokenEventContext = require("./project-spacing-token-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectSpacingTokenCreatedEventToJSON = function transformProjectSpacingTokenCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectSpacingTokenCreatedEventResource.transformProjectSpacingTokenCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectSpacingTokenEventContext.transformProjectSpacingTokenEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectSpacingTokenCreatedEventToJSON = transformProjectSpacingTokenCreatedEventToJSON;

var transformJSONToProjectSpacingTokenCreatedEvent = function transformJSONToProjectSpacingTokenCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectSpacingTokenCreatedEventResource.transformJSONToProjectSpacingTokenCreatedEventResource)(value.resource),
    context: (0, _projectSpacingTokenEventContext.transformJSONToProjectSpacingTokenEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectSpacingTokenCreatedEvent
 */


exports.transformJSONToProjectSpacingTokenCreatedEvent = transformJSONToProjectSpacingTokenCreatedEvent;
