// @zeplin/sdk/dist/models/project-component-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectComponentUpdatedEventToJSON = exports.transformJSONToProjectComponentUpdatedEvent = void 0;

var _projectComponentCreatedEventResource = require("./project-component-created-event-resource");

var _projectComponentEventContext = require("./project-component-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectComponentUpdatedEventToJSON = function transformProjectComponentUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectComponentCreatedEventResource.transformProjectComponentCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectComponentEventContext.transformProjectComponentEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectComponentUpdatedEventToJSON = transformProjectComponentUpdatedEventToJSON;

var transformJSONToProjectComponentUpdatedEvent = function transformJSONToProjectComponentUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectComponentCreatedEventResource.transformJSONToProjectComponentCreatedEventResource)(value.resource),
    context: (0, _projectComponentEventContext.transformJSONToProjectComponentEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectComponentUpdatedEvent
 */


exports.transformJSONToProjectComponentUpdatedEvent = transformJSONToProjectComponentUpdatedEvent;
