// @zeplin/sdk/dist/models/project-member-invited-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectMemberInvitedEventResourceToJSON = exports.transformJSONToProjectMemberInvitedEventResource = void 0;

var _projectMember = require("./project-member");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectMemberInvitedEventResourceToJSON = function transformProjectMemberInvitedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _projectMember.transformProjectMemberToJSON)(value.data)
  };
};

exports.transformProjectMemberInvitedEventResourceToJSON = transformProjectMemberInvitedEventResourceToJSON;

var transformJSONToProjectMemberInvitedEventResource = function transformJSONToProjectMemberInvitedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _projectMember.transformJSONToProjectMember)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectMemberInvitedEventResource
 */


exports.transformJSONToProjectMemberInvitedEventResource = transformJSONToProjectMemberInvitedEventResource;
