// @zeplin/sdk/dist/models/project-note-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteCreatedEventResourceToJSON = exports.transformJSONToProjectNoteCreatedEventResource = void 0;

var _screenNote = require("./screen-note");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteCreatedEventResourceToJSON = function transformProjectNoteCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screenNote.transformScreenNoteToJSON)(value.data)
  };
};

exports.transformProjectNoteCreatedEventResourceToJSON = transformProjectNoteCreatedEventResourceToJSON;

var transformJSONToProjectNoteCreatedEventResource = function transformJSONToProjectNoteCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screenNote.transformJSONToScreenNote)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteCreatedEventResource
 */


exports.transformJSONToProjectNoteCreatedEventResource = transformJSONToProjectNoteCreatedEventResource;
