// @zeplin/sdk/dist/models/project-text-style-deleted-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectTextStyleDeletedEventResourceToJSON = exports.transformJSONToProjectTextStyleDeletedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectTextStyleDeletedEventResourceToJSON = function transformProjectTextStyleDeletedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformProjectTextStyleDeletedEventResourceToJSON = transformProjectTextStyleDeletedEventResourceToJSON;

var transformJSONToProjectTextStyleDeletedEventResource = function transformJSONToProjectTextStyleDeletedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface ProjectTextStyleDeletedEventResource
 */


exports.transformJSONToProjectTextStyleDeletedEventResource = transformJSONToProjectTextStyleDeletedEventResource;
