// @zeplin/sdk/dist/models/project-note-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteUpdatedEventToJSON = exports.transformJSONToProjectNoteUpdatedEvent = void 0;

var _projectNoteCreatedEventResource = require("./project-note-created-event-resource");

var _projectNoteEventContext = require("./project-note-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteUpdatedEventToJSON = function transformProjectNoteUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCreatedEventResource.transformProjectNoteCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectNoteEventContext.transformProjectNoteEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectNoteUpdatedEventToJSON = transformProjectNoteUpdatedEventToJSON;

var transformJSONToProjectNoteUpdatedEvent = function transformJSONToProjectNoteUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCreatedEventResource.transformJSONToProjectNoteCreatedEventResource)(value.resource),
    context: (0, _projectNoteEventContext.transformJSONToProjectNoteEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteUpdatedEvent
 */


exports.transformJSONToProjectNoteUpdatedEvent = transformJSONToProjectNoteUpdatedEvent;
