// @zeplin/sdk/dist/models/project-screen-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenCreatedEventResourceToJSON = exports.transformJSONToProjectScreenCreatedEventResource = void 0;

var _screen = require("./screen");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenCreatedEventResourceToJSON = function transformProjectScreenCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screen.transformScreenToJSON)(value.data)
  };
};

exports.transformProjectScreenCreatedEventResourceToJSON = transformProjectScreenCreatedEventResourceToJSON;

var transformJSONToProjectScreenCreatedEventResource = function transformJSONToProjectScreenCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screen.transformJSONToScreen)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenCreatedEventResource
 */


exports.transformJSONToProjectScreenCreatedEventResource = transformJSONToProjectScreenCreatedEventResource;
