// @zeplin/sdk/dist/models/project-color-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectColorCreatedEventResourceToJSON = exports.transformJSONToProjectColorCreatedEventResource = void 0;

var _color = require("./color");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectColorCreatedEventResourceToJSON = function transformProjectColorCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _color.transformColorToJSON)(value.data)
  };
};

exports.transformProjectColorCreatedEventResourceToJSON = transformProjectColorCreatedEventResourceToJSON;

var transformJSONToProjectColorCreatedEventResource = function transformJSONToProjectColorCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _color.transformJSONToColor)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectColorCreatedEventResource
 */


exports.transformJSONToProjectColorCreatedEventResource = transformJSONToProjectColorCreatedEventResource;
