// @zeplin/sdk/dist/models/project-component-deleted-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectComponentDeletedEventResourceToJSON = exports.transformJSONToProjectComponentDeletedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectComponentDeletedEventResourceToJSON = function transformProjectComponentDeletedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformProjectComponentDeletedEventResourceToJSON = transformProjectComponentDeletedEventResourceToJSON;

var transformJSONToProjectComponentDeletedEventResource = function transformJSONToProjectComponentDeletedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface ProjectComponentDeletedEventResource
 */


exports.transformJSONToProjectComponentDeletedEventResource = transformJSONToProjectComponentDeletedEventResource;
