// @zeplin/sdk/dist/models/project-member.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectMemberToJSON = exports.transformJSONToProjectMember = void 0;

var _user = require("./user");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectMemberToJSON = function transformProjectMemberToJSON(value) {
  return {
    user: (0, _user.transformUserToJSON)(value.user),
    role: value.role
  };
};

exports.transformProjectMemberToJSON = transformProjectMemberToJSON;

var transformJSONToProjectMember = function transformJSONToProjectMember(value) {
  return {
    user: (0, _user.transformJSONToUser)(value.user),
    role: value.role
  };
};
/**
 * 
 * @export
 * @interface ProjectMember
 */


exports.transformJSONToProjectMember = transformJSONToProjectMember;
