// @zeplin/sdk/dist/models/project-text-style-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectTextStyleCreatedEventResourceToJSON = exports.transformJSONToProjectTextStyleCreatedEventResource = void 0;

var _textStyle = require("./text-style");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectTextStyleCreatedEventResourceToJSON = function transformProjectTextStyleCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _textStyle.transformTextStyleToJSON)(value.data)
  };
};

exports.transformProjectTextStyleCreatedEventResourceToJSON = transformProjectTextStyleCreatedEventResourceToJSON;

var transformJSONToProjectTextStyleCreatedEventResource = function transformJSONToProjectTextStyleCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _textStyle.transformJSONToTextStyle)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectTextStyleCreatedEventResource
 */


exports.transformJSONToProjectTextStyleCreatedEventResource = transformJSONToProjectTextStyleCreatedEventResource;
