// @zeplin/sdk/dist/models/project-screen-version-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenVersionUpdatedEventToJSON = exports.transformJSONToProjectScreenVersionUpdatedEvent = void 0;

var _projectScreenVersionCreatedEventResource = require("./project-screen-version-created-event-resource");

var _projectScreenVersionEventContext = require("./project-screen-version-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenVersionUpdatedEventToJSON = function transformProjectScreenVersionUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenVersionCreatedEventResource.transformProjectScreenVersionCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectScreenVersionEventContext.transformProjectScreenVersionEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectScreenVersionUpdatedEventToJSON = transformProjectScreenVersionUpdatedEventToJSON;

var transformJSONToProjectScreenVersionUpdatedEvent = function transformJSONToProjectScreenVersionUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenVersionCreatedEventResource.transformJSONToProjectScreenVersionCreatedEventResource)(value.resource),
    context: (0, _projectScreenVersionEventContext.transformJSONToProjectScreenVersionEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenVersionUpdatedEvent
 */


exports.transformJSONToProjectScreenVersionUpdatedEvent = transformJSONToProjectScreenVersionUpdatedEvent;
