// @zeplin/sdk/dist/models/project-status-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectStatusEnumToJSON = exports.transformJSONToProjectStatusEnum = exports.ProjectStatusEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectStatusEnumToJSON = function transformProjectStatusEnumToJSON(value) {
  return value;
};

exports.transformProjectStatusEnumToJSON = transformProjectStatusEnumToJSON;

var transformJSONToProjectStatusEnum = function transformJSONToProjectStatusEnum(value) {
  return value;
};
/**
 * The status of the project
 * @export
 * @enum {string}
 */


exports.transformJSONToProjectStatusEnum = transformJSONToProjectStatusEnum;
var ProjectStatusEnum;
exports.ProjectStatusEnum = ProjectStatusEnum;

(function (ProjectStatusEnum) {
  ProjectStatusEnum["ACTIVE"] = "active";
  ProjectStatusEnum["ARCHIVED"] = "archived";
})(ProjectStatusEnum || (exports.ProjectStatusEnum = ProjectStatusEnum = {}));
