// @zeplin/sdk/dist/models/project-update-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectUpdateBodyToJSON = exports.transformJSONToProjectUpdateBody = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectUpdateBodyToJSON = function transformProjectUpdateBodyToJSON(value) {
  return {
    name: value.name,
    description: value.description,
    workflow_status_id: value.workflowStatusId
  };
};

exports.transformProjectUpdateBodyToJSON = transformProjectUpdateBodyToJSON;

var transformJSONToProjectUpdateBody = function transformJSONToProjectUpdateBody(value) {
  return {
    name: value.name,
    description: value.description,
    workflowStatusId: value.workflow_status_id
  };
};
/**
 * 
 * @export
 * @interface ProjectUpdateBody
 */


exports.transformJSONToProjectUpdateBody = transformJSONToProjectUpdateBody;
