// @zeplin/sdk/dist/models/project-text-style-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectTextStyleDeletedEventToJSON = exports.transformJSONToProjectTextStyleDeletedEvent = void 0;

var _projectTextStyleDeletedEventResource = require("./project-text-style-deleted-event-resource");

var _projectTextStyleEventContext = require("./project-text-style-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectTextStyleDeletedEventToJSON = function transformProjectTextStyleDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectTextStyleDeletedEventResource.transformProjectTextStyleDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectTextStyleEventContext.transformProjectTextStyleEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectTextStyleDeletedEventToJSON = transformProjectTextStyleDeletedEventToJSON;

var transformJSONToProjectTextStyleDeletedEvent = function transformJSONToProjectTextStyleDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectTextStyleDeletedEventResource.transformJSONToProjectTextStyleDeletedEventResource)(value.resource),
    context: (0, _projectTextStyleEventContext.transformJSONToProjectTextStyleEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectTextStyleDeletedEvent
 */


exports.transformJSONToProjectTextStyleDeletedEvent = transformJSONToProjectTextStyleDeletedEvent;
