// @zeplin/sdk/dist/models/project-spacing-token-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectSpacingTokenDeletedEventToJSON = exports.transformJSONToProjectSpacingTokenDeletedEvent = void 0;

var _projectSpacingTokenDeletedEventResource = require("./project-spacing-token-deleted-event-resource");

var _projectSpacingTokenEventContext = require("./project-spacing-token-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectSpacingTokenDeletedEventToJSON = function transformProjectSpacingTokenDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectSpacingTokenDeletedEventResource.transformProjectSpacingTokenDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectSpacingTokenEventContext.transformProjectSpacingTokenEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectSpacingTokenDeletedEventToJSON = transformProjectSpacingTokenDeletedEventToJSON;

var transformJSONToProjectSpacingTokenDeletedEvent = function transformJSONToProjectSpacingTokenDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectSpacingTokenDeletedEventResource.transformJSONToProjectSpacingTokenDeletedEventResource)(value.resource),
    context: (0, _projectSpacingTokenEventContext.transformJSONToProjectSpacingTokenEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectSpacingTokenDeletedEvent
 */


exports.transformJSONToProjectSpacingTokenDeletedEvent = transformJSONToProjectSpacingTokenDeletedEvent;
