// @zeplin/sdk/dist/models/project-flow-board-built-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectFlowBoardBuiltEventResourceToJSON = exports.transformJSONToProjectFlowBoardBuiltEventResource = void 0;

var _flowBoard = require("./flow-board");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectFlowBoardBuiltEventResourceToJSON = function transformProjectFlowBoardBuiltEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _flowBoard.transformFlowBoardToJSON)(value.data)
  };
};

exports.transformProjectFlowBoardBuiltEventResourceToJSON = transformProjectFlowBoardBuiltEventResourceToJSON;

var transformJSONToProjectFlowBoardBuiltEventResource = function transformJSONToProjectFlowBoardBuiltEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _flowBoard.transformJSONToFlowBoard)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectFlowBoardBuiltEventResource
 */


exports.transformJSONToProjectFlowBoardBuiltEventResource = transformJSONToProjectFlowBoardBuiltEventResource;
