// @zeplin/sdk/dist/models/project-component-event-context.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectComponentEventContextToJSON = exports.transformJSONToProjectComponentEventContext = void 0;

var _componentVersionSummary = require("./component-version-summary");

var _project = require("./project");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectComponentEventContextToJSON = function transformProjectComponentEventContextToJSON(value) {
  return {
    project: (0, _project.transformProjectToJSON)(value.project),
    version: (0, _componentVersionSummary.transformComponentVersionSummaryToJSON)(value.version)
  };
};

exports.transformProjectComponentEventContextToJSON = transformProjectComponentEventContextToJSON;

var transformJSONToProjectComponentEventContext = function transformJSONToProjectComponentEventContext(value) {
  return {
    project: (0, _project.transformJSONToProject)(value.project),
    version: (0, _componentVersionSummary.transformJSONToComponentVersionSummary)(value.version)
  };
};
/**
 * 
 * @export
 * @interface ProjectComponentEventContext
 */


exports.transformJSONToProjectComponentEventContext = transformJSONToProjectComponentEventContext;
