// @zeplin/sdk/dist/models/project-screen-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenDeletedEventToJSON = exports.transformJSONToProjectScreenDeletedEvent = void 0;

var _projectScreenDeletedEventContext = require("./project-screen-deleted-event-context");

var _projectScreenDeletedEventResource = require("./project-screen-deleted-event-resource");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenDeletedEventToJSON = function transformProjectScreenDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenDeletedEventResource.transformProjectScreenDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectScreenDeletedEventContext.transformProjectScreenDeletedEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectScreenDeletedEventToJSON = transformProjectScreenDeletedEventToJSON;

var transformJSONToProjectScreenDeletedEvent = function transformJSONToProjectScreenDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenDeletedEventResource.transformJSONToProjectScreenDeletedEventResource)(value.resource),
    context: (0, _projectScreenDeletedEventContext.transformJSONToProjectScreenDeletedEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenDeletedEvent
 */


exports.transformJSONToProjectScreenDeletedEvent = transformJSONToProjectScreenDeletedEvent;
