// @zeplin/sdk/dist/models/project-webhook-event-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectWebhookEventEnumToJSON = exports.transformJSONToProjectWebhookEventEnum = exports.ProjectWebhookEventEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectWebhookEventEnumToJSON = function transformProjectWebhookEventEnumToJSON(value) {
  return value;
};

exports.transformProjectWebhookEventEnumToJSON = transformProjectWebhookEventEnumToJSON;

var transformJSONToProjectWebhookEventEnum = function transformJSONToProjectWebhookEventEnum(value) {
  return value;
};
/**
 * 
 * @export
 * @enum {string}
 */


exports.transformJSONToProjectWebhookEventEnum = transformJSONToProjectWebhookEventEnum;
var ProjectWebhookEventEnum;
exports.ProjectWebhookEventEnum = ProjectWebhookEventEnum;

(function (ProjectWebhookEventEnum) {
  ProjectWebhookEventEnum["STAR"] = "project.*";
  ProjectWebhookEventEnum["COLOR"] = "project.color";
  ProjectWebhookEventEnum["COMPONENT"] = "project.component";
  ProjectWebhookEventEnum["MEMBER"] = "project.member";
  ProjectWebhookEventEnum["NOTE"] = "project.note";
  ProjectWebhookEventEnum["NOTE_COMMENT"] = "project.note.comment";
  ProjectWebhookEventEnum["SCREEN"] = "project.screen";
  ProjectWebhookEventEnum["SCREEN_VERSION"] = "project.screen.version";
  ProjectWebhookEventEnum["SPACING_TOKEN"] = "project.spacing_token";
  ProjectWebhookEventEnum["TEXT_STYLE"] = "project.text_style";
  ProjectWebhookEventEnum["FLOW_BOARD"] = "project.flow_board";
})(ProjectWebhookEventEnum || (exports.ProjectWebhookEventEnum = ProjectWebhookEventEnum = {}));
