// @zeplin/sdk/dist/models/project-flow-board-built-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectFlowBoardBuiltEventToJSON = exports.transformJSONToProjectFlowBoardBuiltEvent = void 0;

var _projectFlowBoardBuiltEventResource = require("./project-flow-board-built-event-resource");

var _projectFlowBoardEventContext = require("./project-flow-board-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectFlowBoardBuiltEventToJSON = function transformProjectFlowBoardBuiltEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectFlowBoardBuiltEventResource.transformProjectFlowBoardBuiltEventResourceToJSON)(value.resource),
    context: (0, _projectFlowBoardEventContext.transformProjectFlowBoardEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectFlowBoardBuiltEventToJSON = transformProjectFlowBoardBuiltEventToJSON;

var transformJSONToProjectFlowBoardBuiltEvent = function transformJSONToProjectFlowBoardBuiltEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectFlowBoardBuiltEventResource.transformJSONToProjectFlowBoardBuiltEventResource)(value.resource),
    context: (0, _projectFlowBoardEventContext.transformJSONToProjectFlowBoardEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectFlowBoardBuiltEvent
 */


exports.transformJSONToProjectFlowBoardBuiltEvent = transformJSONToProjectFlowBoardBuiltEvent;
