// @zeplin/sdk/dist/models/project-screen-version-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenVersionCreatedEventResourceToJSON = exports.transformJSONToProjectScreenVersionCreatedEventResource = void 0;

var _screenVersionSummary = require("./screen-version-summary");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenVersionCreatedEventResourceToJSON = function transformProjectScreenVersionCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screenVersionSummary.transformScreenVersionSummaryToJSON)(value.data)
  };
};

exports.transformProjectScreenVersionCreatedEventResourceToJSON = transformProjectScreenVersionCreatedEventResourceToJSON;

var transformJSONToProjectScreenVersionCreatedEventResource = function transformJSONToProjectScreenVersionCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screenVersionSummary.transformJSONToScreenVersionSummary)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenVersionCreatedEventResource
 */


exports.transformJSONToProjectScreenVersionCreatedEventResource = transformJSONToProjectScreenVersionCreatedEventResource;
