// @zeplin/sdk/dist/models/project-color-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectColorDeletedEventToJSON = exports.transformJSONToProjectColorDeletedEvent = void 0;

var _projectColorDeletedEventResource = require("./project-color-deleted-event-resource");

var _projectColorEventContext = require("./project-color-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectColorDeletedEventToJSON = function transformProjectColorDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectColorDeletedEventResource.transformProjectColorDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectColorEventContext.transformProjectColorEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectColorDeletedEventToJSON = transformProjectColorDeletedEventToJSON;

var transformJSONToProjectColorDeletedEvent = function transformJSONToProjectColorDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectColorDeletedEventResource.transformJSONToProjectColorDeletedEventResource)(value.resource),
    context: (0, _projectColorEventContext.transformJSONToProjectColorEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectColorDeletedEvent
 */


exports.transformJSONToProjectColorDeletedEvent = transformJSONToProjectColorDeletedEvent;
