// @zeplin/sdk/dist/models/project-note-deleted-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteDeletedEventResourceToJSON = exports.transformJSONToProjectNoteDeletedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteDeletedEventResourceToJSON = function transformProjectNoteDeletedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformProjectNoteDeletedEventResourceToJSON = transformProjectNoteDeletedEventResourceToJSON;

var transformJSONToProjectNoteDeletedEventResource = function transformJSONToProjectNoteDeletedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteDeletedEventResource
 */


exports.transformJSONToProjectNoteDeletedEventResource = transformJSONToProjectNoteDeletedEventResource;
