// @zeplin/sdk/dist/models/project-screen-deleted-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenDeletedEventResourceToJSON = exports.transformJSONToProjectScreenDeletedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenDeletedEventResourceToJSON = function transformProjectScreenDeletedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformProjectScreenDeletedEventResourceToJSON = transformProjectScreenDeletedEventResourceToJSON;

var transformJSONToProjectScreenDeletedEventResource = function transformJSONToProjectScreenDeletedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenDeletedEventResource
 */


exports.transformJSONToProjectScreenDeletedEventResource = transformJSONToProjectScreenDeletedEventResource;
