// @zeplin/sdk/dist/models/project-screen-version-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenVersionDeletedEventToJSON = exports.transformJSONToProjectScreenVersionDeletedEvent = void 0;

var _projectScreenVersionDeletedEventResource = require("./project-screen-version-deleted-event-resource");

var _projectScreenVersionEventContext = require("./project-screen-version-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenVersionDeletedEventToJSON = function transformProjectScreenVersionDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenVersionDeletedEventResource.transformProjectScreenVersionDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectScreenVersionEventContext.transformProjectScreenVersionEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectScreenVersionDeletedEventToJSON = transformProjectScreenVersionDeletedEventToJSON;

var transformJSONToProjectScreenVersionDeletedEvent = function transformJSONToProjectScreenVersionDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectScreenVersionDeletedEventResource.transformJSONToProjectScreenVersionDeletedEventResource)(value.resource),
    context: (0, _projectScreenVersionEventContext.transformJSONToProjectScreenVersionEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenVersionDeletedEvent
 */


exports.transformJSONToProjectScreenVersionDeletedEvent = transformJSONToProjectScreenVersionDeletedEvent;
