// @zeplin/sdk/dist/models/project-screen-version-deleted-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectScreenVersionDeletedEventResourceToJSON = exports.transformJSONToProjectScreenVersionDeletedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectScreenVersionDeletedEventResourceToJSON = function transformProjectScreenVersionDeletedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformProjectScreenVersionDeletedEventResourceToJSON = transformProjectScreenVersionDeletedEventResourceToJSON;

var transformJSONToProjectScreenVersionDeletedEventResource = function transformJSONToProjectScreenVersionDeletedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface ProjectScreenVersionDeletedEventResource
 */


exports.transformJSONToProjectScreenVersionDeletedEventResource = transformJSONToProjectScreenVersionDeletedEventResource;
