// @zeplin/sdk/dist/models/project-note-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteDeletedEventToJSON = exports.transformJSONToProjectNoteDeletedEvent = void 0;

var _projectNoteDeletedEventResource = require("./project-note-deleted-event-resource");

var _projectNoteEventContext = require("./project-note-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteDeletedEventToJSON = function transformProjectNoteDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteDeletedEventResource.transformProjectNoteDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectNoteEventContext.transformProjectNoteEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectNoteDeletedEventToJSON = transformProjectNoteDeletedEventToJSON;

var transformJSONToProjectNoteDeletedEvent = function transformJSONToProjectNoteDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteDeletedEventResource.transformJSONToProjectNoteDeletedEventResource)(value.resource),
    context: (0, _projectNoteEventContext.transformJSONToProjectNoteEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteDeletedEvent
 */


exports.transformJSONToProjectNoteDeletedEvent = transformJSONToProjectNoteDeletedEvent;
