// @zeplin/sdk/dist/models/project-note-comment-event-context.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteCommentEventContextToJSON = exports.transformJSONToProjectNoteCommentEventContext = void 0;

var _project = require("./project");

var _screen = require("./screen");

var _screenNote = require("./screen-note");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteCommentEventContextToJSON = function transformProjectNoteCommentEventContextToJSON(value) {
  return {
    project: (0, _project.transformProjectToJSON)(value.project),
    screen: (0, _screen.transformScreenToJSON)(value.screen),
    note: (0, _screenNote.transformScreenNoteToJSON)(value.note)
  };
};

exports.transformProjectNoteCommentEventContextToJSON = transformProjectNoteCommentEventContextToJSON;

var transformJSONToProjectNoteCommentEventContext = function transformJSONToProjectNoteCommentEventContext(value) {
  return {
    project: (0, _project.transformJSONToProject)(value.project),
    screen: (0, _screen.transformJSONToScreen)(value.screen),
    note: (0, _screenNote.transformJSONToScreenNote)(value.note)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteCommentEventContext
 */


exports.transformJSONToProjectNoteCommentEventContext = transformJSONToProjectNoteCommentEventContext;
