// @zeplin/sdk/dist/models/project-note-comment-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteCommentDeletedEventToJSON = exports.transformJSONToProjectNoteCommentDeletedEvent = void 0;

var _projectNoteCommentDeletedEventResource = require("./project-note-comment-deleted-event-resource");

var _projectNoteCommentEventContext = require("./project-note-comment-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteCommentDeletedEventToJSON = function transformProjectNoteCommentDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCommentDeletedEventResource.transformProjectNoteCommentDeletedEventResourceToJSON)(value.resource),
    context: (0, _projectNoteCommentEventContext.transformProjectNoteCommentEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectNoteCommentDeletedEventToJSON = transformProjectNoteCommentDeletedEventToJSON;

var transformJSONToProjectNoteCommentDeletedEvent = function transformJSONToProjectNoteCommentDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectNoteCommentDeletedEventResource.transformJSONToProjectNoteCommentDeletedEventResource)(value.resource),
    context: (0, _projectNoteCommentEventContext.transformJSONToProjectNoteCommentEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteCommentDeletedEvent
 */


exports.transformJSONToProjectNoteCommentDeletedEvent = transformJSONToProjectNoteCommentDeletedEvent;
