// @zeplin/sdk/dist/models/project-color-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectColorCreatedEventToJSON = exports.transformJSONToProjectColorCreatedEvent = void 0;

var _projectColorCreatedEventResource = require("./project-color-created-event-resource");

var _projectColorEventContext = require("./project-color-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectColorCreatedEventToJSON = function transformProjectColorCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectColorCreatedEventResource.transformProjectColorCreatedEventResourceToJSON)(value.resource),
    context: (0, _projectColorEventContext.transformProjectColorEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformProjectColorCreatedEventToJSON = transformProjectColorCreatedEventToJSON;

var transformJSONToProjectColorCreatedEvent = function transformJSONToProjectColorCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectColorCreatedEventResource.transformJSONToProjectColorCreatedEventResource)(value.resource),
    context: (0, _projectColorEventContext.transformJSONToProjectColorEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface ProjectColorCreatedEvent
 */


exports.transformJSONToProjectColorCreatedEvent = transformJSONToProjectColorCreatedEvent;
