// @zeplin/sdk/dist/models/project-note-comment-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformProjectNoteCommentCreatedEventResourceToJSON = exports.transformJSONToProjectNoteCommentCreatedEventResource = void 0;

var _screenNoteComment = require("./screen-note-comment");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformProjectNoteCommentCreatedEventResourceToJSON = function transformProjectNoteCommentCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screenNoteComment.transformScreenNoteCommentToJSON)(value.data)
  };
};

exports.transformProjectNoteCommentCreatedEventResourceToJSON = transformProjectNoteCommentCreatedEventResourceToJSON;

var transformJSONToProjectNoteCommentCreatedEventResource = function transformJSONToProjectNoteCommentCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _screenNoteComment.transformJSONToScreenNoteComment)(value.data)
  };
};
/**
 * 
 * @export
 * @interface ProjectNoteCommentCreatedEventResource
 */


exports.transformJSONToProjectNoteCommentCreatedEventResource = transformJSONToProjectNoteCommentCreatedEventResource;
