// @zeplin/sdk/dist/models/resource-source.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformResourceSourceToJSON = exports.transformJSONToResourceSource = void 0;

var _resourceProjectSource = require("./resource-project-source");

var _resourceStyleguideSource = require("./resource-styleguide-source");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformResourceSourceToJSON = function transformResourceSourceToJSON(value) {
  return {
    project: value.project && (0, _resourceProjectSource.transformResourceProjectSourceToJSON)(value.project),
    styleguide: value.styleguide && (0, _resourceStyleguideSource.transformResourceStyleguideSourceToJSON)(value.styleguide)
  };
};

exports.transformResourceSourceToJSON = transformResourceSourceToJSON;

var transformJSONToResourceSource = function transformJSONToResourceSource(value) {
  return {
    project: value.project && (0, _resourceProjectSource.transformJSONToResourceProjectSource)(value.project),
    styleguide: value.styleguide && (0, _resourceStyleguideSource.transformJSONToResourceStyleguideSource)(value.styleguide)
  };
};
/**
 * Source details for resource. It has to be either `project` or `styleguide`.
 * @export
 * @interface ResourceSource
 */


exports.transformJSONToResourceSource = transformJSONToResourceSource;
