// @zeplin/sdk/dist/models/resource-styleguide-source.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformResourceStyleguideSourceToJSON = exports.transformJSONToResourceStyleguideSource = void 0;

require("core-js/modules/es.function.name.js");

var _entityReference = require("./entity-reference");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformResourceStyleguideSourceToJSON = function transformResourceStyleguideSourceToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    platform: value.platform,
    parent: value.parent && (0, _entityReference.transformEntityReferenceToJSON)(value.parent)
  };
};

exports.transformResourceStyleguideSourceToJSON = transformResourceStyleguideSourceToJSON;

var transformJSONToResourceStyleguideSource = function transformJSONToResourceStyleguideSource(value) {
  return {
    id: value.id,
    name: value.name,
    platform: value.platform,
    parent: value.parent && (0, _entityReference.transformJSONToEntityReference)(value.parent)
  };
};
/**
 * The source styleguide of the resource
 * @export
 * @interface ResourceStyleguideSource
 */


exports.transformJSONToResourceStyleguideSource = transformJSONToResourceStyleguideSource;
