// @zeplin/sdk/dist/models/screen-note-create-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteCreateBodyToJSON = exports.transformJSONToScreenNoteCreateBody = void 0;

var _screenNoteColorNameEnum = require("./screen-note-color-name-enum");

var _screenNotePosition = require("./screen-note-position");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteCreateBodyToJSON = function transformScreenNoteCreateBodyToJSON(value) {
  return {
    content: value.content,
    position: (0, _screenNotePosition.transformScreenNotePositionToJSON)(value.position),
    color: (0, _screenNoteColorNameEnum.transformScreenNoteColorNameEnumToJSON)(value.color)
  };
};

exports.transformScreenNoteCreateBodyToJSON = transformScreenNoteCreateBodyToJSON;

var transformJSONToScreenNoteCreateBody = function transformJSONToScreenNoteCreateBody(value) {
  return {
    content: value.content,
    position: (0, _screenNotePosition.transformJSONToScreenNotePosition)(value.position),
    color: (0, _screenNoteColorNameEnum.transformJSONToScreenNoteColorNameEnum)(value.color)
  };
};
/**
 * 
 * @export
 * @interface ScreenNoteCreateBody
 */


exports.transformJSONToScreenNoteCreateBody = transformJSONToScreenNoteCreateBody;
