// @zeplin/sdk/dist/models/screen-note.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteToJSON = exports.transformJSONToScreenNote = void 0;

require("core-js/modules/es.array.map.js");

var _screenNoteColor = require("./screen-note-color");

var _screenNoteComment = require("./screen-note-comment");

var _screenNotePosition = require("./screen-note-position");

var _screenNoteStatusEnum = require("./screen-note-status-enum");

var _user = require("./user");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteToJSON = function transformScreenNoteToJSON(value) {
  return {
    id: value.id,
    creator: (0, _user.transformUserToJSON)(value.creator),
    order: value.order,
    status: (0, _screenNoteStatusEnum.transformScreenNoteStatusEnumToJSON)(value.status),
    position: (0, _screenNotePosition.transformScreenNotePositionToJSON)(value.position),
    color: (0, _screenNoteColor.transformScreenNoteColorToJSON)(value.color),
    comments: value.comments.map(_screenNoteComment.transformScreenNoteCommentToJSON),
    created: value.created
  };
};

exports.transformScreenNoteToJSON = transformScreenNoteToJSON;

var transformJSONToScreenNote = function transformJSONToScreenNote(value) {
  return {
    id: value.id,
    creator: (0, _user.transformJSONToUser)(value.creator),
    order: value.order,
    status: (0, _screenNoteStatusEnum.transformJSONToScreenNoteStatusEnum)(value.status),
    position: (0, _screenNotePosition.transformJSONToScreenNotePosition)(value.position),
    color: (0, _screenNoteColor.transformJSONToScreenNoteColor)(value.color),
    comments: value.comments.map(_screenNoteComment.transformJSONToScreenNoteComment),
    created: value.created
  };
};
/**
 * 
 * @export
 * @interface ScreenNote
 */


exports.transformJSONToScreenNote = transformJSONToScreenNote;
