// @zeplin/sdk/dist/models/screen-variant-group-value.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenVariantGroupValueToJSON = exports.transformJSONToScreenVariantGroupValue = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenVariantGroupValueToJSON = function transformScreenVariantGroupValueToJSON(value) {
  return {
    screen_id: value.screenId,
    value: value.value
  };
};

exports.transformScreenVariantGroupValueToJSON = transformScreenVariantGroupValueToJSON;

var transformJSONToScreenVariantGroupValue = function transformJSONToScreenVariantGroupValue(value) {
  return {
    screenId: value.screen_id,
    value: value.value
  };
};
/**
 * 
 * @export
 * @interface ScreenVariantGroupValue
 */


exports.transformJSONToScreenVariantGroupValue = transformJSONToScreenVariantGroupValue;
