// @zeplin/sdk/dist/models/screen-note-color-name-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteColorNameEnumToJSON = exports.transformJSONToScreenNoteColorNameEnum = exports.ScreenNoteColorNameEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteColorNameEnumToJSON = function transformScreenNoteColorNameEnumToJSON(value) {
  return value;
};

exports.transformScreenNoteColorNameEnumToJSON = transformScreenNoteColorNameEnumToJSON;

var transformJSONToScreenNoteColorNameEnum = function transformJSONToScreenNoteColorNameEnum(value) {
  return value;
};
/**
 * 
 * @export
 * @enum {string}
 */


exports.transformJSONToScreenNoteColorNameEnum = transformJSONToScreenNoteColorNameEnum;
var ScreenNoteColorNameEnum;
exports.ScreenNoteColorNameEnum = ScreenNoteColorNameEnum;

(function (ScreenNoteColorNameEnum) {
  ScreenNoteColorNameEnum["YELLOW"] = "yellow";
  ScreenNoteColorNameEnum["ORANGE"] = "orange";
  ScreenNoteColorNameEnum["PEACH"] = "peach";
  ScreenNoteColorNameEnum["GREEN"] = "green";
  ScreenNoteColorNameEnum["TURQUOISE"] = "turquoise";
  ScreenNoteColorNameEnum["CORNFLOWER_BLUE"] = "cornflower_blue";
  ScreenNoteColorNameEnum["DEEP_PURPLE"] = "deep_purple";
})(ScreenNoteColorNameEnum || (exports.ScreenNoteColorNameEnum = ScreenNoteColorNameEnum = {}));
