// @zeplin/sdk/dist/models/screen-note-update-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteUpdateBodyToJSON = exports.transformJSONToScreenNoteUpdateBody = void 0;

var _screenNoteColorNameEnum = require("./screen-note-color-name-enum");

var _screenNotePosition = require("./screen-note-position");

var _screenNoteStatusEnum = require("./screen-note-status-enum");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteUpdateBodyToJSON = function transformScreenNoteUpdateBodyToJSON(value) {
  return {
    status: value.status && (0, _screenNoteStatusEnum.transformScreenNoteStatusEnumToJSON)(value.status),
    position: value.position && (0, _screenNotePosition.transformScreenNotePositionToJSON)(value.position),
    color: value.color && (0, _screenNoteColorNameEnum.transformScreenNoteColorNameEnumToJSON)(value.color)
  };
};

exports.transformScreenNoteUpdateBodyToJSON = transformScreenNoteUpdateBodyToJSON;

var transformJSONToScreenNoteUpdateBody = function transformJSONToScreenNoteUpdateBody(value) {
  return {
    status: value.status && (0, _screenNoteStatusEnum.transformJSONToScreenNoteStatusEnum)(value.status),
    position: value.position && (0, _screenNotePosition.transformJSONToScreenNotePosition)(value.position),
    color: value.color && (0, _screenNoteColorNameEnum.transformJSONToScreenNoteColorNameEnum)(value.color)
  };
};
/**
 * 
 * @export
 * @interface ScreenNoteUpdateBody
 */


exports.transformJSONToScreenNoteUpdateBody = transformJSONToScreenNoteUpdateBody;
