// @zeplin/sdk/dist/models/screen-variant-group.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenVariantGroupToJSON = exports.transformJSONToScreenVariantGroup = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.array.map.js");

var _screenVariantGroupValue = require("./screen-variant-group-value");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenVariantGroupToJSON = function transformScreenVariantGroupToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    variants: value.variants.map(_screenVariantGroupValue.transformScreenVariantGroupValueToJSON)
  };
};

exports.transformScreenVariantGroupToJSON = transformScreenVariantGroupToJSON;

var transformJSONToScreenVariantGroup = function transformJSONToScreenVariantGroup(value) {
  return {
    id: value.id,
    name: value.name,
    variants: value.variants.map(_screenVariantGroupValue.transformJSONToScreenVariantGroupValue)
  };
};
/**
 * 
 * @export
 * @interface ScreenVariantGroup
 */


exports.transformJSONToScreenVariantGroup = transformJSONToScreenVariantGroup;
