// @zeplin/sdk/dist/models/screen-variant-group-reference.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenVariantGroupReferenceToJSON = exports.transformJSONToScreenVariantGroupReference = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenVariantGroupReferenceToJSON = function transformScreenVariantGroupReferenceToJSON(value) {
  return {
    id: value.id,
    name: value.name
  };
};

exports.transformScreenVariantGroupReferenceToJSON = transformScreenVariantGroupReferenceToJSON;

var transformJSONToScreenVariantGroupReference = function transformJSONToScreenVariantGroupReference(value) {
  return {
    id: value.id,
    name: value.name
  };
};
/**
 * Variant group that contains the screen
 * @export
 * @interface ScreenVariantGroupReference
 */


exports.transformJSONToScreenVariantGroupReference = transformJSONToScreenVariantGroupReference;
