// @zeplin/sdk/dist/models/screen.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenToJSON = exports.transformJSONToScreen = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

var _entityReference = require("./entity-reference");

var _screenVariant = require("./screen-variant");

var _snapshotImage = require("./snapshot-image");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenToJSON = function transformScreenToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    tags: value.tags,
    image: (0, _snapshotImage.transformSnapshotImageToJSON)(value.image),
    created: value.created,
    updated: value.updated,
    number_of_versions: value.numberOfVersions,
    number_of_notes: value.numberOfNotes,
    section: value.section && (0, _entityReference.transformEntityReferenceToJSON)(value.section),
    variant: value.variant && (0, _screenVariant.transformScreenVariantToJSON)(value.variant)
  };
};

exports.transformScreenToJSON = transformScreenToJSON;

var transformJSONToScreen = function transformJSONToScreen(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    tags: value.tags,
    image: (0, _snapshotImage.transformJSONToSnapshotImage)(value.image),
    created: value.created,
    updated: value.updated,
    numberOfVersions: value.number_of_versions,
    numberOfNotes: value.number_of_notes,
    section: value.section && (0, _entityReference.transformJSONToEntityReference)(value.section),
    variant: value.variant && (0, _screenVariant.transformJSONToScreenVariant)(value.variant)
  };
};
/**
 * 
 * @export
 * @interface Screen
 */


exports.transformJSONToScreen = transformJSONToScreen;
