// @zeplin/sdk/dist/models/screen-note-comment.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteCommentToJSON = exports.transformJSONToScreenNoteComment = void 0;

var _user = require("./user");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteCommentToJSON = function transformScreenNoteCommentToJSON(value) {
  return {
    id: value.id,
    content: value.content,
    author: (0, _user.transformUserToJSON)(value.author),
    updated: value.updated
  };
};

exports.transformScreenNoteCommentToJSON = transformScreenNoteCommentToJSON;

var transformJSONToScreenNoteComment = function transformJSONToScreenNoteComment(value) {
  return {
    id: value.id,
    content: value.content,
    author: (0, _user.transformJSONToUser)(value.author),
    updated: value.updated
  };
};
/**
 * 
 * @export
 * @interface ScreenNoteComment
 */


exports.transformJSONToScreenNoteComment = transformJSONToScreenNoteComment;
