// @zeplin/sdk/dist/models/screen-note-position.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNotePositionToJSON = exports.transformJSONToScreenNotePosition = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNotePositionToJSON = function transformScreenNotePositionToJSON(value) {
  return {
    x: value.x,
    y: value.y
  };
};

exports.transformScreenNotePositionToJSON = transformScreenNotePositionToJSON;

var transformJSONToScreenNotePosition = function transformJSONToScreenNotePosition(value) {
  return {
    x: value.x,
    y: value.y
  };
};
/**
 * Position of the note with respect to top left corner. Values are normalized in [0, 1]
 * @export
 * @interface ScreenNotePosition
 */


exports.transformJSONToScreenNotePosition = transformJSONToScreenNotePosition;
