// @zeplin/sdk/dist/models/screen-note-color.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteColorToJSON = exports.transformJSONToScreenNoteColor = void 0;

require("core-js/modules/es.function.name.js");

var _screenNoteColorNameEnum = require("./screen-note-color-name-enum");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteColorToJSON = function transformScreenNoteColorToJSON(value) {
  return {
    name: value.name && (0, _screenNoteColorNameEnum.transformScreenNoteColorNameEnumToJSON)(value.name),
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};

exports.transformScreenNoteColorToJSON = transformScreenNoteColorToJSON;

var transformJSONToScreenNoteColor = function transformJSONToScreenNoteColor(value) {
  return {
    name: value.name && (0, _screenNoteColorNameEnum.transformJSONToScreenNoteColorNameEnum)(value.name),
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};
/**
 * 
 * @export
 * @interface ScreenNoteColor
 */


exports.transformJSONToScreenNoteColor = transformJSONToScreenNoteColor;
