// @zeplin/sdk/dist/models/screen-note-status-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformScreenNoteStatusEnumToJSON = exports.transformJSONToScreenNoteStatusEnum = exports.ScreenNoteStatusEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformScreenNoteStatusEnumToJSON = function transformScreenNoteStatusEnumToJSON(value) {
  return value;
};

exports.transformScreenNoteStatusEnumToJSON = transformScreenNoteStatusEnumToJSON;

var transformJSONToScreenNoteStatusEnum = function transformJSONToScreenNoteStatusEnum(value) {
  return value;
};
/**
 * Status of the note
 * @export
 * @enum {string}
 */


exports.transformJSONToScreenNoteStatusEnum = transformJSONToScreenNoteStatusEnum;
var ScreenNoteStatusEnum;
exports.ScreenNoteStatusEnum = ScreenNoteStatusEnum;

(function (ScreenNoteStatusEnum) {
  ScreenNoteStatusEnum["OPEN"] = "open";
  ScreenNoteStatusEnum["RESOLVED"] = "resolved";
})(ScreenNoteStatusEnum || (exports.ScreenNoteStatusEnum = ScreenNoteStatusEnum = {}));
