// @zeplin/sdk/dist/models/snapshot-image.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformSnapshotImageToJSON = exports.transformJSONToSnapshotImage = void 0;

var _thumbnails = require("./thumbnails");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformSnapshotImageToJSON = function transformSnapshotImageToJSON(value) {
  return {
    width: value.width,
    height: value.height,
    thumbnails: (0, _thumbnails.transformThumbnailsToJSON)(value.thumbnails),
    original_url: value.originalUrl
  };
};

exports.transformSnapshotImageToJSON = transformSnapshotImageToJSON;

var transformJSONToSnapshotImage = function transformJSONToSnapshotImage(value) {
  return {
    width: value.width,
    height: value.height,
    thumbnails: (0, _thumbnails.transformJSONToThumbnails)(value.thumbnails),
    originalUrl: value.original_url
  };
};
/**
 * 
 * @export
 * @interface SnapshotImage
 */


exports.transformJSONToSnapshotImage = transformJSONToSnapshotImage;
