// @zeplin/sdk/dist/models/source-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformSourceEnumToJSON = exports.transformJSONToSourceEnum = exports.SourceEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformSourceEnumToJSON = function transformSourceEnumToJSON(value) {
  return value;
};

exports.transformSourceEnumToJSON = transformSourceEnumToJSON;

var transformJSONToSourceEnum = function transformJSONToSourceEnum(value) {
  return value;
};
/**
 * Source application of the design
 * @export
 * @enum {string}
 */


exports.transformJSONToSourceEnum = transformJSONToSourceEnum;
var SourceEnum;
exports.SourceEnum = SourceEnum;

(function (SourceEnum) {
  SourceEnum["SKETCH"] = "sketch";
  SourceEnum["XD"] = "xd";
  SourceEnum["FIGMA"] = "figma";
  SourceEnum["PSD"] = "psd";
  SourceEnum["BITMAP"] = "bitmap";
})(SourceEnum || (exports.SourceEnum = SourceEnum = {}));
