// @zeplin/sdk/dist/models/spacing-token-color.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformSpacingTokenColorToJSON = exports.transformJSONToSpacingTokenColor = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformSpacingTokenColorToJSON = function transformSpacingTokenColorToJSON(value) {
  return {
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};

exports.transformSpacingTokenColorToJSON = transformSpacingTokenColorToJSON;

var transformJSONToSpacingTokenColor = function transformJSONToSpacingTokenColor(value) {
  return {
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};
/**
 * 
 * @export
 * @interface SpacingTokenColor
 */


exports.transformJSONToSpacingTokenColor = transformJSONToSpacingTokenColor;
