// @zeplin/sdk/dist/models/spacing-section.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformSpacingSectionToJSON = exports.transformJSONToSpacingSection = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

var _entityReference = require("./entity-reference");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformSpacingSectionToJSON = function transformSpacingSectionToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    base_token: (0, _entityReference.transformEntityReferenceToJSON)(value.baseToken)
  };
};

exports.transformSpacingSectionToJSON = transformSpacingSectionToJSON;

var transformJSONToSpacingSection = function transformJSONToSpacingSection(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    baseToken: (0, _entityReference.transformJSONToEntityReference)(value.base_token)
  };
};
/**
 * 
 * @export
 * @interface SpacingSection
 */


exports.transformJSONToSpacingSection = transformJSONToSpacingSection;
