// @zeplin/sdk/dist/models/spacing-design-token-metadata.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformSpacingDesignTokenMetadataToJSON = exports.transformJSONToSpacingDesignTokenMetadata = void 0;

var _designTokenResource = require("./design-token-resource");

var _designTokenSource = require("./design-token-source");

var _entityReference = require("./entity-reference");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformSpacingDesignTokenMetadataToJSON = function transformSpacingDesignTokenMetadataToJSON(value) {
  return {
    source: (0, _designTokenSource.transformDesignTokenSourceToJSON)(value.source),
    resource: (0, _designTokenResource.transformDesignTokenResourceToJSON)(value.resource),
    spacing_section: (0, _entityReference.transformEntityReferenceToJSON)(value.spacingSection)
  };
};

exports.transformSpacingDesignTokenMetadataToJSON = transformSpacingDesignTokenMetadataToJSON;

var transformJSONToSpacingDesignTokenMetadata = function transformJSONToSpacingDesignTokenMetadata(value) {
  return {
    source: (0, _designTokenSource.transformJSONToDesignTokenSource)(value.source),
    resource: (0, _designTokenResource.transformJSONToDesignTokenResource)(value.resource),
    spacingSection: (0, _entityReference.transformJSONToEntityReference)(value.spacing_section)
  };
};
/**
 * Additional information about the spacing token
 * @export
 * @interface SpacingDesignTokenMetadata
 */


exports.transformJSONToSpacingDesignTokenMetadata = transformJSONToSpacingDesignTokenMetadata;
