// @zeplin/sdk/dist/models/spacing-token.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformSpacingTokenToJSON = exports.transformJSONToSpacingToken = void 0;

require("core-js/modules/es.function.name.js");

var _entityReference = require("./entity-reference");

var _resourceSource = require("./resource-source");

var _spacingTokenColor = require("./spacing-token-color");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformSpacingTokenToJSON = function transformSpacingTokenToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    value: value.value,
    color: (0, _spacingTokenColor.transformSpacingTokenColorToJSON)(value.color),
    section: (0, _entityReference.transformEntityReferenceToJSON)(value.section),
    source: value.source && (0, _resourceSource.transformResourceSourceToJSON)(value.source)
  };
};

exports.transformSpacingTokenToJSON = transformSpacingTokenToJSON;

var transformJSONToSpacingToken = function transformJSONToSpacingToken(value) {
  return {
    id: value.id,
    name: value.name,
    value: value.value,
    color: (0, _spacingTokenColor.transformJSONToSpacingTokenColor)(value.color),
    section: (0, _entityReference.transformJSONToEntityReference)(value.section),
    source: value.source && (0, _resourceSource.transformJSONToResourceSource)(value.source)
  };
};
/**
 * 
 * @export
 * @interface SpacingToken
 */


exports.transformJSONToSpacingToken = transformJSONToSpacingToken;
