// @zeplin/sdk/dist/models/styleguide-component-version-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideComponentVersionCreatedEventToJSON = exports.transformJSONToStyleguideComponentVersionCreatedEvent = void 0;

var _projectComponentCreatedEventResource = require("./project-component-created-event-resource");

var _styleguideComponentEventContext = require("./styleguide-component-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideComponentVersionCreatedEventToJSON = function transformStyleguideComponentVersionCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectComponentCreatedEventResource.transformProjectComponentCreatedEventResourceToJSON)(value.resource),
    context: (0, _styleguideComponentEventContext.transformStyleguideComponentEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideComponentVersionCreatedEventToJSON = transformStyleguideComponentVersionCreatedEventToJSON;

var transformJSONToStyleguideComponentVersionCreatedEvent = function transformJSONToStyleguideComponentVersionCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectComponentCreatedEventResource.transformJSONToProjectComponentCreatedEventResource)(value.resource),
    context: (0, _styleguideComponentEventContext.transformJSONToStyleguideComponentEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideComponentVersionCreatedEvent
 */


exports.transformJSONToStyleguideComponentVersionCreatedEvent = transformJSONToStyleguideComponentVersionCreatedEvent;
