// @zeplin/sdk/dist/models/styleguide-member.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideMemberToJSON = exports.transformJSONToStyleguideMember = void 0;

var _user = require("./user");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideMemberToJSON = function transformStyleguideMemberToJSON(value) {
  return {
    user: (0, _user.transformUserToJSON)(value.user),
    role: value.role
  };
};

exports.transformStyleguideMemberToJSON = transformStyleguideMemberToJSON;

var transformJSONToStyleguideMember = function transformJSONToStyleguideMember(value) {
  return {
    user: (0, _user.transformJSONToUser)(value.user),
    role: value.role
  };
};
/**
 * 
 * @export
 * @interface StyleguideMember
 */


exports.transformJSONToStyleguideMember = transformJSONToStyleguideMember;
