// @zeplin/sdk/dist/models/styleguide-text-style-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideTextStyleCreatedEventToJSON = exports.transformJSONToStyleguideTextStyleCreatedEvent = void 0;

var _projectTextStyleCreatedEventResource = require("./project-text-style-created-event-resource");

var _styleguideTextStyleEventContext = require("./styleguide-text-style-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideTextStyleCreatedEventToJSON = function transformStyleguideTextStyleCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectTextStyleCreatedEventResource.transformProjectTextStyleCreatedEventResourceToJSON)(value.resource),
    context: (0, _styleguideTextStyleEventContext.transformStyleguideTextStyleEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideTextStyleCreatedEventToJSON = transformStyleguideTextStyleCreatedEventToJSON;

var transformJSONToStyleguideTextStyleCreatedEvent = function transformJSONToStyleguideTextStyleCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectTextStyleCreatedEventResource.transformJSONToProjectTextStyleCreatedEventResource)(value.resource),
    context: (0, _styleguideTextStyleEventContext.transformJSONToStyleguideTextStyleEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideTextStyleCreatedEvent
 */


exports.transformJSONToStyleguideTextStyleCreatedEvent = transformJSONToStyleguideTextStyleCreatedEvent;
