// @zeplin/sdk/dist/models/styleguide-component-event-context.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideComponentEventContextToJSON = exports.transformJSONToStyleguideComponentEventContext = void 0;

var _componentVersionSummary = require("./component-version-summary");

var _styleguide = require("./styleguide");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideComponentEventContextToJSON = function transformStyleguideComponentEventContextToJSON(value) {
  return {
    styleguide: (0, _styleguide.transformStyleguideToJSON)(value.styleguide),
    version: (0, _componentVersionSummary.transformComponentVersionSummaryToJSON)(value.version)
  };
};

exports.transformStyleguideComponentEventContextToJSON = transformStyleguideComponentEventContextToJSON;

var transformJSONToStyleguideComponentEventContext = function transformJSONToStyleguideComponentEventContext(value) {
  return {
    styleguide: (0, _styleguide.transformJSONToStyleguide)(value.styleguide),
    version: (0, _componentVersionSummary.transformJSONToComponentVersionSummary)(value.version)
  };
};
/**
 * 
 * @export
 * @interface StyleguideComponentEventContext
 */


exports.transformJSONToStyleguideComponentEventContext = transformJSONToStyleguideComponentEventContext;
