// @zeplin/sdk/dist/models/styleguide-spacing-token-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideSpacingTokenUpdatedEventToJSON = exports.transformJSONToStyleguideSpacingTokenUpdatedEvent = void 0;

var _projectSpacingTokenCreatedEventResource = require("./project-spacing-token-created-event-resource");

var _styleguideSpacingTokenEventContext = require("./styleguide-spacing-token-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideSpacingTokenUpdatedEventToJSON = function transformStyleguideSpacingTokenUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectSpacingTokenCreatedEventResource.transformProjectSpacingTokenCreatedEventResourceToJSON)(value.resource),
    context: (0, _styleguideSpacingTokenEventContext.transformStyleguideSpacingTokenEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideSpacingTokenUpdatedEventToJSON = transformStyleguideSpacingTokenUpdatedEventToJSON;

var transformJSONToStyleguideSpacingTokenUpdatedEvent = function transformJSONToStyleguideSpacingTokenUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectSpacingTokenCreatedEventResource.transformJSONToProjectSpacingTokenCreatedEventResource)(value.resource),
    context: (0, _styleguideSpacingTokenEventContext.transformJSONToStyleguideSpacingTokenEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideSpacingTokenUpdatedEvent
 */


exports.transformJSONToStyleguideSpacingTokenUpdatedEvent = transformJSONToStyleguideSpacingTokenUpdatedEvent;
