// @zeplin/sdk/dist/models/styleguide-color-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideColorDeletedEventToJSON = exports.transformJSONToStyleguideColorDeletedEvent = void 0;

var _projectColorDeletedEventResource = require("./project-color-deleted-event-resource");

var _styleguideColorEventContext = require("./styleguide-color-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideColorDeletedEventToJSON = function transformStyleguideColorDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectColorDeletedEventResource.transformProjectColorDeletedEventResourceToJSON)(value.resource),
    context: (0, _styleguideColorEventContext.transformStyleguideColorEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideColorDeletedEventToJSON = transformStyleguideColorDeletedEventToJSON;

var transformJSONToStyleguideColorDeletedEvent = function transformJSONToStyleguideColorDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectColorDeletedEventResource.transformJSONToProjectColorDeletedEventResource)(value.resource),
    context: (0, _styleguideColorEventContext.transformJSONToStyleguideColorEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideColorDeletedEvent
 */


exports.transformJSONToStyleguideColorDeletedEvent = transformJSONToStyleguideColorDeletedEvent;
