// @zeplin/sdk/dist/models/styleguide-member-role-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideMemberRoleUpdatedEventToJSON = exports.transformJSONToStyleguideMemberRoleUpdatedEvent = void 0;

var _styleguideMemberEventContext = require("./styleguide-member-event-context");

var _styleguideMemberInvitedEventResource = require("./styleguide-member-invited-event-resource");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideMemberRoleUpdatedEventToJSON = function transformStyleguideMemberRoleUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _styleguideMemberInvitedEventResource.transformStyleguideMemberInvitedEventResourceToJSON)(value.resource),
    context: (0, _styleguideMemberEventContext.transformStyleguideMemberEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideMemberRoleUpdatedEventToJSON = transformStyleguideMemberRoleUpdatedEventToJSON;

var transformJSONToStyleguideMemberRoleUpdatedEvent = function transformJSONToStyleguideMemberRoleUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _styleguideMemberInvitedEventResource.transformJSONToStyleguideMemberInvitedEventResource)(value.resource),
    context: (0, _styleguideMemberEventContext.transformJSONToStyleguideMemberEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideMemberRoleUpdatedEvent
 */


exports.transformJSONToStyleguideMemberRoleUpdatedEvent = transformJSONToStyleguideMemberRoleUpdatedEvent;
