// @zeplin/sdk/dist/models/styleguide-text-style-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideTextStyleUpdatedEventToJSON = exports.transformJSONToStyleguideTextStyleUpdatedEvent = void 0;

var _projectTextStyleCreatedEventResource = require("./project-text-style-created-event-resource");

var _styleguideTextStyleEventContext = require("./styleguide-text-style-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideTextStyleUpdatedEventToJSON = function transformStyleguideTextStyleUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectTextStyleCreatedEventResource.transformProjectTextStyleCreatedEventResourceToJSON)(value.resource),
    context: (0, _styleguideTextStyleEventContext.transformStyleguideTextStyleEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideTextStyleUpdatedEventToJSON = transformStyleguideTextStyleUpdatedEventToJSON;

var transformJSONToStyleguideTextStyleUpdatedEvent = function transformJSONToStyleguideTextStyleUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectTextStyleCreatedEventResource.transformJSONToProjectTextStyleCreatedEventResource)(value.resource),
    context: (0, _styleguideTextStyleEventContext.transformJSONToStyleguideTextStyleEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideTextStyleUpdatedEvent
 */


exports.transformJSONToStyleguideTextStyleUpdatedEvent = transformJSONToStyleguideTextStyleUpdatedEvent;
