// @zeplin/sdk/dist/models/styleguide-status-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideStatusEnumToJSON = exports.transformJSONToStyleguideStatusEnum = exports.StyleguideStatusEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideStatusEnumToJSON = function transformStyleguideStatusEnumToJSON(value) {
  return value;
};

exports.transformStyleguideStatusEnumToJSON = transformStyleguideStatusEnumToJSON;

var transformJSONToStyleguideStatusEnum = function transformJSONToStyleguideStatusEnum(value) {
  return value;
};
/**
 * The status of the styleguide
 * @export
 * @enum {string}
 */


exports.transformJSONToStyleguideStatusEnum = transformJSONToStyleguideStatusEnum;
var StyleguideStatusEnum;
exports.StyleguideStatusEnum = StyleguideStatusEnum;

(function (StyleguideStatusEnum) {
  StyleguideStatusEnum["ACTIVE"] = "active";
  StyleguideStatusEnum["ARCHIVED"] = "archived";
})(StyleguideStatusEnum || (exports.StyleguideStatusEnum = StyleguideStatusEnum = {}));
