// @zeplin/sdk/dist/models/styleguide-component-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideComponentDeletedEventToJSON = exports.transformJSONToStyleguideComponentDeletedEvent = void 0;

var _projectComponentDeletedEventResource = require("./project-component-deleted-event-resource");

var _styleguideComponentDeletedEventContext = require("./styleguide-component-deleted-event-context");

var _webhookEventActor = require("./webhook-event-actor");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideComponentDeletedEventToJSON = function transformStyleguideComponentDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectComponentDeletedEventResource.transformProjectComponentDeletedEventResourceToJSON)(value.resource),
    context: (0, _styleguideComponentDeletedEventContext.transformStyleguideComponentDeletedEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformStyleguideComponentDeletedEventToJSON = transformStyleguideComponentDeletedEventToJSON;

var transformJSONToStyleguideComponentDeletedEvent = function transformJSONToStyleguideComponentDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _projectComponentDeletedEventResource.transformJSONToProjectComponentDeletedEventResource)(value.resource),
    context: (0, _styleguideComponentDeletedEventContext.transformJSONToStyleguideComponentDeletedEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface StyleguideComponentDeletedEvent
 */


exports.transformJSONToStyleguideComponentDeletedEvent = transformJSONToStyleguideComponentDeletedEvent;
