// @zeplin/sdk/dist/models/styleguide-update-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideUpdateBodyToJSON = exports.transformJSONToStyleguideUpdateBody = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideUpdateBodyToJSON = function transformStyleguideUpdateBodyToJSON(value) {
  return {
    name: value.name,
    description: value.description
  };
};

exports.transformStyleguideUpdateBodyToJSON = transformStyleguideUpdateBodyToJSON;

var transformJSONToStyleguideUpdateBody = function transformJSONToStyleguideUpdateBody(value) {
  return {
    name: value.name,
    description: value.description
  };
};
/**
 * 
 * @export
 * @interface StyleguideUpdateBody
 */


exports.transformJSONToStyleguideUpdateBody = transformJSONToStyleguideUpdateBody;
