// @zeplin/sdk/dist/models/styleguide-member-removed-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideMemberRemovedEventResourceToJSON = exports.transformJSONToStyleguideMemberRemovedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideMemberRemovedEventResourceToJSON = function transformStyleguideMemberRemovedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformStyleguideMemberRemovedEventResourceToJSON = transformStyleguideMemberRemovedEventResourceToJSON;

var transformJSONToStyleguideMemberRemovedEventResource = function transformJSONToStyleguideMemberRemovedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface StyleguideMemberRemovedEventResource
 */


exports.transformJSONToStyleguideMemberRemovedEventResource = transformJSONToStyleguideMemberRemovedEventResource;
