// @zeplin/sdk/dist/models/styleguide-webhook-event-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformStyleguideWebhookEventEnumToJSON = exports.transformJSONToStyleguideWebhookEventEnum = exports.StyleguideWebhookEventEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformStyleguideWebhookEventEnumToJSON = function transformStyleguideWebhookEventEnumToJSON(value) {
  return value;
};

exports.transformStyleguideWebhookEventEnumToJSON = transformStyleguideWebhookEventEnumToJSON;

var transformJSONToStyleguideWebhookEventEnum = function transformJSONToStyleguideWebhookEventEnum(value) {
  return value;
};
/**
 * 
 * @export
 * @enum {string}
 */


exports.transformJSONToStyleguideWebhookEventEnum = transformJSONToStyleguideWebhookEventEnum;
var StyleguideWebhookEventEnum;
exports.StyleguideWebhookEventEnum = StyleguideWebhookEventEnum;

(function (StyleguideWebhookEventEnum) {
  StyleguideWebhookEventEnum["STAR"] = "styleguide.*";
  StyleguideWebhookEventEnum["COLOR"] = "styleguide.color";
  StyleguideWebhookEventEnum["COMPONENT"] = "styleguide.component";
  StyleguideWebhookEventEnum["MEMBER"] = "styleguide.member";
  StyleguideWebhookEventEnum["SPACING_TOKEN"] = "styleguide.spacing_token";
  StyleguideWebhookEventEnum["TEXT_STYLE"] = "styleguide.text_style";
})(StyleguideWebhookEventEnum || (exports.StyleguideWebhookEventEnum = StyleguideWebhookEventEnum = {}));
