// @zeplin/sdk/dist/models/text-style-design-token-metadata.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformTextStyleDesignTokenMetadataToJSON = exports.transformJSONToTextStyleDesignTokenMetadata = void 0;

var _designTokenResource = require("./design-token-resource");

var _designTokenSource = require("./design-token-source");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformTextStyleDesignTokenMetadataToJSON = function transformTextStyleDesignTokenMetadataToJSON(value) {
  return {
    source: (0, _designTokenSource.transformDesignTokenSourceToJSON)(value.source),
    resource: (0, _designTokenResource.transformDesignTokenResourceToJSON)(value.resource)
  };
};

exports.transformTextStyleDesignTokenMetadataToJSON = transformTextStyleDesignTokenMetadataToJSON;

var transformJSONToTextStyleDesignTokenMetadata = function transformJSONToTextStyleDesignTokenMetadata(value) {
  return {
    source: (0, _designTokenSource.transformJSONToDesignTokenSource)(value.source),
    resource: (0, _designTokenResource.transformJSONToDesignTokenResource)(value.resource)
  };
};
/**
 * Additional information about the text style token
 * @export
 * @interface TextStyleDesignTokenMetadata
 */


exports.transformJSONToTextStyleDesignTokenMetadata = transformJSONToTextStyleDesignTokenMetadata;
