// @zeplin/sdk/dist/models/text-style-update-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformTextStyleUpdateBodyToJSON = exports.transformJSONToTextStyleUpdateBody = void 0;

require("core-js/modules/es.function.name.js");

var _colorData = require("./color-data");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformTextStyleUpdateBodyToJSON = function transformTextStyleUpdateBodyToJSON(value) {
  return {
    name: value.name,
    color: value.color && (0, _colorData.transformColorDataToJSON)(value.color)
  };
};

exports.transformTextStyleUpdateBodyToJSON = transformTextStyleUpdateBodyToJSON;

var transformJSONToTextStyleUpdateBody = function transformJSONToTextStyleUpdateBody(value) {
  return {
    name: value.name,
    color: value.color && (0, _colorData.transformJSONToColorData)(value.color)
  };
};
/**
 * 
 * @export
 * @interface TextStyleUpdateBody
 */


exports.transformJSONToTextStyleUpdateBody = transformJSONToTextStyleUpdateBody;
