// @zeplin/sdk/dist/models/text-style-design-token-value.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformTextStyleDesignTokenValueToJSON = exports.transformJSONToTextStyleDesignTokenValue = void 0;

var _designTokenFont = require("./design-token-font");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformTextStyleDesignTokenValueToJSON = function transformTextStyleDesignTokenValueToJSON(value) {
  return {
    letter_spacing: value.letterSpacing,
    line_height: value.lineHeight,
    alignment: value.alignment,
    font: (0, _designTokenFont.transformDesignTokenFontToJSON)(value.font),
    color: value.color
  };
};

exports.transformTextStyleDesignTokenValueToJSON = transformTextStyleDesignTokenValueToJSON;

var transformJSONToTextStyleDesignTokenValue = function transformJSONToTextStyleDesignTokenValue(value) {
  return {
    letterSpacing: value.letter_spacing,
    lineHeight: value.line_height,
    alignment: value.alignment,
    font: (0, _designTokenFont.transformJSONToDesignTokenFont)(value.font),
    color: value.color
  };
};
/**
 * Value of the token
 * @export
 * @interface TextStyleDesignTokenValue
 */


exports.transformJSONToTextStyleDesignTokenValue = transformJSONToTextStyleDesignTokenValue;
