// @zeplin/sdk/dist/models/token-response.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformTokenResponseToJSON = exports.transformJSONToTokenResponse = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformTokenResponseToJSON = function transformTokenResponseToJSON(value) {
  return {
    access_token: value.accessToken,
    expires_in: value.expiresIn,
    refresh_token: value.refreshToken,
    refresh_expires_in: value.refreshExpiresIn,
    token_type: value.tokenType
  };
};

exports.transformTokenResponseToJSON = transformTokenResponseToJSON;

var transformJSONToTokenResponse = function transformJSONToTokenResponse(value) {
  return {
    accessToken: value.access_token,
    expiresIn: value.expires_in,
    refreshToken: value.refresh_token,
    refreshExpiresIn: value.refresh_expires_in,
    tokenType: value.token_type
  };
};
/**
 * 
 * @export
 * @interface TokenResponse
 */


exports.transformJSONToTokenResponse = transformJSONToTokenResponse;
