// @zeplin/sdk/dist/models/token-create-refresh-token-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformTokenCreateRefreshTokenBodyToJSON = exports.transformJSONToTokenCreateRefreshTokenBody = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformTokenCreateRefreshTokenBodyToJSON = function transformTokenCreateRefreshTokenBodyToJSON(value) {
  return {
    grant_type: value.grantType,
    refresh_token: value.refreshToken,
    client_id: value.clientId,
    client_secret: value.clientSecret,
    code_verifier: value.codeVerifier
  };
};

exports.transformTokenCreateRefreshTokenBodyToJSON = transformTokenCreateRefreshTokenBodyToJSON;

var transformJSONToTokenCreateRefreshTokenBody = function transformJSONToTokenCreateRefreshTokenBody(value) {
  return {
    grantType: value.grant_type,
    refreshToken: value.refresh_token,
    clientId: value.client_id,
    clientSecret: value.client_secret,
    codeVerifier: value.code_verifier
  };
};
/**
 * 
 * @export
 * @interface TokenCreateRefreshTokenBody
 */


exports.transformJSONToTokenCreateRefreshTokenBody = transformJSONToTokenCreateRefreshTokenBody;
