// @zeplin/sdk/dist/models/token-create-authorization-code-body.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformTokenCreateAuthorizationCodeBodyToJSON = exports.transformJSONToTokenCreateAuthorizationCodeBody = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformTokenCreateAuthorizationCodeBodyToJSON = function transformTokenCreateAuthorizationCodeBodyToJSON(value) {
  return {
    grant_type: value.grantType,
    code: value.code,
    redirect_uri: value.redirectUri,
    client_id: value.clientId,
    client_secret: value.clientSecret,
    code_verifier: value.codeVerifier
  };
};

exports.transformTokenCreateAuthorizationCodeBodyToJSON = transformTokenCreateAuthorizationCodeBodyToJSON;

var transformJSONToTokenCreateAuthorizationCodeBody = function transformJSONToTokenCreateAuthorizationCodeBody(value) {
  return {
    grantType: value.grant_type,
    code: value.code,
    redirectUri: value.redirect_uri,
    clientId: value.client_id,
    clientSecret: value.client_secret,
    codeVerifier: value.code_verifier
  };
};
/**
 * 
 * @export
 * @interface TokenCreateAuthorizationCodeBody
 */


exports.transformJSONToTokenCreateAuthorizationCodeBody = transformJSONToTokenCreateAuthorizationCodeBody;
