// @zeplin/sdk/dist/models/user-webhook-event-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformUserWebhookEventEnumToJSON = exports.transformJSONToUserWebhookEventEnum = exports.UserWebhookEventEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformUserWebhookEventEnumToJSON = function transformUserWebhookEventEnumToJSON(value) {
  return value;
};

exports.transformUserWebhookEventEnumToJSON = transformUserWebhookEventEnumToJSON;

var transformJSONToUserWebhookEventEnum = function transformJSONToUserWebhookEventEnum(value) {
  return value;
};
/**
 * 
 * @export
 * @enum {string}
 */


exports.transformJSONToUserWebhookEventEnum = transformJSONToUserWebhookEventEnum;
var UserWebhookEventEnum;
exports.UserWebhookEventEnum = UserWebhookEventEnum;

(function (UserWebhookEventEnum) {
  UserWebhookEventEnum["STAR"] = "*";
  UserWebhookEventEnum["PROJECT"] = "project.*";
  UserWebhookEventEnum["PROJECT_COLOR"] = "project.color";
  UserWebhookEventEnum["PROJECT_COMPONENT"] = "project.component";
  UserWebhookEventEnum["PROJECT_MEMBER"] = "project.member";
  UserWebhookEventEnum["PROJECT_NOTE"] = "project.note";
  UserWebhookEventEnum["PROJECT_NOTE_COMMENT"] = "project.note.comment";
  UserWebhookEventEnum["PROJECT_SCREEN"] = "project.screen";
  UserWebhookEventEnum["PROJECT_SCREEN_VERSION"] = "project.screen.version";
  UserWebhookEventEnum["PROJECT_SPACING_TOKEN"] = "project.spacing_token";
  UserWebhookEventEnum["PROJECT_TEXT_STYLE"] = "project.text_style";
  UserWebhookEventEnum["STYLEGUIDE"] = "styleguide.*";
  UserWebhookEventEnum["STYLEGUIDE_COLOR"] = "styleguide.color";
  UserWebhookEventEnum["STYLEGUIDE_COMPONENT"] = "styleguide.component";
  UserWebhookEventEnum["STYLEGUIDE_MEMBER"] = "styleguide.member";
  UserWebhookEventEnum["STYLEGUIDE_SPACING_TOKEN"] = "styleguide.spacing_token";
  UserWebhookEventEnum["STYLEGUIDE_TEXT_STYLE"] = "styleguide.text_style";
  UserWebhookEventEnum["WORKSPACE"] = "workspace.*";
  UserWebhookEventEnum["WORKSPACE_NOTIFICATION"] = "workspace.notification";
  UserWebhookEventEnum["WORKSPACE_PROJECT"] = "workspace.project";
  UserWebhookEventEnum["WORKSPACE_STYLEGUIDE"] = "workspace.styleguide";
})(UserWebhookEventEnum || (exports.UserWebhookEventEnum = UserWebhookEventEnum = {}));
