// @zeplin/sdk/dist/models/version-commit-color.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformVersionCommitColorToJSON = exports.transformJSONToVersionCommitColor = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformVersionCommitColorToJSON = function transformVersionCommitColorToJSON(value) {
  return {
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};

exports.transformVersionCommitColorToJSON = transformVersionCommitColorToJSON;

var transformJSONToVersionCommitColor = function transformJSONToVersionCommitColor(value) {
  return {
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};
/**
 * 
 * @export
 * @interface VersionCommitColor
 */


exports.transformJSONToVersionCommitColor = transformJSONToVersionCommitColor;
