// @zeplin/sdk/dist/models/version-commit.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformVersionCommitToJSON = exports.transformJSONToVersionCommit = void 0;

var _user = require("./user");

var _versionCommitColor = require("./version-commit-color");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformVersionCommitToJSON = function transformVersionCommitToJSON(value) {
  return {
    message: value.message,
    author: value.author && (0, _user.transformUserToJSON)(value.author),
    color: value.color && (0, _versionCommitColor.transformVersionCommitColorToJSON)(value.color)
  };
};

exports.transformVersionCommitToJSON = transformVersionCommitToJSON;

var transformJSONToVersionCommit = function transformJSONToVersionCommit(value) {
  return {
    message: value.message,
    author: value.author && (0, _user.transformJSONToUser)(value.author),
    color: value.color && (0, _versionCommitColor.transformJSONToVersionCommitColor)(value.color)
  };
};
/**
 * 
 * @export
 * @interface VersionCommit
 */


exports.transformJSONToVersionCommit = transformJSONToVersionCommit;
