// @zeplin/sdk/dist/models/vertical-grid.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformVerticalGridToJSON = exports.transformJSONToVerticalGrid = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformVerticalGridToJSON = function transformVerticalGridToJSON(value) {
  return {
    gutter_width: value.gutterWidth,
    column_width: value.columnWidth,
    number_of_cols: value.numberOfCols,
    gutters_on_outside: value.guttersOnOutside
  };
};

exports.transformVerticalGridToJSON = transformVerticalGridToJSON;

var transformJSONToVerticalGrid = function transformJSONToVerticalGrid(value) {
  return {
    gutterWidth: value.gutter_width,
    columnWidth: value.column_width,
    numberOfCols: value.number_of_cols,
    guttersOnOutside: value.gutters_on_outside
  };
};
/**
 * 
 * @export
 * @interface VerticalGrid
 */


exports.transformJSONToVerticalGrid = transformJSONToVerticalGrid;
