// @zeplin/sdk/dist/models/webhook-status-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWebhookStatusEnumToJSON = exports.transformJSONToWebhookStatusEnum = exports.WebhookStatusEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWebhookStatusEnumToJSON = function transformWebhookStatusEnumToJSON(value) {
  return value;
};

exports.transformWebhookStatusEnumToJSON = transformWebhookStatusEnumToJSON;

var transformJSONToWebhookStatusEnum = function transformJSONToWebhookStatusEnum(value) {
  return value;
};
/**
 * The status of the webhook
 * @export
 * @enum {string}
 */


exports.transformJSONToWebhookStatusEnum = transformJSONToWebhookStatusEnum;
var WebhookStatusEnum;
exports.WebhookStatusEnum = WebhookStatusEnum;

(function (WebhookStatusEnum) {
  WebhookStatusEnum["ACTIVE"] = "active";
  WebhookStatusEnum["INACTIVE"] = "inactive";
})(WebhookStatusEnum || (exports.WebhookStatusEnum = WebhookStatusEnum = {}));
