// @zeplin/sdk/dist/models/webhook-health-enum.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWebhookHealthEnumToJSON = exports.transformJSONToWebhookHealthEnum = exports.WebhookHealthEnum = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWebhookHealthEnumToJSON = function transformWebhookHealthEnumToJSON(value) {
  return value;
};

exports.transformWebhookHealthEnumToJSON = transformWebhookHealthEnumToJSON;

var transformJSONToWebhookHealthEnum = function transformJSONToWebhookHealthEnum(value) {
  return value;
};
/**
 * The health of a webhook URL
 * @export
 * @enum {string}
 */


exports.transformJSONToWebhookHealthEnum = transformJSONToWebhookHealthEnum;
var WebhookHealthEnum;
exports.WebhookHealthEnum = WebhookHealthEnum;

(function (WebhookHealthEnum) {
  WebhookHealthEnum["HEALTHY"] = "healthy";
  WebhookHealthEnum["UNHEALTHY"] = "unhealthy";
})(WebhookHealthEnum || (exports.WebhookHealthEnum = WebhookHealthEnum = {}));
