// @zeplin/sdk/dist/models/workflow-status.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkflowStatusToJSON = exports.transformJSONToWorkflowStatus = void 0;

require("core-js/modules/es.function.name.js");

var _workflowStatusColor = require("./workflow-status-color");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkflowStatusToJSON = function transformWorkflowStatusToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    color: (0, _workflowStatusColor.transformWorkflowStatusColorToJSON)(value.color)
  };
};

exports.transformWorkflowStatusToJSON = transformWorkflowStatusToJSON;

var transformJSONToWorkflowStatus = function transformJSONToWorkflowStatus(value) {
  return {
    id: value.id,
    name: value.name,
    color: (0, _workflowStatusColor.transformJSONToWorkflowStatusColor)(value.color)
  };
};
/**
 * 
 * @export
 * @interface WorkflowStatus
 */


exports.transformJSONToWorkflowStatus = transformJSONToWorkflowStatus;
