// @zeplin/sdk/dist/models/workspace-organization-member-tag-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceOrganizationMemberTagUpdatedEventToJSON = exports.transformJSONToWorkspaceOrganizationMemberTagUpdatedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceOrganizationMemberEventContext = require("./workspace-organization-member-event-context");

var _workspaceOrganizationMemberInvitedEventResource = require("./workspace-organization-member-invited-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceOrganizationMemberTagUpdatedEventToJSON = function transformWorkspaceOrganizationMemberTagUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceOrganizationMemberInvitedEventResource.transformWorkspaceOrganizationMemberInvitedEventResourceToJSON)(value.resource),
    context: (0, _workspaceOrganizationMemberEventContext.transformWorkspaceOrganizationMemberEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceOrganizationMemberTagUpdatedEventToJSON = transformWorkspaceOrganizationMemberTagUpdatedEventToJSON;

var transformJSONToWorkspaceOrganizationMemberTagUpdatedEvent = function transformJSONToWorkspaceOrganizationMemberTagUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceOrganizationMemberInvitedEventResource.transformJSONToWorkspaceOrganizationMemberInvitedEventResource)(value.resource),
    context: (0, _workspaceOrganizationMemberEventContext.transformJSONToWorkspaceOrganizationMemberEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceOrganizationMemberTagUpdatedEvent
 */


exports.transformJSONToWorkspaceOrganizationMemberTagUpdatedEvent = transformJSONToWorkspaceOrganizationMemberTagUpdatedEvent;
