// @zeplin/sdk/dist/models/workflow-status-color.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkflowStatusColorToJSON = exports.transformJSONToWorkflowStatusColor = void 0;

require("core-js/modules/es.function.name.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkflowStatusColorToJSON = function transformWorkflowStatusColorToJSON(value) {
  return {
    name: value.name,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};

exports.transformWorkflowStatusColorToJSON = transformWorkflowStatusColorToJSON;

var transformJSONToWorkflowStatusColor = function transformJSONToWorkflowStatusColor(value) {
  return {
    name: value.name,
    r: value.r,
    g: value.g,
    b: value.b,
    a: value.a
  };
};
/**
 * 
 * @export
 * @interface WorkflowStatusColor
 */


exports.transformJSONToWorkflowStatusColor = transformJSONToWorkflowStatusColor;
