// @zeplin/sdk/dist/models/workspace-styleguide-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceStyleguideDeletedEventToJSON = exports.transformJSONToWorkspaceStyleguideDeletedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceStyleguideDeletedEventResource = require("./workspace-styleguide-deleted-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceStyleguideDeletedEventToJSON = function transformWorkspaceStyleguideDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceStyleguideDeletedEventResource.transformWorkspaceStyleguideDeletedEventResourceToJSON)(value.resource),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceStyleguideDeletedEventToJSON = transformWorkspaceStyleguideDeletedEventToJSON;

var transformJSONToWorkspaceStyleguideDeletedEvent = function transformJSONToWorkspaceStyleguideDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceStyleguideDeletedEventResource.transformJSONToWorkspaceStyleguideDeletedEventResource)(value.resource),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceStyleguideDeletedEvent
 */


exports.transformJSONToWorkspaceStyleguideDeletedEvent = transformJSONToWorkspaceStyleguideDeletedEvent;
