// @zeplin/sdk/dist/models/workspace-styleguide-deleted-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceStyleguideDeletedEventResourceToJSON = exports.transformJSONToWorkspaceStyleguideDeletedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceStyleguideDeletedEventResourceToJSON = function transformWorkspaceStyleguideDeletedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformWorkspaceStyleguideDeletedEventResourceToJSON = transformWorkspaceStyleguideDeletedEventResourceToJSON;

var transformJSONToWorkspaceStyleguideDeletedEventResource = function transformJSONToWorkspaceStyleguideDeletedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface WorkspaceStyleguideDeletedEventResource
 */


exports.transformJSONToWorkspaceStyleguideDeletedEventResource = transformJSONToWorkspaceStyleguideDeletedEventResource;
