// @zeplin/sdk/dist/models/workspace-styleguide-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceStyleguideCreatedEventResourceToJSON = exports.transformJSONToWorkspaceStyleguideCreatedEventResource = void 0;

var _styleguide = require("./styleguide");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceStyleguideCreatedEventResourceToJSON = function transformWorkspaceStyleguideCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _styleguide.transformStyleguideToJSON)(value.data)
  };
};

exports.transformWorkspaceStyleguideCreatedEventResourceToJSON = transformWorkspaceStyleguideCreatedEventResourceToJSON;

var transformJSONToWorkspaceStyleguideCreatedEventResource = function transformJSONToWorkspaceStyleguideCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _styleguide.transformJSONToStyleguide)(value.data)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceStyleguideCreatedEventResource
 */


exports.transformJSONToWorkspaceStyleguideCreatedEventResource = transformJSONToWorkspaceStyleguideCreatedEventResource;
