// @zeplin/sdk/dist/models/workspace-notification-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceNotificationUpdatedEventToJSON = exports.transformJSONToWorkspaceNotificationUpdatedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceNotificationCreatedEventResource = require("./workspace-notification-created-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceNotificationUpdatedEventToJSON = function transformWorkspaceNotificationUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceNotificationCreatedEventResource.transformWorkspaceNotificationCreatedEventResourceToJSON)(value.resource),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceNotificationUpdatedEventToJSON = transformWorkspaceNotificationUpdatedEventToJSON;

var transformJSONToWorkspaceNotificationUpdatedEvent = function transformJSONToWorkspaceNotificationUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceNotificationCreatedEventResource.transformJSONToWorkspaceNotificationCreatedEventResource)(value.resource),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceNotificationUpdatedEvent
 */


exports.transformJSONToWorkspaceNotificationUpdatedEvent = transformJSONToWorkspaceNotificationUpdatedEvent;
