// @zeplin/sdk/dist/models/workspace-organization-member-invited-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceOrganizationMemberInvitedEventResourceToJSON = exports.transformJSONToWorkspaceOrganizationMemberInvitedEventResource = void 0;

var _organizationMember = require("./organization-member");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceOrganizationMemberInvitedEventResourceToJSON = function transformWorkspaceOrganizationMemberInvitedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _organizationMember.transformOrganizationMemberToJSON)(value.data)
  };
};

exports.transformWorkspaceOrganizationMemberInvitedEventResourceToJSON = transformWorkspaceOrganizationMemberInvitedEventResourceToJSON;

var transformJSONToWorkspaceOrganizationMemberInvitedEventResource = function transformJSONToWorkspaceOrganizationMemberInvitedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _organizationMember.transformJSONToOrganizationMember)(value.data)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceOrganizationMemberInvitedEventResource
 */


exports.transformJSONToWorkspaceOrganizationMemberInvitedEventResource = transformJSONToWorkspaceOrganizationMemberInvitedEventResource;
