// @zeplin/sdk/dist/models/workspace-organization-member-removed-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceOrganizationMemberRemovedEventResourceToJSON = exports.transformJSONToWorkspaceOrganizationMemberRemovedEventResource = void 0;

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceOrganizationMemberRemovedEventResourceToJSON = function transformWorkspaceOrganizationMemberRemovedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type
  };
};

exports.transformWorkspaceOrganizationMemberRemovedEventResourceToJSON = transformWorkspaceOrganizationMemberRemovedEventResourceToJSON;

var transformJSONToWorkspaceOrganizationMemberRemovedEventResource = function transformJSONToWorkspaceOrganizationMemberRemovedEventResource(value) {
  return {
    id: value.id,
    type: value.type
  };
};
/**
 * 
 * @export
 * @interface WorkspaceOrganizationMemberRemovedEventResource
 */


exports.transformJSONToWorkspaceOrganizationMemberRemovedEventResource = transformJSONToWorkspaceOrganizationMemberRemovedEventResource;
