// @zeplin/sdk/dist/models/workspace-project-deleted-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceProjectDeletedEventToJSON = exports.transformJSONToWorkspaceProjectDeletedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceProjectDeletedEventResource = require("./workspace-project-deleted-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceProjectDeletedEventToJSON = function transformWorkspaceProjectDeletedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceProjectDeletedEventResource.transformWorkspaceProjectDeletedEventResourceToJSON)(value.resource),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceProjectDeletedEventToJSON = transformWorkspaceProjectDeletedEventToJSON;

var transformJSONToWorkspaceProjectDeletedEvent = function transformJSONToWorkspaceProjectDeletedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceProjectDeletedEventResource.transformJSONToWorkspaceProjectDeletedEventResource)(value.resource),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceProjectDeletedEvent
 */


exports.transformJSONToWorkspaceProjectDeletedEvent = transformJSONToWorkspaceProjectDeletedEvent;
