// @zeplin/sdk/dist/models/workspace-organization-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceOrganizationUpdatedEventToJSON = exports.transformJSONToWorkspaceOrganizationUpdatedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceOrganizationUpdatedEventResource = require("./workspace-organization-updated-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceOrganizationUpdatedEventToJSON = function transformWorkspaceOrganizationUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceOrganizationUpdatedEventResource.transformWorkspaceOrganizationUpdatedEventResourceToJSON)(value.resource),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceOrganizationUpdatedEventToJSON = transformWorkspaceOrganizationUpdatedEventToJSON;

var transformJSONToWorkspaceOrganizationUpdatedEvent = function transformJSONToWorkspaceOrganizationUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceOrganizationUpdatedEventResource.transformJSONToWorkspaceOrganizationUpdatedEventResource)(value.resource),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceOrganizationUpdatedEvent
 */


exports.transformJSONToWorkspaceOrganizationUpdatedEvent = transformJSONToWorkspaceOrganizationUpdatedEvent;
