// @zeplin/sdk/dist/models/workspace-organization-updated-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceOrganizationUpdatedEventResourceToJSON = exports.transformJSONToWorkspaceOrganizationUpdatedEventResource = void 0;

var _organizationSummary = require("./organization-summary");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceOrganizationUpdatedEventResourceToJSON = function transformWorkspaceOrganizationUpdatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _organizationSummary.transformOrganizationSummaryToJSON)(value.data)
  };
};

exports.transformWorkspaceOrganizationUpdatedEventResourceToJSON = transformWorkspaceOrganizationUpdatedEventResourceToJSON;

var transformJSONToWorkspaceOrganizationUpdatedEventResource = function transformJSONToWorkspaceOrganizationUpdatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _organizationSummary.transformJSONToOrganizationSummary)(value.data)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceOrganizationUpdatedEventResource
 */


exports.transformJSONToWorkspaceOrganizationUpdatedEventResource = transformJSONToWorkspaceOrganizationUpdatedEventResource;
