// @zeplin/sdk/dist/models/workspace-organization-member-removed-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceOrganizationMemberRemovedEventToJSON = exports.transformJSONToWorkspaceOrganizationMemberRemovedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceOrganizationMemberEventContext = require("./workspace-organization-member-event-context");

var _workspaceOrganizationMemberRemovedEventResource = require("./workspace-organization-member-removed-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceOrganizationMemberRemovedEventToJSON = function transformWorkspaceOrganizationMemberRemovedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceOrganizationMemberRemovedEventResource.transformWorkspaceOrganizationMemberRemovedEventResourceToJSON)(value.resource),
    context: (0, _workspaceOrganizationMemberEventContext.transformWorkspaceOrganizationMemberEventContextToJSON)(value.context),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceOrganizationMemberRemovedEventToJSON = transformWorkspaceOrganizationMemberRemovedEventToJSON;

var transformJSONToWorkspaceOrganizationMemberRemovedEvent = function transformJSONToWorkspaceOrganizationMemberRemovedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceOrganizationMemberRemovedEventResource.transformJSONToWorkspaceOrganizationMemberRemovedEventResource)(value.resource),
    context: (0, _workspaceOrganizationMemberEventContext.transformJSONToWorkspaceOrganizationMemberEventContext)(value.context),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceOrganizationMemberRemovedEvent
 */


exports.transformJSONToWorkspaceOrganizationMemberRemovedEvent = transformJSONToWorkspaceOrganizationMemberRemovedEvent;
