// @zeplin/sdk/dist/models/workspace-project-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceProjectCreatedEventResourceToJSON = exports.transformJSONToWorkspaceProjectCreatedEventResource = void 0;

var _project = require("./project");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceProjectCreatedEventResourceToJSON = function transformWorkspaceProjectCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _project.transformProjectToJSON)(value.data)
  };
};

exports.transformWorkspaceProjectCreatedEventResourceToJSON = transformWorkspaceProjectCreatedEventResourceToJSON;

var transformJSONToWorkspaceProjectCreatedEventResource = function transformJSONToWorkspaceProjectCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _project.transformJSONToProject)(value.data)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceProjectCreatedEventResource
 */


exports.transformJSONToWorkspaceProjectCreatedEventResource = transformJSONToWorkspaceProjectCreatedEventResource;
