// @zeplin/sdk/dist/models/workspace-notification-created-event-resource.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceNotificationCreatedEventResourceToJSON = exports.transformJSONToWorkspaceNotificationCreatedEventResource = void 0;

var _notification = require("./notification");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceNotificationCreatedEventResourceToJSON = function transformWorkspaceNotificationCreatedEventResourceToJSON(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _notification.transformNotificationToJSON)(value.data)
  };
};

exports.transformWorkspaceNotificationCreatedEventResourceToJSON = transformWorkspaceNotificationCreatedEventResourceToJSON;

var transformJSONToWorkspaceNotificationCreatedEventResource = function transformJSONToWorkspaceNotificationCreatedEventResource(value) {
  return {
    id: value.id,
    type: value.type,
    data: (0, _notification.transformJSONToNotification)(value.data)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceNotificationCreatedEventResource
 */


exports.transformJSONToWorkspaceNotificationCreatedEventResource = transformJSONToWorkspaceNotificationCreatedEventResource;
