// @zeplin/sdk/dist/models/workspace-project-created-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceProjectCreatedEventToJSON = exports.transformJSONToWorkspaceProjectCreatedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceProjectCreatedEventResource = require("./workspace-project-created-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceProjectCreatedEventToJSON = function transformWorkspaceProjectCreatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceProjectCreatedEventResource.transformWorkspaceProjectCreatedEventResourceToJSON)(value.resource),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceProjectCreatedEventToJSON = transformWorkspaceProjectCreatedEventToJSON;

var transformJSONToWorkspaceProjectCreatedEvent = function transformJSONToWorkspaceProjectCreatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceProjectCreatedEventResource.transformJSONToWorkspaceProjectCreatedEventResource)(value.resource),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceProjectCreatedEvent
 */


exports.transformJSONToWorkspaceProjectCreatedEvent = transformJSONToWorkspaceProjectCreatedEvent;
