// @zeplin/sdk/dist/models/workspace-styleguide-updated-event.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformWorkspaceStyleguideUpdatedEventToJSON = exports.transformJSONToWorkspaceStyleguideUpdatedEvent = void 0;

var _webhookEventActor = require("./webhook-event-actor");

var _workspaceStyleguideCreatedEventResource = require("./workspace-styleguide-created-event-resource");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformWorkspaceStyleguideUpdatedEventToJSON = function transformWorkspaceStyleguideUpdatedEventToJSON(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceStyleguideCreatedEventResource.transformWorkspaceStyleguideCreatedEventResourceToJSON)(value.resource),
    actor: (0, _webhookEventActor.transformWebhookEventActorToJSON)(value.actor)
  };
};

exports.transformWorkspaceStyleguideUpdatedEventToJSON = transformWorkspaceStyleguideUpdatedEventToJSON;

var transformJSONToWorkspaceStyleguideUpdatedEvent = function transformJSONToWorkspaceStyleguideUpdatedEvent(value) {
  return {
    event: value.event,
    action: value.action,
    timestamp: value.timestamp,
    resource: (0, _workspaceStyleguideCreatedEventResource.transformJSONToWorkspaceStyleguideCreatedEventResource)(value.resource),
    actor: (0, _webhookEventActor.transformJSONToWebhookEventActor)(value.actor)
  };
};
/**
 * 
 * @export
 * @interface WorkspaceStyleguideUpdatedEvent
 */


exports.transformJSONToWorkspaceStyleguideUpdatedEvent = transformJSONToWorkspaceStyleguideUpdatedEvent;
