// @zeplin/sdk/dist/models/zeplin-application.js
"use strict";

require("core-js/modules/es.object.define-property.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformZeplinApplicationToJSON = exports.transformJSONToZeplinApplication = void 0;

require("core-js/modules/es.function.name.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.symbol.description.js");

/* tslint:disable */

/* eslint-disable */

/**
 * Zeplin API
 * Access your resources in Zeplin
 *
 * Contact: support@zeplin.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var transformZeplinApplicationToJSON = function transformZeplinApplicationToJSON(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    website: value.website,
    logo: value.logo
  };
};

exports.transformZeplinApplicationToJSON = transformZeplinApplicationToJSON;

var transformJSONToZeplinApplication = function transformJSONToZeplinApplication(value) {
  return {
    id: value.id,
    name: value.name,
    description: value.description,
    website: value.website,
    logo: value.logo
  };
};
/**
 * 
 * @export
 * @interface ZeplinApplication
 */


exports.transformJSONToZeplinApplication = transformJSONToZeplinApplication;
