// @zeplin/sdk/dist/apis/design-tokens-api.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

require("core-js/modules/es.object.define-property.js");

require("core-js/modules/es.object.keys.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.array.filter.js");

require("core-js/modules/es.object.get-own-property-descriptor.js");

require("core-js/modules/es.array.for-each.js");

require("core-js/modules/web.dom-collections.for-each.js");

require("core-js/modules/es.object.get-own-property-descriptors.js");

require("core-js/modules/es.object.define-properties.js");

require("core-js/modules/es.object.set-prototype-of.js");

require("core-js/modules/es.object.get-prototype-of.js");

require("core-js/modules/es.reflect.construct.js");

require("core-js/modules/es.object.create.js");

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.symbol.iterator.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DesignTokensApiFp = exports.DesignTokensApiAxiosParamCreator = exports.DesignTokensApi = void 0;

require("regenerator-runtime/runtime.js");

require("core-js/modules/es.regexp.exec.js");

require("core-js/modules/es.string.replace.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.object.to-string.js");

require("core-js/modules/es.string.iterator.js");

require("core-js/modules/web.dom-collections.iterator.js");

require("core-js/modules/web.url.js");

require("core-js/modules/es.promise.js");

var _axios = _interopRequireDefault(require("axios"));

var _common = require("../common");

var _base = require("../base");

var _models = require("../models");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

/**
 * DesignTokensApi - axios parameter creator
 * @export
 */
var DesignTokensApiAxiosParamCreator = function DesignTokensApiAxiosParamCreator(configuration) {
  return {
    /**
     * Fetch all design tokens of the project
     * @summary Get project design tokens
     * @param {string} projectId Project id
     * @param {boolean} [includeLinkedStyleguides] Whether to include linked styleguides or not
     * @param {'kebab' | 'snake' | 'camel' | 'pascal' | 'constant' | 'original'} [tokenNameCase] Case for token names
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectDesignTokens: function () {
      var _getProjectDesignTokens = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(projectId, includeLinkedStyleguides, tokenNameCase) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args = arguments;
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                options = _args.length > 3 && _args[3] !== undefined ? _args[3] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectDesignTokens', 'projectId', projectId);
                localVarPath = "/v1/projects/{project_id}/design_tokens".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context.next = 10;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 10:
                _context.next = 12;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 12:
                if (includeLinkedStyleguides !== undefined) {
                  localVarQueryParameter['include_linked_styleguides'] = includeLinkedStyleguides;
                }

                if (tokenNameCase !== undefined) {
                  localVarQueryParameter['token_name_case'] = tokenNameCase;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 18:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));

      function getProjectDesignTokens(_x, _x2, _x3) {
        return _getProjectDesignTokens.apply(this, arguments);
      }

      return getProjectDesignTokens;
    }(),

    /**
     * Fetch all design tokens of the styleguide
     * @summary Get styleguide design tokens
     * @param {string} styleguideId Styleguide id
     * @param {boolean} [includeLinkedStyleguides] Whether to include linked styleguides or not
     * @param {'kebab' | 'snake' | 'camel' | 'pascal' | 'constant' | 'original'} [tokenNameCase] Case for token names
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getStyleguideDesignTokens: function () {
      var _getStyleguideDesignTokens = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2(styleguideId, includeLinkedStyleguides, tokenNameCase) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args2 = arguments;
        return regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                options = _args2.length > 3 && _args2[3] !== undefined ? _args2[3] : {};
                // verify required parameter 'styleguideId' is not null or undefined
                (0, _common.assertParamExists)('getStyleguideDesignTokens', 'styleguideId', styleguideId);
                localVarPath = "/v1/styleguides/{styleguide_id}/design_tokens".replace("{".concat("styleguide_id", "}"), encodeURIComponent(String(styleguideId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context2.next = 10;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 10:
                _context2.next = 12;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 12:
                if (includeLinkedStyleguides !== undefined) {
                  localVarQueryParameter['include_linked_styleguides'] = includeLinkedStyleguides;
                }

                if (tokenNameCase !== undefined) {
                  localVarQueryParameter['token_name_case'] = tokenNameCase;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context2.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 18:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2);
      }));

      function getStyleguideDesignTokens(_x4, _x5, _x6) {
        return _getStyleguideDesignTokens.apply(this, arguments);
      }

      return getStyleguideDesignTokens;
    }()
  };
};
/**
 * DesignTokensApi - functional programming interface
 * @export
 */


exports.DesignTokensApiAxiosParamCreator = DesignTokensApiAxiosParamCreator;

var DesignTokensApiFp = function DesignTokensApiFp(configuration) {
  var localVarAxiosParamCreator = DesignTokensApiAxiosParamCreator(configuration);
  return {
    /**
     * Fetch all design tokens of the project
     * @summary Get project design tokens
     * @param {string} projectId Project id
     * @param {boolean} [includeLinkedStyleguides] Whether to include linked styleguides or not
     * @param {'kebab' | 'snake' | 'camel' | 'pascal' | 'constant' | 'original'} [tokenNameCase] Case for token names
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectDesignTokens: function getProjectDesignTokens(projectId, includeLinkedStyleguides, tokenNameCase, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _context3.next = 2;
                return localVarAxiosParamCreator.getProjectDesignTokens(projectId, includeLinkedStyleguides, tokenNameCase, options);

              case 2:
                localVarAxiosArgs = _context3.sent;
                return _context3.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3);
      }))();
    },

    /**
     * Fetch all design tokens of the styleguide
     * @summary Get styleguide design tokens
     * @param {string} styleguideId Styleguide id
     * @param {boolean} [includeLinkedStyleguides] Whether to include linked styleguides or not
     * @param {'kebab' | 'snake' | 'camel' | 'pascal' | 'constant' | 'original'} [tokenNameCase] Case for token names
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getStyleguideDesignTokens: function getStyleguideDesignTokens(styleguideId, includeLinkedStyleguides, tokenNameCase, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee4$(_context4) {
          while (1) {
            switch (_context4.prev = _context4.next) {
              case 0:
                _context4.next = 2;
                return localVarAxiosParamCreator.getStyleguideDesignTokens(styleguideId, includeLinkedStyleguides, tokenNameCase, options);

              case 2:
                localVarAxiosArgs = _context4.sent;
                return _context4.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context4.stop();
            }
          }
        }, _callee4);
      }))();
    }
  };
};
/**
 * Search parameters for getProjectDesignTokens operation in DesignTokensApi.
 * @export
 * @interface DesignTokensApiGetProjectDesignTokensSearchParams
 */


exports.DesignTokensApiFp = DesignTokensApiFp;

/**
 * DesignTokensApi - object-oriented interface
 * @export
 * @class DesignTokensApi
 * @extends {BaseAPI}
 */
var DesignTokensApi = /*#__PURE__*/function (_BaseAPI) {
  _inherits(DesignTokensApi, _BaseAPI);

  var _super = _createSuper(DesignTokensApi);

  function DesignTokensApi() {
    _classCallCheck(this, DesignTokensApi);

    return _super.apply(this, arguments);
  }

  _createClass(DesignTokensApi, [{
    key: "getProjectDesignTokens",
    value:
    /**
     * Fetch all design tokens of the project
     * @summary Get project design tokens
     * @param {string} projectId Project id
     * @param {DesignTokensApiGetProjectDesignTokensSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DesignTokensApi
     */
    function () {
      var _getProjectDesignTokens2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee5(projectId) {
        var searchParams,
            options,
            designTokensApiFp,
            request,
            response,
            _args5 = arguments;
        return regeneratorRuntime.wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                searchParams = _args5.length > 1 && _args5[1] !== undefined ? _args5[1] : {};
                options = _args5.length > 2 ? _args5[2] : undefined;
                designTokensApiFp = DesignTokensApiFp(this.configuration);
                _context5.next = 5;
                return designTokensApiFp.getProjectDesignTokens(projectId, searchParams.includeLinkedStyleguides, searchParams.tokenNameCase, options);

              case 5:
                request = _context5.sent;
                _context5.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context5.sent;
                return _context5.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToDesignTokens)(response.data)
                }));

              case 10:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5, this);
      }));

      function getProjectDesignTokens(_x7) {
        return _getProjectDesignTokens2.apply(this, arguments);
      }

      return getProjectDesignTokens;
    }()
    /**
     * Fetch all design tokens of the styleguide
     * @summary Get styleguide design tokens
     * @param {string} styleguideId Styleguide id
     * @param {DesignTokensApiGetStyleguideDesignTokensSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DesignTokensApi
     */

  }, {
    key: "getStyleguideDesignTokens",
    value: function () {
      var _getStyleguideDesignTokens2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee6(styleguideId) {
        var searchParams,
            options,
            designTokensApiFp,
            request,
            response,
            _args6 = arguments;
        return regeneratorRuntime.wrap(function _callee6$(_context6) {
          while (1) {
            switch (_context6.prev = _context6.next) {
              case 0:
                searchParams = _args6.length > 1 && _args6[1] !== undefined ? _args6[1] : {};
                options = _args6.length > 2 ? _args6[2] : undefined;
                designTokensApiFp = DesignTokensApiFp(this.configuration);
                _context6.next = 5;
                return designTokensApiFp.getStyleguideDesignTokens(styleguideId, searchParams.includeLinkedStyleguides, searchParams.tokenNameCase, options);

              case 5:
                request = _context6.sent;
                _context6.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context6.sent;
                return _context6.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToDesignTokens)(response.data)
                }));

              case 10:
              case "end":
                return _context6.stop();
            }
          }
        }, _callee6, this);
      }));

      function getStyleguideDesignTokens(_x8) {
        return _getStyleguideDesignTokens2.apply(this, arguments);
      }

      return getStyleguideDesignTokens;
    }()
  }]);

  return DesignTokensApi;
}(_base.BaseAPI);

exports.DesignTokensApi = DesignTokensApi;
