// @zeplin/sdk/dist/apis/flows-api.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

require("core-js/modules/es.object.define-property.js");

require("core-js/modules/es.object.keys.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.array.filter.js");

require("core-js/modules/es.object.get-own-property-descriptor.js");

require("core-js/modules/es.array.for-each.js");

require("core-js/modules/web.dom-collections.for-each.js");

require("core-js/modules/es.object.get-own-property-descriptors.js");

require("core-js/modules/es.object.define-properties.js");

require("core-js/modules/es.object.set-prototype-of.js");

require("core-js/modules/es.object.get-prototype-of.js");

require("core-js/modules/es.reflect.construct.js");

require("core-js/modules/es.object.create.js");

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.symbol.iterator.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FlowsApiFp = exports.FlowsApiAxiosParamCreator = exports.FlowsApi = void 0;

require("regenerator-runtime/runtime.js");

require("core-js/modules/es.regexp.exec.js");

require("core-js/modules/es.string.replace.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.object.to-string.js");

require("core-js/modules/es.string.iterator.js");

require("core-js/modules/web.dom-collections.iterator.js");

require("core-js/modules/web.url.js");

require("core-js/modules/es.array.map.js");

require("core-js/modules/es.promise.js");

var _axios = _interopRequireDefault(require("axios"));

var _common = require("../common");

var _base = require("../base");

var _models = require("../models");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

/**
 * FlowsApi - axios parameter creator
 * @export
 */
var FlowsApiAxiosParamCreator = function FlowsApiAxiosParamCreator(configuration) {
  return {
    /**
     * Get flow board details related to a project
     * @summary Get a single project flow board
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoard: function () {
      var _getProjectFlowBoard = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(projectId, flowBoardId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args = arguments;
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                options = _args.length > 2 && _args[2] !== undefined ? _args[2] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoard', 'projectId', projectId); // verify required parameter 'flowBoardId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoard', 'flowBoardId', flowBoardId);
                localVarPath = "/v1/projects/{project_id}/flow_boards/{flow_board_id}".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("flow_board_id", "}"), encodeURIComponent(String(flowBoardId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 17:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));

      function getProjectFlowBoard(_x, _x2) {
        return _getProjectFlowBoard.apply(this, arguments);
      }

      return getProjectFlowBoard;
    }(),

    /**
     * Get details of the project flow board connector
     * @summary Get a single project flow board connector
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} connectorId Board connector id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardConnector: function () {
      var _getProjectFlowBoardConnector = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2(projectId, flowBoardId, connectorId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args2 = arguments;
        return regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                options = _args2.length > 3 && _args2[3] !== undefined ? _args2[3] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoardConnector', 'projectId', projectId); // verify required parameter 'flowBoardId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardConnector', 'flowBoardId', flowBoardId); // verify required parameter 'connectorId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardConnector', 'connectorId', connectorId);
                localVarPath = "/v1/projects/{project_id}/flow_boards/{flow_board_id}/connectors/{connector_id}".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("flow_board_id", "}"), encodeURIComponent(String(flowBoardId))).replace("{".concat("connector_id", "}"), encodeURIComponent(String(connectorId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context2.next = 12;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 12:
                _context2.next = 14;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 14:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context2.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 18:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2);
      }));

      function getProjectFlowBoardConnector(_x3, _x4, _x5) {
        return _getProjectFlowBoardConnector.apply(this, arguments);
      }

      return getProjectFlowBoardConnector;
    }(),

    /**
     * List all connectors of the flow board in a project
     * @summary Get project flow board connectors
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} [startingNodeId] Starting node id
     * @param {string} [endingNodeId] Ending node id
     * @param {number} [limit] Pagination limit
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardConnectors: function () {
      var _getProjectFlowBoardConnectors = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3(projectId, flowBoardId, startingNodeId, endingNodeId, limit, offset) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args3 = arguments;
        return regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                options = _args3.length > 6 && _args3[6] !== undefined ? _args3[6] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoardConnectors', 'projectId', projectId); // verify required parameter 'flowBoardId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardConnectors', 'flowBoardId', flowBoardId);
                localVarPath = "/v1/projects/{project_id}/flow_boards/{flow_board_id}/connectors".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("flow_board_id", "}"), encodeURIComponent(String(flowBoardId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context3.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context3.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                if (startingNodeId !== undefined) {
                  localVarQueryParameter['starting_node_id'] = startingNodeId;
                }

                if (endingNodeId !== undefined) {
                  localVarQueryParameter['ending_node_id'] = endingNodeId;
                }

                if (limit !== undefined) {
                  localVarQueryParameter['limit'] = limit;
                }

                if (offset !== undefined) {
                  localVarQueryParameter['offset'] = offset;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context3.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 21:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3);
      }));

      function getProjectFlowBoardConnectors(_x6, _x7, _x8, _x9, _x10, _x11) {
        return _getProjectFlowBoardConnectors.apply(this, arguments);
      }

      return getProjectFlowBoardConnectors;
    }(),

    /**
     * List all flow board groups in a project.
     * @summary Get project flow board groups
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardGroups: function () {
      var _getProjectFlowBoardGroups = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4(projectId, flowBoardId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args4 = arguments;
        return regeneratorRuntime.wrap(function _callee4$(_context4) {
          while (1) {
            switch (_context4.prev = _context4.next) {
              case 0:
                options = _args4.length > 2 && _args4[2] !== undefined ? _args4[2] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoardGroups', 'projectId', projectId); // verify required parameter 'flowBoardId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardGroups', 'flowBoardId', flowBoardId);
                localVarPath = "/v1/projects/{project_id}/flow_boards/{flow_board_id}/groups".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("flow_board_id", "}"), encodeURIComponent(String(flowBoardId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context4.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context4.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context4.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 17:
              case "end":
                return _context4.stop();
            }
          }
        }, _callee4);
      }));

      function getProjectFlowBoardGroups(_x12, _x13) {
        return _getProjectFlowBoardGroups.apply(this, arguments);
      }

      return getProjectFlowBoardGroups;
    }(),

    /**
     * Get details of the project flow board node
     * @summary Get a single project flow board node
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} nodeId Board node id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardNode: function () {
      var _getProjectFlowBoardNode = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee5(projectId, flowBoardId, nodeId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args5 = arguments;
        return regeneratorRuntime.wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                options = _args5.length > 3 && _args5[3] !== undefined ? _args5[3] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoardNode', 'projectId', projectId); // verify required parameter 'flowBoardId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardNode', 'flowBoardId', flowBoardId); // verify required parameter 'nodeId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardNode', 'nodeId', nodeId);
                localVarPath = "/v1/projects/{project_id}/flow_boards/{flow_board_id}/nodes/{node_id}".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("flow_board_id", "}"), encodeURIComponent(String(flowBoardId))).replace("{".concat("node_id", "}"), encodeURIComponent(String(nodeId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context5.next = 12;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 12:
                _context5.next = 14;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 14:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context5.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 18:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5);
      }));

      function getProjectFlowBoardNode(_x14, _x15, _x16) {
        return _getProjectFlowBoardNode.apply(this, arguments);
      }

      return getProjectFlowBoardNode;
    }(),

    /**
     * List all nodes of the flow board in a project
     * @summary Get project flow board nodes
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} [groupId] Group id
     * @param {number} [limit] Pagination limit
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardNodes: function () {
      var _getProjectFlowBoardNodes = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee6(projectId, flowBoardId, groupId, limit, offset) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args6 = arguments;
        return regeneratorRuntime.wrap(function _callee6$(_context6) {
          while (1) {
            switch (_context6.prev = _context6.next) {
              case 0:
                options = _args6.length > 5 && _args6[5] !== undefined ? _args6[5] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoardNodes', 'projectId', projectId); // verify required parameter 'flowBoardId' is not null or undefined

                (0, _common.assertParamExists)('getProjectFlowBoardNodes', 'flowBoardId', flowBoardId);
                localVarPath = "/v1/projects/{project_id}/flow_boards/{flow_board_id}/nodes".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("flow_board_id", "}"), encodeURIComponent(String(flowBoardId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context6.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context6.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                if (groupId !== undefined) {
                  localVarQueryParameter['group_id'] = groupId;
                }

                if (limit !== undefined) {
                  localVarQueryParameter['limit'] = limit;
                }

                if (offset !== undefined) {
                  localVarQueryParameter['offset'] = offset;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context6.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 20:
              case "end":
                return _context6.stop();
            }
          }
        }, _callee6);
      }));

      function getProjectFlowBoardNodes(_x17, _x18, _x19, _x20, _x21) {
        return _getProjectFlowBoardNodes.apply(this, arguments);
      }

      return getProjectFlowBoardNodes;
    }(),

    /**
     * List all flow boards in a project.
     * @summary Get project flow boards
     * @param {string} projectId Project id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoards: function () {
      var _getProjectFlowBoards = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee7(projectId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args7 = arguments;
        return regeneratorRuntime.wrap(function _callee7$(_context7) {
          while (1) {
            switch (_context7.prev = _context7.next) {
              case 0:
                options = _args7.length > 1 && _args7[1] !== undefined ? _args7[1] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectFlowBoards', 'projectId', projectId);
                localVarPath = "/v1/projects/{project_id}/flow_boards".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context7.next = 10;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 10:
                _context7.next = 12;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 12:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context7.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 16:
              case "end":
                return _context7.stop();
            }
          }
        }, _callee7);
      }));

      function getProjectFlowBoards(_x22) {
        return _getProjectFlowBoards.apply(this, arguments);
      }

      return getProjectFlowBoards;
    }()
  };
};
/**
 * FlowsApi - functional programming interface
 * @export
 */


exports.FlowsApiAxiosParamCreator = FlowsApiAxiosParamCreator;

var FlowsApiFp = function FlowsApiFp(configuration) {
  var localVarAxiosParamCreator = FlowsApiAxiosParamCreator(configuration);
  return {
    /**
     * Get flow board details related to a project
     * @summary Get a single project flow board
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoard: function getProjectFlowBoard(projectId, flowBoardId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee8() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee8$(_context8) {
          while (1) {
            switch (_context8.prev = _context8.next) {
              case 0:
                _context8.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoard(projectId, flowBoardId, options);

              case 2:
                localVarAxiosArgs = _context8.sent;
                return _context8.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context8.stop();
            }
          }
        }, _callee8);
      }))();
    },

    /**
     * Get details of the project flow board connector
     * @summary Get a single project flow board connector
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} connectorId Board connector id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardConnector: function getProjectFlowBoardConnector(projectId, flowBoardId, connectorId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee9() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee9$(_context9) {
          while (1) {
            switch (_context9.prev = _context9.next) {
              case 0:
                _context9.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoardConnector(projectId, flowBoardId, connectorId, options);

              case 2:
                localVarAxiosArgs = _context9.sent;
                return _context9.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context9.stop();
            }
          }
        }, _callee9);
      }))();
    },

    /**
     * List all connectors of the flow board in a project
     * @summary Get project flow board connectors
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} [startingNodeId] Starting node id
     * @param {string} [endingNodeId] Ending node id
     * @param {number} [limit] Pagination limit
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardConnectors: function getProjectFlowBoardConnectors(projectId, flowBoardId, startingNodeId, endingNodeId, limit, offset, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee10() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee10$(_context10) {
          while (1) {
            switch (_context10.prev = _context10.next) {
              case 0:
                _context10.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoardConnectors(projectId, flowBoardId, startingNodeId, endingNodeId, limit, offset, options);

              case 2:
                localVarAxiosArgs = _context10.sent;
                return _context10.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context10.stop();
            }
          }
        }, _callee10);
      }))();
    },

    /**
     * List all flow board groups in a project.
     * @summary Get project flow board groups
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardGroups: function getProjectFlowBoardGroups(projectId, flowBoardId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee11() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee11$(_context11) {
          while (1) {
            switch (_context11.prev = _context11.next) {
              case 0:
                _context11.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoardGroups(projectId, flowBoardId, options);

              case 2:
                localVarAxiosArgs = _context11.sent;
                return _context11.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context11.stop();
            }
          }
        }, _callee11);
      }))();
    },

    /**
     * Get details of the project flow board node
     * @summary Get a single project flow board node
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} nodeId Board node id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardNode: function getProjectFlowBoardNode(projectId, flowBoardId, nodeId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee12() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee12$(_context12) {
          while (1) {
            switch (_context12.prev = _context12.next) {
              case 0:
                _context12.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoardNode(projectId, flowBoardId, nodeId, options);

              case 2:
                localVarAxiosArgs = _context12.sent;
                return _context12.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context12.stop();
            }
          }
        }, _callee12);
      }))();
    },

    /**
     * List all nodes of the flow board in a project
     * @summary Get project flow board nodes
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} [groupId] Group id
     * @param {number} [limit] Pagination limit
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoardNodes: function getProjectFlowBoardNodes(projectId, flowBoardId, groupId, limit, offset, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee13() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee13$(_context13) {
          while (1) {
            switch (_context13.prev = _context13.next) {
              case 0:
                _context13.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoardNodes(projectId, flowBoardId, groupId, limit, offset, options);

              case 2:
                localVarAxiosArgs = _context13.sent;
                return _context13.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context13.stop();
            }
          }
        }, _callee13);
      }))();
    },

    /**
     * List all flow boards in a project.
     * @summary Get project flow boards
     * @param {string} projectId Project id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectFlowBoards: function getProjectFlowBoards(projectId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee14() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee14$(_context14) {
          while (1) {
            switch (_context14.prev = _context14.next) {
              case 0:
                _context14.next = 2;
                return localVarAxiosParamCreator.getProjectFlowBoards(projectId, options);

              case 2:
                localVarAxiosArgs = _context14.sent;
                return _context14.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context14.stop();
            }
          }
        }, _callee14);
      }))();
    }
  };
};
/**
 * Search parameters for getProjectFlowBoardConnectors operation in FlowsApi.
 * @export
 * @interface FlowsApiGetProjectFlowBoardConnectorsSearchParams
 */


exports.FlowsApiFp = FlowsApiFp;

/**
 * FlowsApi - object-oriented interface
 * @export
 * @class FlowsApi
 * @extends {BaseAPI}
 */
var FlowsApi = /*#__PURE__*/function (_BaseAPI) {
  _inherits(FlowsApi, _BaseAPI);

  var _super = _createSuper(FlowsApi);

  function FlowsApi() {
    _classCallCheck(this, FlowsApi);

    return _super.apply(this, arguments);
  }

  _createClass(FlowsApi, [{
    key: "getProjectFlowBoard",
    value:
    /**
     * Get flow board details related to a project
     * @summary Get a single project flow board
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */
    function () {
      var _getProjectFlowBoard2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee15(projectId, flowBoardId, options) {
        var flowsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee15$(_context15) {
          while (1) {
            switch (_context15.prev = _context15.next) {
              case 0:
                flowsApiFp = FlowsApiFp(this.configuration);
                _context15.next = 3;
                return flowsApiFp.getProjectFlowBoard(projectId, flowBoardId, options);

              case 3:
                request = _context15.sent;
                _context15.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context15.sent;
                return _context15.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToFlowBoard)(response.data)
                }));

              case 8:
              case "end":
                return _context15.stop();
            }
          }
        }, _callee15, this);
      }));

      function getProjectFlowBoard(_x23, _x24, _x25) {
        return _getProjectFlowBoard2.apply(this, arguments);
      }

      return getProjectFlowBoard;
    }()
    /**
     * Get details of the project flow board connector
     * @summary Get a single project flow board connector
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} connectorId Board connector id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */

  }, {
    key: "getProjectFlowBoardConnector",
    value: function () {
      var _getProjectFlowBoardConnector2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee16(projectId, flowBoardId, connectorId, options) {
        var flowsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee16$(_context16) {
          while (1) {
            switch (_context16.prev = _context16.next) {
              case 0:
                flowsApiFp = FlowsApiFp(this.configuration);
                _context16.next = 3;
                return flowsApiFp.getProjectFlowBoardConnector(projectId, flowBoardId, connectorId, options);

              case 3:
                request = _context16.sent;
                _context16.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context16.sent;
                return _context16.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToFlowBoardConnector)(response.data)
                }));

              case 8:
              case "end":
                return _context16.stop();
            }
          }
        }, _callee16, this);
      }));

      function getProjectFlowBoardConnector(_x26, _x27, _x28, _x29) {
        return _getProjectFlowBoardConnector2.apply(this, arguments);
      }

      return getProjectFlowBoardConnector;
    }()
    /**
     * List all connectors of the flow board in a project
     * @summary Get project flow board connectors
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {FlowsApiGetProjectFlowBoardConnectorsSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */

  }, {
    key: "getProjectFlowBoardConnectors",
    value: function () {
      var _getProjectFlowBoardConnectors2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee17(projectId, flowBoardId) {
        var searchParams,
            options,
            flowsApiFp,
            request,
            response,
            _args17 = arguments;
        return regeneratorRuntime.wrap(function _callee17$(_context17) {
          while (1) {
            switch (_context17.prev = _context17.next) {
              case 0:
                searchParams = _args17.length > 2 && _args17[2] !== undefined ? _args17[2] : {};
                options = _args17.length > 3 ? _args17[3] : undefined;
                flowsApiFp = FlowsApiFp(this.configuration);
                _context17.next = 5;
                return flowsApiFp.getProjectFlowBoardConnectors(projectId, flowBoardId, searchParams.startingNodeId, searchParams.endingNodeId, searchParams.limit, searchParams.offset, options);

              case 5:
                request = _context17.sent;
                _context17.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context17.sent;
                return _context17.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: response.data.map(_models.transformJSONToFlowBoardConnector)
                }));

              case 10:
              case "end":
                return _context17.stop();
            }
          }
        }, _callee17, this);
      }));

      function getProjectFlowBoardConnectors(_x30, _x31) {
        return _getProjectFlowBoardConnectors2.apply(this, arguments);
      }

      return getProjectFlowBoardConnectors;
    }()
    /**
     * List all flow board groups in a project.
     * @summary Get project flow board groups
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */

  }, {
    key: "getProjectFlowBoardGroups",
    value: function () {
      var _getProjectFlowBoardGroups2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee18(projectId, flowBoardId, options) {
        var flowsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee18$(_context18) {
          while (1) {
            switch (_context18.prev = _context18.next) {
              case 0:
                flowsApiFp = FlowsApiFp(this.configuration);
                _context18.next = 3;
                return flowsApiFp.getProjectFlowBoardGroups(projectId, flowBoardId, options);

              case 3:
                request = _context18.sent;
                _context18.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context18.sent;
                return _context18.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: response.data.map(_models.transformJSONToFlowBoardGroup)
                }));

              case 8:
              case "end":
                return _context18.stop();
            }
          }
        }, _callee18, this);
      }));

      function getProjectFlowBoardGroups(_x32, _x33, _x34) {
        return _getProjectFlowBoardGroups2.apply(this, arguments);
      }

      return getProjectFlowBoardGroups;
    }()
    /**
     * Get details of the project flow board node
     * @summary Get a single project flow board node
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {string} nodeId Board node id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */

  }, {
    key: "getProjectFlowBoardNode",
    value: function () {
      var _getProjectFlowBoardNode2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee19(projectId, flowBoardId, nodeId, options) {
        var flowsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee19$(_context19) {
          while (1) {
            switch (_context19.prev = _context19.next) {
              case 0:
                flowsApiFp = FlowsApiFp(this.configuration);
                _context19.next = 3;
                return flowsApiFp.getProjectFlowBoardNode(projectId, flowBoardId, nodeId, options);

              case 3:
                request = _context19.sent;
                _context19.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context19.sent;
                return _context19.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToFlowBoardNode)(response.data)
                }));

              case 8:
              case "end":
                return _context19.stop();
            }
          }
        }, _callee19, this);
      }));

      function getProjectFlowBoardNode(_x35, _x36, _x37, _x38) {
        return _getProjectFlowBoardNode2.apply(this, arguments);
      }

      return getProjectFlowBoardNode;
    }()
    /**
     * List all nodes of the flow board in a project
     * @summary Get project flow board nodes
     * @param {string} projectId Project id
     * @param {string} flowBoardId Flow Board id
     * @param {FlowsApiGetProjectFlowBoardNodesSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */

  }, {
    key: "getProjectFlowBoardNodes",
    value: function () {
      var _getProjectFlowBoardNodes2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee20(projectId, flowBoardId) {
        var searchParams,
            options,
            flowsApiFp,
            request,
            response,
            _args20 = arguments;
        return regeneratorRuntime.wrap(function _callee20$(_context20) {
          while (1) {
            switch (_context20.prev = _context20.next) {
              case 0:
                searchParams = _args20.length > 2 && _args20[2] !== undefined ? _args20[2] : {};
                options = _args20.length > 3 ? _args20[3] : undefined;
                flowsApiFp = FlowsApiFp(this.configuration);
                _context20.next = 5;
                return flowsApiFp.getProjectFlowBoardNodes(projectId, flowBoardId, searchParams.groupId, searchParams.limit, searchParams.offset, options);

              case 5:
                request = _context20.sent;
                _context20.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context20.sent;
                return _context20.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: response.data.map(_models.transformJSONToFlowBoardNode)
                }));

              case 10:
              case "end":
                return _context20.stop();
            }
          }
        }, _callee20, this);
      }));

      function getProjectFlowBoardNodes(_x39, _x40) {
        return _getProjectFlowBoardNodes2.apply(this, arguments);
      }

      return getProjectFlowBoardNodes;
    }()
    /**
     * List all flow boards in a project.
     * @summary Get project flow boards
     * @param {string} projectId Project id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof FlowsApi
     */

  }, {
    key: "getProjectFlowBoards",
    value: function () {
      var _getProjectFlowBoards2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee21(projectId, options) {
        var flowsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee21$(_context21) {
          while (1) {
            switch (_context21.prev = _context21.next) {
              case 0:
                flowsApiFp = FlowsApiFp(this.configuration);
                _context21.next = 3;
                return flowsApiFp.getProjectFlowBoards(projectId, options);

              case 3:
                request = _context21.sent;
                _context21.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context21.sent;
                return _context21.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: response.data.map(_models.transformJSONToFlowBoard)
                }));

              case 8:
              case "end":
                return _context21.stop();
            }
          }
        }, _callee21, this);
      }));

      function getProjectFlowBoards(_x41, _x42) {
        return _getProjectFlowBoards2.apply(this, arguments);
      }

      return getProjectFlowBoards;
    }()
  }]);

  return FlowsApi;
}(_base.BaseAPI);

exports.FlowsApi = FlowsApi;
