// @zeplin/sdk/dist/apis/projects-api.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

require("core-js/modules/es.object.define-property.js");

require("core-js/modules/es.object.keys.js");

require("core-js/modules/es.symbol.js");

require("core-js/modules/es.array.filter.js");

require("core-js/modules/es.object.get-own-property-descriptor.js");

require("core-js/modules/es.array.for-each.js");

require("core-js/modules/web.dom-collections.for-each.js");

require("core-js/modules/es.object.get-own-property-descriptors.js");

require("core-js/modules/es.object.define-properties.js");

require("core-js/modules/es.object.set-prototype-of.js");

require("core-js/modules/es.object.get-prototype-of.js");

require("core-js/modules/es.reflect.construct.js");

require("core-js/modules/es.object.create.js");

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.symbol.iterator.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProjectsApiFp = exports.ProjectsApiAxiosParamCreator = exports.ProjectsApi = void 0;

require("regenerator-runtime/runtime.js");

require("core-js/modules/es.regexp.exec.js");

require("core-js/modules/es.string.replace.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.object.to-string.js");

require("core-js/modules/es.string.iterator.js");

require("core-js/modules/web.dom-collections.iterator.js");

require("core-js/modules/web.url.js");

require("core-js/modules/es.array.map.js");

require("core-js/modules/es.promise.js");

var _axios = _interopRequireDefault(require("axios"));

var _common = require("../common");

var _base = require("../base");

var _models = require("../models");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } else if (call !== void 0) { throw new TypeError("Derived constructors may only return object or undefined"); } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

/**
 * ProjectsApi - axios parameter creator
 * @export
 */
var ProjectsApiAxiosParamCreator = function ProjectsApiAxiosParamCreator(configuration) {
  return {
    /**
     * Get a project by id
     * @summary Get a single project
     * @param {string} projectId Project id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProject: function () {
      var _getProject = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee(projectId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args = arguments;
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                options = _args.length > 1 && _args[1] !== undefined ? _args[1] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProject', 'projectId', projectId);
                localVarPath = "/v1/projects/{project_id}".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context.next = 10;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 10:
                _context.next = 12;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 12:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 16:
              case "end":
                return _context.stop();
            }
          }
        }, _callee);
      }));

      function getProject(_x) {
        return _getProject.apply(this, arguments);
      }

      return getProject;
    }(),

    /**
     * Fetch all design tokens of the project
     * @summary Get project design tokens
     * @param {string} projectId Project id
     * @param {boolean} [includeLinkedStyleguides] Whether to include linked styleguides or not
     * @param {'kebab' | 'snake' | 'camel' | 'pascal' | 'constant' | 'original'} [tokenNameCase] Case for token names
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectDesignTokens: function () {
      var _getProjectDesignTokens = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2(projectId, includeLinkedStyleguides, tokenNameCase) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args2 = arguments;
        return regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                options = _args2.length > 3 && _args2[3] !== undefined ? _args2[3] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectDesignTokens', 'projectId', projectId);
                localVarPath = "/v1/projects/{project_id}/design_tokens".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context2.next = 10;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 10:
                _context2.next = 12;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 12:
                if (includeLinkedStyleguides !== undefined) {
                  localVarQueryParameter['include_linked_styleguides'] = includeLinkedStyleguides;
                }

                if (tokenNameCase !== undefined) {
                  localVarQueryParameter['token_name_case'] = tokenNameCase;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context2.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 18:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2);
      }));

      function getProjectDesignTokens(_x2, _x3, _x4) {
        return _getProjectDesignTokens.apply(this, arguments);
      }

      return getProjectDesignTokens;
    }(),

    /**
     * List all members of the project
     * @summary Get project members
     * @param {string} projectId Project id
     * @param {number} [limit] Pagination limit
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectMembers: function () {
      var _getProjectMembers = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3(projectId, limit, offset) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args3 = arguments;
        return regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                options = _args3.length > 3 && _args3[3] !== undefined ? _args3[3] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('getProjectMembers', 'projectId', projectId);
                localVarPath = "/v1/projects/{project_id}/members".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context3.next = 10;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 10:
                _context3.next = 12;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 12:
                if (limit !== undefined) {
                  localVarQueryParameter['limit'] = limit;
                }

                if (offset !== undefined) {
                  localVarQueryParameter['offset'] = offset;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context3.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 18:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3);
      }));

      function getProjectMembers(_x5, _x6, _x7) {
        return _getProjectMembers.apply(this, arguments);
      }

      return getProjectMembers;
    }(),

    /**
     * List all projects that user is a member of
     * @summary Get all projects
     * @param {number} [limit] Pagination limit
     * @param {string} [workspace] Workspace of the project, it can be &#x60;personal&#x60; or the id of organization
     * @param {ProjectStatusEnum} [status] Filter by status
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjects: function () {
      var _getProjects = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee4(limit, workspace, status, offset) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args4 = arguments;
        return regeneratorRuntime.wrap(function _callee4$(_context4) {
          while (1) {
            switch (_context4.prev = _context4.next) {
              case 0:
                options = _args4.length > 4 && _args4[4] !== undefined ? _args4[4] : {};
                localVarPath = "/v1/projects"; // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'GET'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context4.next = 9;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 9:
                _context4.next = 11;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 11:
                if (limit !== undefined) {
                  localVarQueryParameter['limit'] = limit;
                }

                if (workspace !== undefined) {
                  localVarQueryParameter['workspace'] = workspace;
                }

                if (status !== undefined) {
                  localVarQueryParameter['status'] = status;
                }

                if (offset !== undefined) {
                  localVarQueryParameter['offset'] = offset;
                }

                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context4.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 19:
              case "end":
                return _context4.stop();
            }
          }
        }, _callee4);
      }));

      function getProjects(_x8, _x9, _x10, _x11) {
        return _getProjects.apply(this, arguments);
      }

      return getProjects;
    }(),

    /**
     * Invite a member to the project.
     * @summary Invite a member
     * @param {string} projectId Project id
     * @param {ProjectMemberInviteBody} projectMemberInviteBody 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    inviteProjectMember: function () {
      var _inviteProjectMember = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee5(projectId, projectMemberInviteBody) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args5 = arguments;
        return regeneratorRuntime.wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                options = _args5.length > 2 && _args5[2] !== undefined ? _args5[2] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('inviteProjectMember', 'projectId', projectId); // verify required parameter 'projectMemberInviteBody' is not null or undefined

                (0, _common.assertParamExists)('inviteProjectMember', 'projectMemberInviteBody', projectMemberInviteBody);
                localVarPath = "/v1/projects/{project_id}/members".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'POST'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context5.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context5.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                localVarHeaderParameter['Content-Type'] = 'application/json';
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                localVarRequestOptions.data = (0, _common.serializeDataIfNeeded)((0, _models.transformProjectMemberInviteBodyToJSON)(projectMemberInviteBody), localVarRequestOptions, configuration);
                return _context5.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 19:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5);
      }));

      function inviteProjectMember(_x12, _x13) {
        return _inviteProjectMember.apply(this, arguments);
      }

      return inviteProjectMember;
    }(),

    /**
     * Remove a member from the project.
     * @summary Remove a member
     * @param {string} projectId Project id
     * @param {string} memberId Member id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    removeProjectMember: function () {
      var _removeProjectMember = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee6(projectId, memberId) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args6 = arguments;
        return regeneratorRuntime.wrap(function _callee6$(_context6) {
          while (1) {
            switch (_context6.prev = _context6.next) {
              case 0:
                options = _args6.length > 2 && _args6[2] !== undefined ? _args6[2] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('removeProjectMember', 'projectId', projectId); // verify required parameter 'memberId' is not null or undefined

                (0, _common.assertParamExists)('removeProjectMember', 'memberId', memberId);
                localVarPath = "/v1/projects/{project_id}/members/{member_id}".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))).replace("{".concat("member_id", "}"), encodeURIComponent(String(memberId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'DELETE'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context6.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context6.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                return _context6.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 17:
              case "end":
                return _context6.stop();
            }
          }
        }, _callee6);
      }));

      function removeProjectMember(_x14, _x15) {
        return _removeProjectMember.apply(this, arguments);
      }

      return removeProjectMember;
    }(),

    /**
     * Update a project\'s name and description
     * @summary Update a project
     * @param {string} projectId Project id
     * @param {ProjectUpdateBody} projectUpdateBody 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateProject: function () {
      var _updateProject = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee7(projectId, projectUpdateBody) {
        var options,
            localVarPath,
            localVarUrlObj,
            baseOptions,
            localVarRequestOptions,
            localVarHeaderParameter,
            localVarQueryParameter,
            headersFromBaseOptions,
            _args7 = arguments;
        return regeneratorRuntime.wrap(function _callee7$(_context7) {
          while (1) {
            switch (_context7.prev = _context7.next) {
              case 0:
                options = _args7.length > 2 && _args7[2] !== undefined ? _args7[2] : {};
                // verify required parameter 'projectId' is not null or undefined
                (0, _common.assertParamExists)('updateProject', 'projectId', projectId); // verify required parameter 'projectUpdateBody' is not null or undefined

                (0, _common.assertParamExists)('updateProject', 'projectUpdateBody', projectUpdateBody);
                localVarPath = "/v1/projects/{project_id}".replace("{".concat("project_id", "}"), encodeURIComponent(String(projectId))); // use dummy base URL string because the URL constructor only accepts absolute URLs.

                localVarUrlObj = new URL(localVarPath, _common.DUMMY_BASE_URL);

                if (configuration) {
                  baseOptions = configuration.baseOptions;
                }

                localVarRequestOptions = _objectSpread(_objectSpread({
                  method: 'PATCH'
                }, baseOptions), options);
                localVarHeaderParameter = {};
                localVarQueryParameter = {}; // authentication OAuth2 required
                // oauth required

                _context7.next = 11;
                return (0, _common.setOAuthToObject)(localVarHeaderParameter, "OAuth2", [], configuration);

              case 11:
                _context7.next = 13;
                return (0, _common.setBearerAuthToObject)(localVarHeaderParameter, configuration);

              case 13:
                localVarHeaderParameter['Content-Type'] = 'application/json';
                (0, _common.setSearchParams)(localVarUrlObj, localVarQueryParameter, options.query);
                headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
                localVarRequestOptions.headers = _objectSpread(_objectSpread(_objectSpread({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
                localVarRequestOptions.data = (0, _common.serializeDataIfNeeded)((0, _models.transformProjectUpdateBodyToJSON)(projectUpdateBody), localVarRequestOptions, configuration);
                return _context7.abrupt("return", {
                  url: (0, _common.toPathString)(localVarUrlObj),
                  options: localVarRequestOptions
                });

              case 19:
              case "end":
                return _context7.stop();
            }
          }
        }, _callee7);
      }));

      function updateProject(_x16, _x17) {
        return _updateProject.apply(this, arguments);
      }

      return updateProject;
    }()
  };
};
/**
 * ProjectsApi - functional programming interface
 * @export
 */


exports.ProjectsApiAxiosParamCreator = ProjectsApiAxiosParamCreator;

var ProjectsApiFp = function ProjectsApiFp(configuration) {
  var localVarAxiosParamCreator = ProjectsApiAxiosParamCreator(configuration);
  return {
    /**
     * Get a project by id
     * @summary Get a single project
     * @param {string} projectId Project id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProject: function getProject(projectId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee8() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee8$(_context8) {
          while (1) {
            switch (_context8.prev = _context8.next) {
              case 0:
                _context8.next = 2;
                return localVarAxiosParamCreator.getProject(projectId, options);

              case 2:
                localVarAxiosArgs = _context8.sent;
                return _context8.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context8.stop();
            }
          }
        }, _callee8);
      }))();
    },

    /**
     * Fetch all design tokens of the project
     * @summary Get project design tokens
     * @param {string} projectId Project id
     * @param {boolean} [includeLinkedStyleguides] Whether to include linked styleguides or not
     * @param {'kebab' | 'snake' | 'camel' | 'pascal' | 'constant' | 'original'} [tokenNameCase] Case for token names
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectDesignTokens: function getProjectDesignTokens(projectId, includeLinkedStyleguides, tokenNameCase, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee9() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee9$(_context9) {
          while (1) {
            switch (_context9.prev = _context9.next) {
              case 0:
                _context9.next = 2;
                return localVarAxiosParamCreator.getProjectDesignTokens(projectId, includeLinkedStyleguides, tokenNameCase, options);

              case 2:
                localVarAxiosArgs = _context9.sent;
                return _context9.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context9.stop();
            }
          }
        }, _callee9);
      }))();
    },

    /**
     * List all members of the project
     * @summary Get project members
     * @param {string} projectId Project id
     * @param {number} [limit] Pagination limit
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjectMembers: function getProjectMembers(projectId, limit, offset, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee10() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee10$(_context10) {
          while (1) {
            switch (_context10.prev = _context10.next) {
              case 0:
                _context10.next = 2;
                return localVarAxiosParamCreator.getProjectMembers(projectId, limit, offset, options);

              case 2:
                localVarAxiosArgs = _context10.sent;
                return _context10.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context10.stop();
            }
          }
        }, _callee10);
      }))();
    },

    /**
     * List all projects that user is a member of
     * @summary Get all projects
     * @param {number} [limit] Pagination limit
     * @param {string} [workspace] Workspace of the project, it can be &#x60;personal&#x60; or the id of organization
     * @param {ProjectStatusEnum} [status] Filter by status
     * @param {number} [offset] Pagination offset
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    getProjects: function getProjects(limit, workspace, status, offset, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee11() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee11$(_context11) {
          while (1) {
            switch (_context11.prev = _context11.next) {
              case 0:
                _context11.next = 2;
                return localVarAxiosParamCreator.getProjects(limit, workspace, status, offset, options);

              case 2:
                localVarAxiosArgs = _context11.sent;
                return _context11.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context11.stop();
            }
          }
        }, _callee11);
      }))();
    },

    /**
     * Invite a member to the project.
     * @summary Invite a member
     * @param {string} projectId Project id
     * @param {ProjectMemberInviteBody} projectMemberInviteBody 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    inviteProjectMember: function inviteProjectMember(projectId, projectMemberInviteBody, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee12() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee12$(_context12) {
          while (1) {
            switch (_context12.prev = _context12.next) {
              case 0:
                _context12.next = 2;
                return localVarAxiosParamCreator.inviteProjectMember(projectId, projectMemberInviteBody, options);

              case 2:
                localVarAxiosArgs = _context12.sent;
                return _context12.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context12.stop();
            }
          }
        }, _callee12);
      }))();
    },

    /**
     * Remove a member from the project.
     * @summary Remove a member
     * @param {string} projectId Project id
     * @param {string} memberId Member id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    removeProjectMember: function removeProjectMember(projectId, memberId, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee13() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee13$(_context13) {
          while (1) {
            switch (_context13.prev = _context13.next) {
              case 0:
                _context13.next = 2;
                return localVarAxiosParamCreator.removeProjectMember(projectId, memberId, options);

              case 2:
                localVarAxiosArgs = _context13.sent;
                return _context13.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context13.stop();
            }
          }
        }, _callee13);
      }))();
    },

    /**
     * Update a project\'s name and description
     * @summary Update a project
     * @param {string} projectId Project id
     * @param {ProjectUpdateBody} projectUpdateBody 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    updateProject: function updateProject(projectId, projectUpdateBody, options) {
      return _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee14() {
        var localVarAxiosArgs;
        return regeneratorRuntime.wrap(function _callee14$(_context14) {
          while (1) {
            switch (_context14.prev = _context14.next) {
              case 0:
                _context14.next = 2;
                return localVarAxiosParamCreator.updateProject(projectId, projectUpdateBody, options);

              case 2:
                localVarAxiosArgs = _context14.sent;
                return _context14.abrupt("return", (0, _common.createRequestFunction)(localVarAxiosArgs, _axios["default"], _base.BASE_PATH, configuration));

              case 4:
              case "end":
                return _context14.stop();
            }
          }
        }, _callee14);
      }))();
    }
  };
};
/**
 * Search parameters for getProjectDesignTokens operation in ProjectsApi.
 * @export
 * @interface ProjectsApiGetProjectDesignTokensSearchParams
 */


exports.ProjectsApiFp = ProjectsApiFp;

/**
 * ProjectsApi - object-oriented interface
 * @export
 * @class ProjectsApi
 * @extends {BaseAPI}
 */
var ProjectsApi = /*#__PURE__*/function (_BaseAPI) {
  _inherits(ProjectsApi, _BaseAPI);

  var _super = _createSuper(ProjectsApi);

  function ProjectsApi() {
    _classCallCheck(this, ProjectsApi);

    return _super.apply(this, arguments);
  }

  _createClass(ProjectsApi, [{
    key: "getProject",
    value:
    /**
     * Get a project by id
     * @summary Get a single project
     * @param {string} projectId Project id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */
    function () {
      var _getProject2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee15(projectId, options) {
        var projectsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee15$(_context15) {
          while (1) {
            switch (_context15.prev = _context15.next) {
              case 0:
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context15.next = 3;
                return projectsApiFp.getProject(projectId, options);

              case 3:
                request = _context15.sent;
                _context15.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context15.sent;
                return _context15.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToProject)(response.data)
                }));

              case 8:
              case "end":
                return _context15.stop();
            }
          }
        }, _callee15, this);
      }));

      function getProject(_x18, _x19) {
        return _getProject2.apply(this, arguments);
      }

      return getProject;
    }()
    /**
     * Fetch all design tokens of the project
     * @summary Get project design tokens
     * @param {string} projectId Project id
     * @param {ProjectsApiGetProjectDesignTokensSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */

  }, {
    key: "getProjectDesignTokens",
    value: function () {
      var _getProjectDesignTokens2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee16(projectId) {
        var searchParams,
            options,
            projectsApiFp,
            request,
            response,
            _args16 = arguments;
        return regeneratorRuntime.wrap(function _callee16$(_context16) {
          while (1) {
            switch (_context16.prev = _context16.next) {
              case 0:
                searchParams = _args16.length > 1 && _args16[1] !== undefined ? _args16[1] : {};
                options = _args16.length > 2 ? _args16[2] : undefined;
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context16.next = 5;
                return projectsApiFp.getProjectDesignTokens(projectId, searchParams.includeLinkedStyleguides, searchParams.tokenNameCase, options);

              case 5:
                request = _context16.sent;
                _context16.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context16.sent;
                return _context16.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: (0, _models.transformJSONToDesignTokens)(response.data)
                }));

              case 10:
              case "end":
                return _context16.stop();
            }
          }
        }, _callee16, this);
      }));

      function getProjectDesignTokens(_x20) {
        return _getProjectDesignTokens2.apply(this, arguments);
      }

      return getProjectDesignTokens;
    }()
    /**
     * List all members of the project
     * @summary Get project members
     * @param {string} projectId Project id
     * @param {ProjectsApiGetProjectMembersSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */

  }, {
    key: "getProjectMembers",
    value: function () {
      var _getProjectMembers2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee17(projectId) {
        var searchParams,
            options,
            projectsApiFp,
            request,
            response,
            _args17 = arguments;
        return regeneratorRuntime.wrap(function _callee17$(_context17) {
          while (1) {
            switch (_context17.prev = _context17.next) {
              case 0:
                searchParams = _args17.length > 1 && _args17[1] !== undefined ? _args17[1] : {};
                options = _args17.length > 2 ? _args17[2] : undefined;
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context17.next = 5;
                return projectsApiFp.getProjectMembers(projectId, searchParams.limit, searchParams.offset, options);

              case 5:
                request = _context17.sent;
                _context17.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context17.sent;
                return _context17.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: response.data.map(_models.transformJSONToProjectMember)
                }));

              case 10:
              case "end":
                return _context17.stop();
            }
          }
        }, _callee17, this);
      }));

      function getProjectMembers(_x21) {
        return _getProjectMembers2.apply(this, arguments);
      }

      return getProjectMembers;
    }()
    /**
     * List all projects that user is a member of
     * @summary Get all projects
     * @param {ProjectsApiGetProjectsSearchParams} [searchParams] Search parameters.
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */

  }, {
    key: "getProjects",
    value: function () {
      var _getProjects2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee18() {
        var searchParams,
            options,
            projectsApiFp,
            request,
            response,
            _args18 = arguments;
        return regeneratorRuntime.wrap(function _callee18$(_context18) {
          while (1) {
            switch (_context18.prev = _context18.next) {
              case 0:
                searchParams = _args18.length > 0 && _args18[0] !== undefined ? _args18[0] : {};
                options = _args18.length > 1 ? _args18[1] : undefined;
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context18.next = 5;
                return projectsApiFp.getProjects(searchParams.limit, searchParams.workspace, searchParams.status, searchParams.offset, options);

              case 5:
                request = _context18.sent;
                _context18.next = 8;
                return request(this.axios, this.basePath);

              case 8:
                response = _context18.sent;
                return _context18.abrupt("return", _objectSpread(_objectSpread({}, response), {}, {
                  data: response.data.map(_models.transformJSONToProject)
                }));

              case 10:
              case "end":
                return _context18.stop();
            }
          }
        }, _callee18, this);
      }));

      function getProjects() {
        return _getProjects2.apply(this, arguments);
      }

      return getProjects;
    }()
    /**
     * Invite a member to the project.
     * @summary Invite a member
     * @param {string} projectId Project id
     * @param {ProjectMemberInviteBody} projectMemberInviteBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */

  }, {
    key: "inviteProjectMember",
    value: function () {
      var _inviteProjectMember2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee19(projectId, projectMemberInviteBody, options) {
        var projectsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee19$(_context19) {
          while (1) {
            switch (_context19.prev = _context19.next) {
              case 0:
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context19.next = 3;
                return projectsApiFp.inviteProjectMember(projectId, projectMemberInviteBody, options);

              case 3:
                request = _context19.sent;
                _context19.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context19.sent;
                return _context19.abrupt("return", response);

              case 8:
              case "end":
                return _context19.stop();
            }
          }
        }, _callee19, this);
      }));

      function inviteProjectMember(_x22, _x23, _x24) {
        return _inviteProjectMember2.apply(this, arguments);
      }

      return inviteProjectMember;
    }()
    /**
     * Remove a member from the project.
     * @summary Remove a member
     * @param {string} projectId Project id
     * @param {string} memberId Member id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */

  }, {
    key: "removeProjectMember",
    value: function () {
      var _removeProjectMember2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee20(projectId, memberId, options) {
        var projectsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee20$(_context20) {
          while (1) {
            switch (_context20.prev = _context20.next) {
              case 0:
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context20.next = 3;
                return projectsApiFp.removeProjectMember(projectId, memberId, options);

              case 3:
                request = _context20.sent;
                _context20.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context20.sent;
                return _context20.abrupt("return", response);

              case 8:
              case "end":
                return _context20.stop();
            }
          }
        }, _callee20, this);
      }));

      function removeProjectMember(_x25, _x26, _x27) {
        return _removeProjectMember2.apply(this, arguments);
      }

      return removeProjectMember;
    }()
    /**
     * Update a project\'s name and description
     * @summary Update a project
     * @param {string} projectId Project id
     * @param {ProjectUpdateBody} projectUpdateBody
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ProjectsApi
     */

  }, {
    key: "updateProject",
    value: function () {
      var _updateProject2 = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee21(projectId, projectUpdateBody, options) {
        var projectsApiFp, request, response;
        return regeneratorRuntime.wrap(function _callee21$(_context21) {
          while (1) {
            switch (_context21.prev = _context21.next) {
              case 0:
                projectsApiFp = ProjectsApiFp(this.configuration);
                _context21.next = 3;
                return projectsApiFp.updateProject(projectId, projectUpdateBody, options);

              case 3:
                request = _context21.sent;
                _context21.next = 6;
                return request(this.axios, this.basePath);

              case 6:
                response = _context21.sent;
                return _context21.abrupt("return", response);

              case 8:
              case "end":
                return _context21.stop();
            }
          }
        }, _callee21, this);
      }));

      function updateProject(_x28, _x29, _x30) {
        return _updateProject2.apply(this, arguments);
      }

      return updateProject;
    }()
  }]);

  return ProjectsApi;
}(_base.BaseAPI);

exports.ProjectsApi = ProjectsApi;
