// storybook-zeplin/dist/utils/messenger.js
import { PARENT_ORIGIN } from "../constants";
const ADDON_SOURCE_NAME = "storybook-zeplin-addon";
class Messenger {
    constructor() {
        this.responderMap = new Map();
        window.addEventListener("message", async ({ data, origin }) => {
            const action = data === null || data === void 0 ? void 0 : data.action;
            if (origin === PARENT_ORIGIN && (data === null || data === void 0 ? void 0 : data.to) === ADDON_SOURCE_NAME && this.responderMap.has(action)) {
                try {
                    const response = await this.responderMap.get(action)(data);
                    this.postMessage(action, response);
                }
                catch (e) {
                    this.postError(action, {
                        message: (e === null || e === void 0 ? void 0 : e.message) || e || "Unknown error"
                    });
                }
            }
        });
    }
    postMessage(action, payload) {
        window.parent.postMessage({
            source: ADDON_SOURCE_NAME,
            action,
            payload
        }, PARENT_ORIGIN);
    }
    postError(action, error) {
        window.parent.postMessage({
            source: ADDON_SOURCE_NAME,
            action,
            error
        }, PARENT_ORIGIN);
    }
    respondOnMessage(action, responder) {
        this.responderMap.set(action, responder);
    }
}
export const messenger = new Messenger();
