// @storybook/test-runner/dist/cjs/playwright/transformPlaywright.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transformPlaywright = exports.testPrefixer = void 0;

var _path = require("path");

var _template = _interopRequireDefault(require("@babel/template"));

var _store = require("@storybook/store");

var _util = require("../util");

var _transformCsf = require("../csf/transformCsf");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const filePrefixer = (0, _template.default)(`
  import global from 'global';
  const { setupPage } = require('@storybook/test-runner');
`);
const testPrefixer = (0, _template.default)(`
    console.log({ id: %%id%%, title: %%title%%, name: %%name%%, storyExport: %%storyExport%% });
    async () => {
      const testFn = async() => {
        const context = { id: %%id%%, title: %%title%%, name: %%name%% };

        page.on('pageerror', (err) => {
          page.evaluate(({ id, err }) => __throwError(id, err), { id: %%id%%, err: err.message });
        });

        if(global.__sbPreRender) {
          await global.__sbPreRender(page, context);
        }

        const result = await page.evaluate(({ id, hasPlayFn }) => __test(id, hasPlayFn), {
          id: %%id%%,
        });
  
        if(global.__sbPostRender) {
          await global.__sbPostRender(page, context);
        }

        return result;
      };

      try {
        await testFn();
      } catch(err) {
        if(err.toString().includes('Execution context was destroyed')) {
          await jestPlaywright.resetPage();
          await setupPage(global.page);
          await testFn();
        } else {
          throw err;
        }
      }
    }
  `, {
  plugins: ['jsx']
});
exports.testPrefixer = testPrefixer;

const getDefaultTitle = filename => {
  const {
    workingDir,
    normalizedStoriesEntries
  } = (0, _util.getStorybookMetadata)();
  const filePath = './' + (0, _path.relative)(workingDir, filename);
  return (0, _store.autoTitle)(filePath, normalizedStoriesEntries);
};

const transformPlaywright = (src, filename) => {
  const defaultTitle = getDefaultTitle(filename);
  const result = (0, _transformCsf.transformCsf)(src, {
    filePrefixer,
    // @ts-ignore
    testPrefixer,
    insertTestIfEmpty: true,
    clearBody: true,
    defaultTitle
  });
  return result;
};

exports.transformPlaywright = transformPlaywright;
